{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "**The purpose of this notebook is to generate the following datasets per ontology**: \n",
    "\n",
    "**TRAIN**: Used for training the PFP models \n",
    "**DEV**: Used for tuning model architecture and hyperparameters beta and gamma. \n",
    "**TEST**: Use for evaluation and benchmarking \n",
    "**PLASMODIUM HOLDOUT**: Used for evaluating the model on manually curated Plamsodium functions and set FDR thresholds \n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/opt/anaconda3/lib/python3.9/site-packages/scipy/__init__.py:146: UserWarning: A NumPy version >=1.16.5 and <1.23.0 is required for this version of SciPy (detected version 1.23.5\n",
      "  warnings.warn(f\"A NumPy version >={np_minversion} and <{np_maxversion}\"\n"
     ]
    }
   ],
   "source": [
    "import numpy as np\n",
    "import pandas as pd\n",
    "import os \n",
    "import networkx as nx\n",
    "import obonet\n",
    "from Bio.SeqRecord import SeqRecord\n",
    "from Bio.Seq import Seq\n",
    "from Bio import SeqIO\n",
    "from collections import defaultdict\n",
    "from sklearn.model_selection import ShuffleSplit\n",
    "from sklearn.preprocessing import MultiLabelBinarizer\n",
    "from tqdm import tqdm\n",
    "from collections import Counter\n",
    "import subprocess\n",
    "import requests, sys\n",
    "import json\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "#set seed for reproducibility\n",
    "np.random.seed(42)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Read in the raw datasets"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "SAR_MA_AA: (791306, 9)\n",
      "Plasmodium_AA: (102355, 9)\n",
      "Plasmodium_MA: (1527, 9)\n",
      "SAR_MA: (11957, 9)\n"
     ]
    }
   ],
   "source": [
    "SAR_MA_AA_path = \"/Users/harsh/Documents/PFP_PUBLISH/raw_data_from_uniprot/SAR_MA_AA.tsv\"\n",
    "Plasmodium_AA_path = \"/Users/harsh/Documents/PFP_PUBLISH/raw_data_from_uniprot/Plasmodium_AA.tsv\"\n",
    "Plasmodium_MA_path = \"/Users/harsh/Documents/PFP_PUBLISH/raw_data_from_uniprot/Plasmodium_MA.tsv\"\n",
    "SAR_MA_path = \"/Users/harsh/Documents/PFP_PUBLISH/raw_data_from_uniprot/SAR_MA.tsv\"\n",
    "\n",
    "#read in the datasets\n",
    "SAR_MA_AA = pd.read_csv(SAR_MA_AA_path, sep=\"\\t\")\n",
    "Plasmodium_AA = pd.read_csv(Plasmodium_AA_path, sep=\"\\t\")\n",
    "Plasmodium_MA = pd.read_csv(Plasmodium_MA_path, sep=\"\\t\")\n",
    "SAR_MA = pd.read_csv(SAR_MA_path, sep=\"\\t\")\n",
    "print(\"SAR_MA_AA:\", SAR_MA_AA.shape)\n",
    "print(\"Plasmodium_AA:\", Plasmodium_AA.shape)\n",
    "print(\"Plasmodium_MA:\", Plasmodium_MA.shape)\n",
    "print(\"SAR_MA:\", SAR_MA.shape)\n",
    "\n",
    "#assert columns are the same\n",
    "assert SAR_MA_AA.columns.tolist() == Plasmodium_AA.columns.tolist() == Plasmodium_MA.columns.tolist() == SAR_MA.columns.tolist()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "os.chdir('./raw_data_from_uniprot')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [],
   "source": [
    "def create_seq_records(data):\n",
    "    seq_records = []\n",
    "    for index, row in data.iterrows():\n",
    "        seq_record = SeqRecord(Seq(row['Sequence']), id=row['Entry'], description=row['Organism'])\n",
    "        seq_records.append(seq_record)\n",
    "    assert len(seq_records) == data.shape[0]\n",
    "    return seq_records"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "#first create a column in dataset1 called GOAssertion \n",
    "SAR_MA_AA[\"GOAssertion\"] = np.nan \n",
    "Plasmodium_AA[\"GOAssertion\"] = \"AA\"\n",
    "\n",
    "SAR_MA_AA.loc[SAR_MA_AA[\"Entry\"].isin(SAR_MA[\"Entry\"]), \"GOAssertion\"] = \"MA\"\n",
    "SAR_MA_AA.loc[SAR_MA_AA[\"GOAssertion\"].isnull(), \"GOAssertion\"] = \"AA\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(716735, 10) (573293, 10) (630944, 10)\n"
     ]
    }
   ],
   "source": [
    "total_SAR = pd.concat([SAR_MA_AA, Plasmodium_AA])\n",
    "total_SAR_function = total_SAR[total_SAR[\"Gene Ontology (molecular function)\"].notnull()]\n",
    "total_SAR_process = total_SAR[total_SAR[\"Gene Ontology (biological process)\"].notnull()]\n",
    "total_SAR_component = total_SAR[total_SAR[\"Gene Ontology (cellular component)\"].notnull()]\n",
    "\n",
    "print(total_SAR_function.shape, total_SAR_process.shape, total_SAR_component.shape)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "893661\n"
     ]
    }
   ],
   "source": [
    "total_SAR_records = create_seq_records(total_SAR)\n",
    "print(len(total_SAR_records))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "893661"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "SeqIO.write(total_SAR_records, \"total_SAR.fasta\", \"fasta\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [],
   "source": [
    "seq_records_SAR_function = create_seq_records(total_SAR_function)\n",
    "seq_records_SAR_process = create_seq_records(total_SAR_process)\n",
    "seq_records_SAR_component = create_seq_records(total_SAR_component)\n",
    "\n",
    "assert len(seq_records_SAR_function) == total_SAR_function.shape[0]\n",
    "assert len(seq_records_SAR_process) == total_SAR_process.shape[0]\n",
    "assert len(seq_records_SAR_component) == total_SAR_component.shape[0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "630944"
      ]
     },
     "execution_count": 19,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "SeqIO.write(seq_records_SAR_function, \"total_SAR_function_for_reduce.fasta\", \"fasta\")\n",
    "SeqIO.write(seq_records_SAR_process, \"total_SAR_process_for_reduce.fasta\", \"fasta\")\n",
    "SeqIO.write(seq_records_SAR_component, \"total_SAR_component_for_reduce.fasta\", \"fasta\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Usage: ./mmseq_cluster.sh <parent_directory> <function_file> <component_file> <process_file> <min_seq_id>\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "CompletedProcess(args=['./mmseq_cluster.sh', '-h'], returncode=1)"
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#run raw_data_from_uniprot/mmseq_cluster.sh\n",
    "\n",
    "subprocess.run([\"./mmseq_cluster.sh\", \"-h\"])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Creating databases...\n",
      "createdb /Users/harsh/Documents/PFP_PUBLISH/raw_data_from_uniprot/total_SAR_function_for_reduce.fasta function/function_redundant_db \n",
      "\n",
      "MMseqs Version:       \t14.7e284\n",
      "Database type         \t0\n",
      "Shuffle input database\ttrue\n",
      "Createdb mode         \t0\n",
      "Write lookup file     \t1\n",
      "Offset of numeric ids \t0\n",
      "Compressed            \t0\n",
      "Verbosity             \t3\n",
      "\n",
      "Converting sequences\n",
      "[=======================================================================\n",
      "Time for merging to function_redundant_db_h: 0h 0m 0s 118ms\n",
      "Time for merging to function_redundant_db: 0h 0m 0s 250ms\n",
      "Database type: Aminoacid\n",
      "Time for processing: 0h 0m 2s 130ms\n",
      "createdb /Users/harsh/Documents/PFP_PUBLISH/raw_data_from_uniprot/total_SAR_component_for_reduce.fasta component/component_redundant_db \n",
      "\n",
      "MMseqs Version:       \t14.7e284\n",
      "Database type         \t0\n",
      "Shuffle input database\ttrue\n",
      "Createdb mode         \t0\n",
      "Write lookup file     \t1\n",
      "Offset of numeric ids \t0\n",
      "Compressed            \t0\n",
      "Verbosity             \t3\n",
      "\n",
      "Converting sequences\n",
      "[===============================================================\n",
      "Time for merging to component_redundant_db_h: 0h 0m 0s 105ms\n",
      "Time for merging to component_redundant_db: 0h 0m 0s 220ms\n",
      "Database type: Aminoacid\n",
      "Time for processing: 0h 0m 1s 805ms\n",
      "createdb /Users/harsh/Documents/PFP_PUBLISH/raw_data_from_uniprot/total_SAR_process_for_reduce.fasta process/process_redundant_db \n",
      "\n",
      "MMseqs Version:       \t14.7e284\n",
      "Database type         \t0\n",
      "Shuffle input database\ttrue\n",
      "Createdb mode         \t0\n",
      "Write lookup file     \t1\n",
      "Offset of numeric ids \t0\n",
      "Compressed            \t0\n",
      "Verbosity             \t3\n",
      "\n",
      "Converting sequences\n",
      "[=========================================================\n",
      "Time for merging to process_redundant_db_h: 0h 0m 0s 101ms\n",
      "Time for merging to process_redundant_db: 0h 0m 0s 199ms\n",
      "Database type: Aminoacid\n",
      "Time for processing: 0h 0m 1s 617ms\n",
      "Clustering sequences...\n",
      "Create directory tmp\n",
      "cluster function/function_redundant_db function/function_clusters_redundant tmp --min-seq-id 0.9 -c 0.5 --cov-mode 1 \n",
      "\n",
      "MMseqs Version:                     \t14.7e284\n",
      "Substitution matrix                 \taa:blosum62.out,nucl:nucleotide.out\n",
      "Seed substitution matrix            \taa:VTML80.out,nucl:nucleotide.out\n",
      "Sensitivity                         \t4\n",
      "k-mer length                        \t0\n",
      "k-score                             \tseq:2147483647,prof:2147483647\n",
      "Alphabet size                       \taa:21,nucl:5\n",
      "Max sequence length                 \t65535\n",
      "Max results per query               \t20\n",
      "Split database                      \t0\n",
      "Split mode                          \t2\n",
      "Split memory limit                  \t0\n",
      "Coverage threshold                  \t0.5\n",
      "Coverage mode                       \t1\n",
      "Compositional bias                  \t1\n",
      "Compositional bias                  \t1\n",
      "Diagonal scoring                    \ttrue\n",
      "Exact k-mer matching                \t0\n",
      "Mask residues                       \t1\n",
      "Mask residues probability           \t0.9\n",
      "Mask lower case residues            \t0\n",
      "Minimum diagonal score              \t15\n",
      "Selected taxa                       \t\n",
      "Include identical seq. id.          \tfalse\n",
      "Spaced k-mers                       \t1\n",
      "Preload mode                        \t0\n",
      "Pseudo count a                      \tsubstitution:1.100,context:1.400\n",
      "Pseudo count b                      \tsubstitution:4.100,context:5.800\n",
      "Spaced k-mer pattern                \t\n",
      "Local temporary path                \t\n",
      "Threads                             \t10\n",
      "Compressed                          \t0\n",
      "Verbosity                           \t3\n",
      "Add backtrace                       \tfalse\n",
      "Alignment mode                      \t3\n",
      "Alignment mode                      \t0\n",
      "Allow wrapped scoring               \tfalse\n",
      "E-value threshold                   \t0.001\n",
      "Seq. id. threshold                  \t0.9\n",
      "Min alignment length                \t0\n",
      "Seq. id. mode                       \t0\n",
      "Alternative alignments              \t0\n",
      "Max reject                          \t2147483647\n",
      "Max accept                          \t2147483647\n",
      "Score bias                          \t0\n",
      "Realign hits                        \tfalse\n",
      "Realign score bias                  \t-0.2\n",
      "Realign max seqs                    \t2147483647\n",
      "Correlation score weight            \t0\n",
      "Gap open cost                       \taa:11,nucl:5\n",
      "Gap extension cost                  \taa:1,nucl:2\n",
      "Zdrop                               \t40\n",
      "Rescore mode                        \t0\n",
      "Remove hits by seq. id. and coverage\tfalse\n",
      "Sort results                        \t0\n",
      "Cluster mode                        \t0\n",
      "Max connected component depth       \t1000\n",
      "Similarity type                     \t2\n",
      "Single step clustering              \tfalse\n",
      "Cascaded clustering steps           \t3\n",
      "Cluster reassign                    \tfalse\n",
      "Remove temporary files              \tfalse\n",
      "Force restart with latest tmp       \tfalse\n",
      "MPI runner                          \t\n",
      "k-mers per sequence                 \t21\n",
      "Scale k-mers per sequence           \taa:0.000,nucl:0.200\n",
      "Adjust k-mer length                 \tfalse\n",
      "Shift hash                          \t67\n",
      "Include only extendable             \tfalse\n",
      "Skip repeating k-mers               \tfalse\n",
      "\n",
      "Set cluster sensitivity to -s 1.000000\n",
      "Set cluster mode GREEDY MEM\n",
      "Set cluster iterations to 1\n",
      "linclust function/function_redundant_db tmp/9696200901604251854/clu_redundancy tmp/9696200901604251854/linclust --cluster-mode 3 --max-iterations 1000 --similarity-type 2 --threads 10 --compressed 0 -v 3 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' -a 0 --alignment-mode 3 --alignment-output-mode 0 --wrapped-scoring 0 -e 0.001 --min-seq-id 0.9 --min-aln-len 0 --seq-id-mode 0 --alt-ali 0 -c 0.5 --cov-mode 1 --max-seq-len 65535 --comp-bias-corr 0 --comp-bias-corr-scale 1 --max-rejected 2147483647 --max-accept 2147483647 --add-self-matches 0 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --score-bias 0 --realign 0 --realign-score-bias -0.2 --realign-max-seqs 2147483647 --corr-score-weight 0 --gap-open aa:11,nucl:5 --gap-extend aa:1,nucl:2 --zdrop 40 --alph-size aa:13,nucl:5 --kmer-per-seq 21 --spaced-kmer-mode 1 --kmer-per-seq-scale aa:0.000,nucl:0.200 --adjust-kmer-len 0 --mask 0 --mask-prob 0.9 --mask-lower-case 0 -k 0 --hash-shift 67 --split-memory-limit 0 --include-only-extendable 0 --ignore-multi-kmer 0 --rescore-mode 0 --filter-hits 0 --sort-results 0 --remove-tmp-files 0 --force-reuse 0 \n",
      "\n",
      "kmermatcher function/function_redundant_db tmp/9696200901604251854/linclust/14271651232352195491/pref --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' --alph-size aa:13,nucl:5 --min-seq-id 0.9 --kmer-per-seq 21 --spaced-kmer-mode 1 --kmer-per-seq-scale aa:0.000,nucl:0.200 --adjust-kmer-len 0 --mask 0 --mask-prob 0.9 --mask-lower-case 0 --cov-mode 1 -k 0 -c 0.5 --max-seq-len 65535 --hash-shift 67 --split-memory-limit 0 --include-only-extendable 0 --ignore-multi-kmer 0 --threads 10 --compressed 0 -v 3 \n",
      "\n",
      "kmermatcher function/function_redundant_db tmp/9696200901604251854/linclust/14271651232352195491/pref --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' --alph-size aa:13,nucl:5 --min-seq-id 0.9 --kmer-per-seq 21 --spaced-kmer-mode 1 --kmer-per-seq-scale aa:0.000,nucl:0.200 --adjust-kmer-len 0 --mask 0 --mask-prob 0.9 --mask-lower-case 0 --cov-mode 1 -k 0 -c 0.5 --max-seq-len 65535 --hash-shift 67 --split-memory-limit 0 --include-only-extendable 0 --ignore-multi-kmer 0 --threads 10 --compressed 0 -v 3 \n",
      "\n",
      "Database size: 716735 type: Aminoacid\n",
      "Reduced amino acid alphabet: (A S T) (C) (D B N) (E Q Z) (F Y) (G) (H) (I V) (K R) (L J M) (P) (W) (X) \n",
      "\n",
      "Generate k-mers list for 1 split\n",
      "[=================================================================] 716.73K 1s 723ms\n",
      "Sort kmer 0h 0m 0s 353ms\n",
      "Sort by rep. sequence 0h 0m 0s 147ms\n",
      "Time for fill: 0h 0m 0s 183ms\n",
      "Time for merging to pref: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 2s 704ms\n",
      "rescorediagonal function/function_redundant_db function/function_redundant_db tmp/9696200901604251854/linclust/14271651232352195491/pref tmp/9696200901604251854/linclust/14271651232352195491/pref_rescore1 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' --rescore-mode 0 --wrapped-scoring 0 --filter-hits 0 -e 0.001 -c 0.5 -a 0 --cov-mode 1 --min-seq-id 0.9 --min-aln-len 0 --seq-id-mode 0 --add-self-matches 0 --sort-results 0 --db-load-mode 0 --threads 10 --compressed 0 -v 3 \n",
      "\n",
      "[=================================================================] 716.73K 0s 308ms\n",
      "Time for merging to pref_rescore1: 0h 0m 0s 255ms\n",
      "Time for processing: 0h 0m 0s 746ms\n",
      "clust function/function_redundant_db tmp/9696200901604251854/linclust/14271651232352195491/pref_rescore1 tmp/9696200901604251854/linclust/14271651232352195491/pre_clust --cluster-mode 3 --max-iterations 1000 --similarity-type 2 --threads 10 --compressed 0 -v 3 \n",
      "\n",
      "Clustering mode: Greedy Low Mem\n",
      "Total time: 0h 0m 0s 104ms\n",
      "\n",
      "Size of the sequence database: 716735\n",
      "Size of the alignment database: 716735\n",
      "Number of clusters: 413197\n",
      "\n",
      "Writing results 0h 0m 0s 109ms\n",
      "Time for merging to pre_clust: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 285ms\n",
      "createsubdb tmp/9696200901604251854/linclust/14271651232352195491/order_redundancy function/function_redundant_db tmp/9696200901604251854/linclust/14271651232352195491/input_step_redundancy -v 3 --subdb-mode 1 \n",
      "\n",
      "Time for merging to input_step_redundancy: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 141ms\n",
      "createsubdb tmp/9696200901604251854/linclust/14271651232352195491/order_redundancy tmp/9696200901604251854/linclust/14271651232352195491/pref tmp/9696200901604251854/linclust/14271651232352195491/pref_filter1 -v 3 --subdb-mode 1 \n",
      "\n",
      "Time for merging to pref_filter1: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 158ms\n",
      "filterdb tmp/9696200901604251854/linclust/14271651232352195491/pref_filter1 tmp/9696200901604251854/linclust/14271651232352195491/pref_filter2 --filter-file tmp/9696200901604251854/linclust/14271651232352195491/order_redundancy --threads 10 --compressed 0 -v 3 \n",
      "\n",
      "Filtering using file(s)\n",
      "[=================================================================] 413.20K 0s 140ms\n",
      "Time for merging to pref_filter2: 0h 0m 0s 166ms\n",
      "Time for processing: 0h 0m 0s 467ms\n",
      "rescorediagonal tmp/9696200901604251854/linclust/14271651232352195491/input_step_redundancy tmp/9696200901604251854/linclust/14271651232352195491/input_step_redundancy tmp/9696200901604251854/linclust/14271651232352195491/pref_filter2 tmp/9696200901604251854/linclust/14271651232352195491/pref_rescore2 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' --rescore-mode 1 --wrapped-scoring 0 --filter-hits 1 -e 0.001 -c 0.5 -a 0 --cov-mode 1 --min-seq-id 0.9 --min-aln-len 0 --seq-id-mode 0 --add-self-matches 0 --sort-results 0 --db-load-mode 0 --threads 10 --compressed 0 -v 3 \n",
      "\n",
      "[=================================================================] 413.20K 0s 198ms\n",
      "Time for merging to pref_rescore2: 0h 0m 0s 236ms\n",
      "Time for processing: 0h 0m 0s 572ms\n",
      "align tmp/9696200901604251854/linclust/14271651232352195491/input_step_redundancy tmp/9696200901604251854/linclust/14271651232352195491/input_step_redundancy tmp/9696200901604251854/linclust/14271651232352195491/pref_rescore2 tmp/9696200901604251854/linclust/14271651232352195491/aln --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' -a 0 --alignment-mode 3 --alignment-output-mode 0 --wrapped-scoring 0 -e 0.001 --min-seq-id 0.9 --min-aln-len 0 --seq-id-mode 0 --alt-ali 0 -c 0.5 --cov-mode 1 --max-seq-len 65535 --comp-bias-corr 0 --comp-bias-corr-scale 1 --max-rejected 2147483647 --max-accept 2147483647 --add-self-matches 0 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --score-bias 0 --realign 0 --realign-score-bias -0.2 --realign-max-seqs 2147483647 --corr-score-weight 0 --gap-open aa:11,nucl:5 --gap-extend aa:1,nucl:2 --zdrop 40 --threads 10 --compressed 0 -v 3 \n",
      "\n",
      "Compute score, coverage and sequence identity\n",
      "Query database size: 413197 type: Aminoacid\n",
      "Target database size: 413197 type: Aminoacid\n",
      "Calculation of alignments\n",
      "[=================================================================] 413.20K 26s 810ms\n",
      "Time for merging to aln: 0h 0m 0s 134ms\n",
      "856321 alignments calculated\n",
      "483273 sequence pairs passed the thresholds (0.564360 of overall calculated)\n",
      "1.169595 hits per query sequence\n",
      "Time for processing: 0h 0m 27s 65ms\n",
      "clust tmp/9696200901604251854/linclust/14271651232352195491/input_step_redundancy tmp/9696200901604251854/linclust/14271651232352195491/aln tmp/9696200901604251854/linclust/14271651232352195491/clust --cluster-mode 3 --max-iterations 1000 --similarity-type 2 --threads 10 --compressed 0 -v 3 \n",
      "\n",
      "Clustering mode: Greedy Low Mem\n",
      "Total time: 0h 0m 0s 94ms\n",
      "\n",
      "Size of the sequence database: 413197\n",
      "Size of the alignment database: 413197\n",
      "Number of clusters: 362510\n",
      "\n",
      "Writing results 0h 0m 0s 92ms\n",
      "Time for merging to clust: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 236ms\n",
      "mergeclusters function/function_redundant_db tmp/9696200901604251854/clu_redundancy tmp/9696200901604251854/linclust/14271651232352195491/pre_clust tmp/9696200901604251854/linclust/14271651232352195491/clust --threads 10 --compressed 0 -v 3 \n",
      "\n",
      "Clustering step 1\n",
      "[=================================================================] 413.20K 0s 44ms\n",
      "Clustering step 2\n",
      "[=================================================================] 362.51K 0s 96ms\n",
      "Write merged clustering\n",
      "[=================================================================] 716.73K 0s 163ms\n",
      "Time for merging to clu_redundancy: 0h 0m 0s 135ms\n",
      "Time for processing: 0h 0m 0s 432ms\n",
      "createsubdb tmp/9696200901604251854/clu_redundancy function/function_redundant_db tmp/9696200901604251854/input_step_redundancy -v 3 --subdb-mode 1 \n",
      "\n",
      "Time for merging to input_step_redundancy: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 137ms\n",
      "prefilter tmp/9696200901604251854/input_step_redundancy tmp/9696200901604251854/input_step_redundancy tmp/9696200901604251854/pref_step0 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' --seed-sub-mat 'aa:VTML80.out,nucl:nucleotide.out' -s 1 -k 0 --k-score seq:2147483647,prof:2147483647 --alph-size aa:21,nucl:5 --max-seq-len 65535 --max-seqs 20 --split 0 --split-mode 2 --split-memory-limit 0 -c 0.5 --cov-mode 1 --comp-bias-corr 0 --comp-bias-corr-scale 1 --diag-score 0 --exact-kmer-matching 0 --mask 1 --mask-prob 0.9 --mask-lower-case 0 --min-ungapped-score 0 --add-self-matches 0 --spaced-kmer-mode 1 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --threads 10 --compressed 0 -v 3 \n",
      "\n",
      "Query database size: 362510 type: Aminoacid\n",
      "Estimated memory consumption: 2G\n",
      "Target database size: 362510 type: Aminoacid\n",
      "Index table k-mer threshold: 154 at k-mer size 6 \n",
      "Index table: counting k-mers\n",
      "[=================================================================] 362.51K 2s 598ms\n",
      "Index table: Masked residues: 2925683\n",
      "Index table: fill\n",
      "[=================================================================] 362.51K 2s 550ms\n",
      "Index statistics\n",
      "Entries:          73974763\n",
      "DB size:          911 MB\n",
      "Avg k-mer size:   1.155856\n",
      "Top 10 k-mers\n",
      "    FAGTGK\t2414\n",
      "    AYQGKT\t2204\n",
      "    WWARVW\t1580\n",
      "    PTFYQY\t1368\n",
      "    HRLPLL\t1134\n",
      "    GPGGLL\t1040\n",
      "    LGGFVY\t1019\n",
      "    GQGGYT\t928\n",
      "    YGTSTY\t897\n",
      "    LYPGKT\t835\n",
      "Time for index table init: 0h 0m 5s 661ms\n",
      "Process prefiltering step 1 of 1\n",
      "\n",
      "k-mer similarity threshold: 154\n",
      "Starting prefiltering scores calculation (step 1 of 1)\n",
      "Query db start 1 to 362510\n",
      "Target db start 1 to 362510\n",
      "[=================================================================] 362.51K 7s 855ms\n",
      "\n",
      "2.038547 k-mers per position\n",
      "3064 DB matches per sequence\n",
      "0 overflows\n",
      "0 queries produce too many hits (truncated result)\n",
      "17 sequences passed prefiltering per query sequence\n",
      "20 median result list length\n",
      "0 sequences with 0 size result lists\n",
      "Time for merging to pref_step0: 0h 0m 0s 115ms\n",
      "Time for processing: 0h 0m 13s 994ms\n",
      "align tmp/9696200901604251854/input_step_redundancy tmp/9696200901604251854/input_step_redundancy tmp/9696200901604251854/pref_step0 tmp/9696200901604251854/aln_step0 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' -a 0 --alignment-mode 3 --alignment-output-mode 0 --wrapped-scoring 0 -e 0.001 --min-seq-id 0.9 --min-aln-len 0 --seq-id-mode 0 --alt-ali 0 -c 0.5 --cov-mode 1 --max-seq-len 65535 --comp-bias-corr 0 --comp-bias-corr-scale 1 --max-rejected 2147483647 --max-accept 2147483647 --add-self-matches 0 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --score-bias 0 --realign 0 --realign-score-bias -0.2 --realign-max-seqs 2147483647 --corr-score-weight 0 --gap-open aa:11,nucl:5 --gap-extend aa:1,nucl:2 --zdrop 40 --threads 10 --compressed 0 -v 3 \n",
      "\n",
      "Compute score, coverage and sequence identity\n",
      "Query database size: 362510 type: Aminoacid\n",
      "Target database size: 362510 type: Aminoacid\n",
      "Calculation of alignments\n",
      "[=================================================================] 362.51K 7m 30s 621ms\n",
      "Time for merging to aln_step0: 0h 0m 0s 322ms\n",
      "5765293 alignments calculated\n",
      "391673 sequence pairs passed the thresholds (0.067936 of overall calculated)\n",
      "1.080447 hits per query sequence\n",
      "Time for processing: 0h 7m 31s 126ms\n",
      "clust tmp/9696200901604251854/input_step_redundancy tmp/9696200901604251854/aln_step0 tmp/9696200901604251854/clu_step0 --cluster-mode 3 --max-iterations 1000 --similarity-type 2 --threads 10 --compressed 0 -v 3 \n",
      "\n",
      "Clustering mode: Greedy Low Mem\n",
      "Total time: 0h 0m 0s 276ms\n",
      "\n",
      "Size of the sequence database: 362510\n",
      "Size of the alignment database: 362510\n",
      "Number of clusters: 352585\n",
      "\n",
      "Writing results 0h 0m 0s 91ms\n",
      "Time for merging to clu_step0: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 622ms\n",
      "mergeclusters function/function_redundant_db function/function_clusters_redundant tmp/9696200901604251854/clu_redundancy tmp/9696200901604251854/clu_step0 --threads 10 --compressed 0 -v 3 \n",
      "\n",
      "Clustering step 1\n",
      "[=================================================================] 362.51K 0s 36ms\n",
      "Clustering step 2\n",
      "[=================================================================] 352.58K 0s 157ms\n",
      "Write merged clustering\n",
      "[=================================================================] 716.73K 0s 238ms\n",
      "Time for merging to function_clusters_redundant: 0h 0m 0s 295ms\n",
      "Time for processing: 0h 0m 0s 900ms\n",
      "cluster component/component_redundant_db component/component_clusters_redundant tmp --min-seq-id 0.9 -c 0.5 --cov-mode 1 \n",
      "\n",
      "MMseqs Version:                     \t14.7e284\n",
      "Substitution matrix                 \taa:blosum62.out,nucl:nucleotide.out\n",
      "Seed substitution matrix            \taa:VTML80.out,nucl:nucleotide.out\n",
      "Sensitivity                         \t4\n",
      "k-mer length                        \t0\n",
      "k-score                             \tseq:2147483647,prof:2147483647\n",
      "Alphabet size                       \taa:21,nucl:5\n",
      "Max sequence length                 \t65535\n",
      "Max results per query               \t20\n",
      "Split database                      \t0\n",
      "Split mode                          \t2\n",
      "Split memory limit                  \t0\n",
      "Coverage threshold                  \t0.5\n",
      "Coverage mode                       \t1\n",
      "Compositional bias                  \t1\n",
      "Compositional bias                  \t1\n",
      "Diagonal scoring                    \ttrue\n",
      "Exact k-mer matching                \t0\n",
      "Mask residues                       \t1\n",
      "Mask residues probability           \t0.9\n",
      "Mask lower case residues            \t0\n",
      "Minimum diagonal score              \t15\n",
      "Selected taxa                       \t\n",
      "Include identical seq. id.          \tfalse\n",
      "Spaced k-mers                       \t1\n",
      "Preload mode                        \t0\n",
      "Pseudo count a                      \tsubstitution:1.100,context:1.400\n",
      "Pseudo count b                      \tsubstitution:4.100,context:5.800\n",
      "Spaced k-mer pattern                \t\n",
      "Local temporary path                \t\n",
      "Threads                             \t10\n",
      "Compressed                          \t0\n",
      "Verbosity                           \t3\n",
      "Add backtrace                       \tfalse\n",
      "Alignment mode                      \t3\n",
      "Alignment mode                      \t0\n",
      "Allow wrapped scoring               \tfalse\n",
      "E-value threshold                   \t0.001\n",
      "Seq. id. threshold                  \t0.9\n",
      "Min alignment length                \t0\n",
      "Seq. id. mode                       \t0\n",
      "Alternative alignments              \t0\n",
      "Max reject                          \t2147483647\n",
      "Max accept                          \t2147483647\n",
      "Score bias                          \t0\n",
      "Realign hits                        \tfalse\n",
      "Realign score bias                  \t-0.2\n",
      "Realign max seqs                    \t2147483647\n",
      "Correlation score weight            \t0\n",
      "Gap open cost                       \taa:11,nucl:5\n",
      "Gap extension cost                  \taa:1,nucl:2\n",
      "Zdrop                               \t40\n",
      "Rescore mode                        \t0\n",
      "Remove hits by seq. id. and coverage\tfalse\n",
      "Sort results                        \t0\n",
      "Cluster mode                        \t0\n",
      "Max connected component depth       \t1000\n",
      "Similarity type                     \t2\n",
      "Single step clustering              \tfalse\n",
      "Cascaded clustering steps           \t3\n",
      "Cluster reassign                    \tfalse\n",
      "Remove temporary files              \tfalse\n",
      "Force restart with latest tmp       \tfalse\n",
      "MPI runner                          \t\n",
      "k-mers per sequence                 \t21\n",
      "Scale k-mers per sequence           \taa:0.000,nucl:0.200\n",
      "Adjust k-mer length                 \tfalse\n",
      "Shift hash                          \t67\n",
      "Include only extendable             \tfalse\n",
      "Skip repeating k-mers               \tfalse\n",
      "\n",
      "Set cluster sensitivity to -s 1.000000\n",
      "Set cluster mode GREEDY MEM\n",
      "Set cluster iterations to 1\n",
      "linclust component/component_redundant_db tmp/12400988725396477983/clu_redundancy tmp/12400988725396477983/linclust --cluster-mode 3 --max-iterations 1000 --similarity-type 2 --threads 10 --compressed 0 -v 3 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' -a 0 --alignment-mode 3 --alignment-output-mode 0 --wrapped-scoring 0 -e 0.001 --min-seq-id 0.9 --min-aln-len 0 --seq-id-mode 0 --alt-ali 0 -c 0.5 --cov-mode 1 --max-seq-len 65535 --comp-bias-corr 0 --comp-bias-corr-scale 1 --max-rejected 2147483647 --max-accept 2147483647 --add-self-matches 0 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --score-bias 0 --realign 0 --realign-score-bias -0.2 --realign-max-seqs 2147483647 --corr-score-weight 0 --gap-open aa:11,nucl:5 --gap-extend aa:1,nucl:2 --zdrop 40 --alph-size aa:13,nucl:5 --kmer-per-seq 21 --spaced-kmer-mode 1 --kmer-per-seq-scale aa:0.000,nucl:0.200 --adjust-kmer-len 0 --mask 0 --mask-prob 0.9 --mask-lower-case 0 -k 0 --hash-shift 67 --split-memory-limit 0 --include-only-extendable 0 --ignore-multi-kmer 0 --rescore-mode 0 --filter-hits 0 --sort-results 0 --remove-tmp-files 0 --force-reuse 0 \n",
      "\n",
      "kmermatcher component/component_redundant_db tmp/12400988725396477983/linclust/3242237169228371394/pref --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' --alph-size aa:13,nucl:5 --min-seq-id 0.9 --kmer-per-seq 21 --spaced-kmer-mode 1 --kmer-per-seq-scale aa:0.000,nucl:0.200 --adjust-kmer-len 0 --mask 0 --mask-prob 0.9 --mask-lower-case 0 --cov-mode 1 -k 0 -c 0.5 --max-seq-len 65535 --hash-shift 67 --split-memory-limit 0 --include-only-extendable 0 --ignore-multi-kmer 0 --threads 10 --compressed 0 -v 3 \n",
      "\n",
      "kmermatcher component/component_redundant_db tmp/12400988725396477983/linclust/3242237169228371394/pref --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' --alph-size aa:13,nucl:5 --min-seq-id 0.9 --kmer-per-seq 21 --spaced-kmer-mode 1 --kmer-per-seq-scale aa:0.000,nucl:0.200 --adjust-kmer-len 0 --mask 0 --mask-prob 0.9 --mask-lower-case 0 --cov-mode 1 -k 0 -c 0.5 --max-seq-len 65535 --hash-shift 67 --split-memory-limit 0 --include-only-extendable 0 --ignore-multi-kmer 0 --threads 10 --compressed 0 -v 3 \n",
      "\n",
      "Database size: 630944 type: Aminoacid\n",
      "Reduced amino acid alphabet: (A S T) (C) (D B N) (E Q Z) (F Y) (G) (H) (I V) (K R) (L J M) (P) (W) (X) \n",
      "\n",
      "Generate k-mers list for 1 split\n",
      "[=================================================================] 630.94K 1s 113ms\n",
      "Sort kmer 0h 0m 0s 391ms\n",
      "Sort by rep. sequence 0h 0m 0s 136ms\n",
      "Time for fill: 0h 0m 0s 154ms\n",
      "Time for merging to pref: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 2s 66ms\n",
      "rescorediagonal component/component_redundant_db component/component_redundant_db tmp/12400988725396477983/linclust/3242237169228371394/pref tmp/12400988725396477983/linclust/3242237169228371394/pref_rescore1 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' --rescore-mode 0 --wrapped-scoring 0 --filter-hits 0 -e 0.001 -c 0.5 -a 0 --cov-mode 1 --min-seq-id 0.9 --min-aln-len 0 --seq-id-mode 0 --add-self-matches 0 --sort-results 0 --db-load-mode 0 --threads 10 --compressed 0 -v 3 \n",
      "\n",
      "[=================================================================] 630.94K 0s 241ms\n",
      "Time for merging to pref_rescore1: 0h 0m 0s 409ms\n",
      "Time for processing: 0h 0m 0s 813ms\n",
      "clust component/component_redundant_db tmp/12400988725396477983/linclust/3242237169228371394/pref_rescore1 tmp/12400988725396477983/linclust/3242237169228371394/pre_clust --cluster-mode 3 --max-iterations 1000 --similarity-type 2 --threads 10 --compressed 0 -v 3 \n",
      "\n",
      "Clustering mode: Greedy Low Mem\n",
      "Total time: 0h 0m 0s 96ms\n",
      "\n",
      "Size of the sequence database: 630944\n",
      "Size of the alignment database: 630944\n",
      "Number of clusters: 358277\n",
      "\n",
      "Writing results 0h 0m 0s 94ms\n",
      "Time for merging to pre_clust: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 292ms\n",
      "createsubdb tmp/12400988725396477983/linclust/3242237169228371394/order_redundancy component/component_redundant_db tmp/12400988725396477983/linclust/3242237169228371394/input_step_redundancy -v 3 --subdb-mode 1 \n",
      "\n",
      "Time for merging to input_step_redundancy: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 126ms\n",
      "createsubdb tmp/12400988725396477983/linclust/3242237169228371394/order_redundancy tmp/12400988725396477983/linclust/3242237169228371394/pref tmp/12400988725396477983/linclust/3242237169228371394/pref_filter1 -v 3 --subdb-mode 1 \n",
      "\n",
      "Time for merging to pref_filter1: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 139ms\n",
      "filterdb tmp/12400988725396477983/linclust/3242237169228371394/pref_filter1 tmp/12400988725396477983/linclust/3242237169228371394/pref_filter2 --filter-file tmp/12400988725396477983/linclust/3242237169228371394/order_redundancy --threads 10 --compressed 0 -v 3 \n",
      "\n",
      "Filtering using file(s)\n",
      "[=================================================================] 358.28K 0s 115ms\n",
      "Time for merging to pref_filter2: 0h 0m 0s 148ms\n",
      "Time for processing: 0h 0m 0s 407ms\n",
      "rescorediagonal tmp/12400988725396477983/linclust/3242237169228371394/input_step_redundancy tmp/12400988725396477983/linclust/3242237169228371394/input_step_redundancy tmp/12400988725396477983/linclust/3242237169228371394/pref_filter2 tmp/12400988725396477983/linclust/3242237169228371394/pref_rescore2 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' --rescore-mode 1 --wrapped-scoring 0 --filter-hits 1 -e 0.001 -c 0.5 -a 0 --cov-mode 1 --min-seq-id 0.9 --min-aln-len 0 --seq-id-mode 0 --add-self-matches 0 --sort-results 0 --db-load-mode 0 --threads 10 --compressed 0 -v 3 \n",
      "\n",
      "[=================================================================] 358.28K 0s 155ms\n",
      "Time for merging to pref_rescore2: 0h 0m 0s 153ms\n",
      "Time for processing: 0h 0m 0s 457ms\n",
      "align tmp/12400988725396477983/linclust/3242237169228371394/input_step_redundancy tmp/12400988725396477983/linclust/3242237169228371394/input_step_redundancy tmp/12400988725396477983/linclust/3242237169228371394/pref_rescore2 tmp/12400988725396477983/linclust/3242237169228371394/aln --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' -a 0 --alignment-mode 3 --alignment-output-mode 0 --wrapped-scoring 0 -e 0.001 --min-seq-id 0.9 --min-aln-len 0 --seq-id-mode 0 --alt-ali 0 -c 0.5 --cov-mode 1 --max-seq-len 65535 --comp-bias-corr 0 --comp-bias-corr-scale 1 --max-rejected 2147483647 --max-accept 2147483647 --add-self-matches 0 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --score-bias 0 --realign 0 --realign-score-bias -0.2 --realign-max-seqs 2147483647 --corr-score-weight 0 --gap-open aa:11,nucl:5 --gap-extend aa:1,nucl:2 --zdrop 40 --threads 10 --compressed 0 -v 3 \n",
      "\n",
      "Compute score, coverage and sequence identity\n",
      "Query database size: 358277 type: Aminoacid\n",
      "Target database size: 358277 type: Aminoacid\n",
      "Calculation of alignments\n",
      "[=================================================================] 358.28K 20s 915ms\n",
      "Time for merging to aln: 0h 0m 0s 116ms\n",
      "728286 alignments calculated\n",
      "418670 sequence pairs passed the thresholds (0.574870 of overall calculated)\n",
      "1.168565 hits per query sequence\n",
      "Time for processing: 0h 0m 21s 135ms\n",
      "clust tmp/12400988725396477983/linclust/3242237169228371394/input_step_redundancy tmp/12400988725396477983/linclust/3242237169228371394/aln tmp/12400988725396477983/linclust/3242237169228371394/clust --cluster-mode 3 --max-iterations 1000 --similarity-type 2 --threads 10 --compressed 0 -v 3 \n",
      "\n",
      "Clustering mode: Greedy Low Mem\n",
      "Total time: 0h 0m 0s 277ms\n",
      "\n",
      "Size of the sequence database: 358277\n",
      "Size of the alignment database: 358277\n",
      "Number of clusters: 315024\n",
      "\n",
      "Writing results 0h 0m 0s 78ms\n",
      "Time for merging to clust: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 624ms\n",
      "mergeclusters component/component_redundant_db tmp/12400988725396477983/clu_redundancy tmp/12400988725396477983/linclust/3242237169228371394/pre_clust tmp/12400988725396477983/linclust/3242237169228371394/clust --threads 10 --compressed 0 -v 3 \n",
      "\n",
      "Clustering step 1\n",
      "[=================================================================] 358.28K 0s 29ms\n",
      "Clustering step 2\n",
      "[=================================================================] 315.02K 0s 140ms\n",
      "Write merged clustering\n",
      "[=================================================================] 630.94K 0s 202ms\n",
      "Time for merging to clu_redundancy: 0h 0m 0s 158ms\n",
      "Time for processing: 0h 0m 0s 535ms\n",
      "createsubdb tmp/12400988725396477983/clu_redundancy component/component_redundant_db tmp/12400988725396477983/input_step_redundancy -v 3 --subdb-mode 1 \n",
      "\n",
      "Time for merging to input_step_redundancy: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 117ms\n",
      "prefilter tmp/12400988725396477983/input_step_redundancy tmp/12400988725396477983/input_step_redundancy tmp/12400988725396477983/pref_step0 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' --seed-sub-mat 'aa:VTML80.out,nucl:nucleotide.out' -s 1 -k 0 --k-score seq:2147483647,prof:2147483647 --alph-size aa:21,nucl:5 --max-seq-len 65535 --max-seqs 20 --split 0 --split-mode 2 --split-memory-limit 0 -c 0.5 --cov-mode 1 --comp-bias-corr 0 --comp-bias-corr-scale 1 --diag-score 0 --exact-kmer-matching 0 --mask 1 --mask-prob 0.9 --mask-lower-case 0 --min-ungapped-score 0 --add-self-matches 0 --spaced-kmer-mode 1 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --threads 10 --compressed 0 -v 3 \n",
      "\n",
      "Query database size: 315024 type: Aminoacid\n",
      "Estimated memory consumption: 1G\n",
      "Target database size: 315024 type: Aminoacid\n",
      "Index table k-mer threshold: 154 at k-mer size 6 \n",
      "Index table: counting k-mers\n",
      "[=================================================================] 315.02K 2s 396ms\n",
      "Index table: Masked residues: 2743359\n",
      "Index table: fill\n",
      "[=================================================================] 315.02K 1s 327ms\n",
      "Index statistics\n",
      "Entries:          60688137\n",
      "DB size:          835 MB\n",
      "Avg k-mer size:   0.948252\n",
      "Top 10 k-mers\n",
      "    LRDWNR\t2191\n",
      "    AYQGKT\t1872\n",
      "    WWARVW\t1581\n",
      "    PTFYQY\t1368\n",
      "    GPGGLL\t834\n",
      "    GQGGHT\t823\n",
      "    GQGGYT\t752\n",
      "    YGTSTY\t725\n",
      "    PQLWEW\t669\n",
      "    NYKRWW\t662\n",
      "Time for index table init: 0h 0m 4s 123ms\n",
      "Process prefiltering step 1 of 1\n",
      "\n",
      "k-mer similarity threshold: 154\n",
      "Starting prefiltering scores calculation (step 1 of 1)\n",
      "Query db start 1 to 315024\n",
      "Target db start 1 to 315024\n",
      "[=================================================================] 315.02K 5s 627ms\n",
      "\n",
      "2.010304 k-mers per position\n",
      "2446 DB matches per sequence\n",
      "0 overflows\n",
      "0 queries produce too many hits (truncated result)\n",
      "16 sequences passed prefiltering per query sequence\n",
      "20 median result list length\n",
      "0 sequences with 0 size result lists\n",
      "Time for merging to pref_step0: 0h 0m 0s 102ms\n",
      "Time for processing: 0h 0m 10s 209ms\n",
      "align tmp/12400988725396477983/input_step_redundancy tmp/12400988725396477983/input_step_redundancy tmp/12400988725396477983/pref_step0 tmp/12400988725396477983/aln_step0 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' -a 0 --alignment-mode 3 --alignment-output-mode 0 --wrapped-scoring 0 -e 0.001 --min-seq-id 0.9 --min-aln-len 0 --seq-id-mode 0 --alt-ali 0 -c 0.5 --cov-mode 1 --max-seq-len 65535 --comp-bias-corr 0 --comp-bias-corr-scale 1 --max-rejected 2147483647 --max-accept 2147483647 --add-self-matches 0 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --score-bias 0 --realign 0 --realign-score-bias -0.2 --realign-max-seqs 2147483647 --corr-score-weight 0 --gap-open aa:11,nucl:5 --gap-extend aa:1,nucl:2 --zdrop 40 --threads 10 --compressed 0 -v 3 \n",
      "\n",
      "Compute score, coverage and sequence identity\n",
      "Query database size: 315024 type: Aminoacid\n",
      "Target database size: 315024 type: Aminoacid\n",
      "Calculation of alignments\n",
      "[=================================================================] 315.02K 6m 0s 752ms\n",
      "Time for merging to aln_step0: 0h 0m 0s 102ms\n",
      "4800201 alignments calculated\n",
      "341848 sequence pairs passed the thresholds (0.071215 of overall calculated)\n",
      "1.085149 hits per query sequence\n",
      "Time for processing: 0h 6m 1s 12ms\n",
      "clust tmp/12400988725396477983/input_step_redundancy tmp/12400988725396477983/aln_step0 tmp/12400988725396477983/clu_step0 --cluster-mode 3 --max-iterations 1000 --similarity-type 2 --threads 10 --compressed 0 -v 3 \n",
      "\n",
      "Clustering mode: Greedy Low Mem\n",
      "Total time: 0h 0m 0s 108ms\n",
      "\n",
      "Size of the sequence database: 315024\n",
      "Size of the alignment database: 315024\n",
      "Number of clusters: 306559\n",
      "\n",
      "Writing results 0h 0m 0s 75ms\n",
      "Time for merging to clu_step0: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 324ms\n",
      "mergeclusters component/component_redundant_db component/component_clusters_redundant tmp/12400988725396477983/clu_redundancy tmp/12400988725396477983/clu_step0 --threads 10 --compressed 0 -v 3 \n",
      "\n",
      "Clustering step 1\n",
      "[=================================================================] 315.02K 0s 30ms\n",
      "Clustering step 2\n",
      "[=================================================================] 306.56K 0s 73ms\n",
      "Write merged clustering\n",
      "[=================================================================] 630.94K 0s 128ms\n",
      "Time for merging to component_clusters_redundant: 0h 0m 0s 291ms\n",
      "Time for processing: 0h 0m 0s 536ms\n",
      "cluster process/process_redundant_db process/process_clusters_redundant tmp --min-seq-id 0.9 -c 0.5 --cov-mode 1 \n",
      "\n",
      "MMseqs Version:                     \t14.7e284\n",
      "Substitution matrix                 \taa:blosum62.out,nucl:nucleotide.out\n",
      "Seed substitution matrix            \taa:VTML80.out,nucl:nucleotide.out\n",
      "Sensitivity                         \t4\n",
      "k-mer length                        \t0\n",
      "k-score                             \tseq:2147483647,prof:2147483647\n",
      "Alphabet size                       \taa:21,nucl:5\n",
      "Max sequence length                 \t65535\n",
      "Max results per query               \t20\n",
      "Split database                      \t0\n",
      "Split mode                          \t2\n",
      "Split memory limit                  \t0\n",
      "Coverage threshold                  \t0.5\n",
      "Coverage mode                       \t1\n",
      "Compositional bias                  \t1\n",
      "Compositional bias                  \t1\n",
      "Diagonal scoring                    \ttrue\n",
      "Exact k-mer matching                \t0\n",
      "Mask residues                       \t1\n",
      "Mask residues probability           \t0.9\n",
      "Mask lower case residues            \t0\n",
      "Minimum diagonal score              \t15\n",
      "Selected taxa                       \t\n",
      "Include identical seq. id.          \tfalse\n",
      "Spaced k-mers                       \t1\n",
      "Preload mode                        \t0\n",
      "Pseudo count a                      \tsubstitution:1.100,context:1.400\n",
      "Pseudo count b                      \tsubstitution:4.100,context:5.800\n",
      "Spaced k-mer pattern                \t\n",
      "Local temporary path                \t\n",
      "Threads                             \t10\n",
      "Compressed                          \t0\n",
      "Verbosity                           \t3\n",
      "Add backtrace                       \tfalse\n",
      "Alignment mode                      \t3\n",
      "Alignment mode                      \t0\n",
      "Allow wrapped scoring               \tfalse\n",
      "E-value threshold                   \t0.001\n",
      "Seq. id. threshold                  \t0.9\n",
      "Min alignment length                \t0\n",
      "Seq. id. mode                       \t0\n",
      "Alternative alignments              \t0\n",
      "Max reject                          \t2147483647\n",
      "Max accept                          \t2147483647\n",
      "Score bias                          \t0\n",
      "Realign hits                        \tfalse\n",
      "Realign score bias                  \t-0.2\n",
      "Realign max seqs                    \t2147483647\n",
      "Correlation score weight            \t0\n",
      "Gap open cost                       \taa:11,nucl:5\n",
      "Gap extension cost                  \taa:1,nucl:2\n",
      "Zdrop                               \t40\n",
      "Rescore mode                        \t0\n",
      "Remove hits by seq. id. and coverage\tfalse\n",
      "Sort results                        \t0\n",
      "Cluster mode                        \t0\n",
      "Max connected component depth       \t1000\n",
      "Similarity type                     \t2\n",
      "Single step clustering              \tfalse\n",
      "Cascaded clustering steps           \t3\n",
      "Cluster reassign                    \tfalse\n",
      "Remove temporary files              \tfalse\n",
      "Force restart with latest tmp       \tfalse\n",
      "MPI runner                          \t\n",
      "k-mers per sequence                 \t21\n",
      "Scale k-mers per sequence           \taa:0.000,nucl:0.200\n",
      "Adjust k-mer length                 \tfalse\n",
      "Shift hash                          \t67\n",
      "Include only extendable             \tfalse\n",
      "Skip repeating k-mers               \tfalse\n",
      "\n",
      "Set cluster sensitivity to -s 1.000000\n",
      "Set cluster mode GREEDY MEM\n",
      "Set cluster iterations to 1\n",
      "linclust process/process_redundant_db tmp/8656781693095139340/clu_redundancy tmp/8656781693095139340/linclust --cluster-mode 3 --max-iterations 1000 --similarity-type 2 --threads 10 --compressed 0 -v 3 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' -a 0 --alignment-mode 3 --alignment-output-mode 0 --wrapped-scoring 0 -e 0.001 --min-seq-id 0.9 --min-aln-len 0 --seq-id-mode 0 --alt-ali 0 -c 0.5 --cov-mode 1 --max-seq-len 65535 --comp-bias-corr 0 --comp-bias-corr-scale 1 --max-rejected 2147483647 --max-accept 2147483647 --add-self-matches 0 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --score-bias 0 --realign 0 --realign-score-bias -0.2 --realign-max-seqs 2147483647 --corr-score-weight 0 --gap-open aa:11,nucl:5 --gap-extend aa:1,nucl:2 --zdrop 40 --alph-size aa:13,nucl:5 --kmer-per-seq 21 --spaced-kmer-mode 1 --kmer-per-seq-scale aa:0.000,nucl:0.200 --adjust-kmer-len 0 --mask 0 --mask-prob 0.9 --mask-lower-case 0 -k 0 --hash-shift 67 --split-memory-limit 0 --include-only-extendable 0 --ignore-multi-kmer 0 --rescore-mode 0 --filter-hits 0 --sort-results 0 --remove-tmp-files 0 --force-reuse 0 \n",
      "\n",
      "kmermatcher process/process_redundant_db tmp/8656781693095139340/linclust/9215862896326619405/pref --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' --alph-size aa:13,nucl:5 --min-seq-id 0.9 --kmer-per-seq 21 --spaced-kmer-mode 1 --kmer-per-seq-scale aa:0.000,nucl:0.200 --adjust-kmer-len 0 --mask 0 --mask-prob 0.9 --mask-lower-case 0 --cov-mode 1 -k 0 -c 0.5 --max-seq-len 65535 --hash-shift 67 --split-memory-limit 0 --include-only-extendable 0 --ignore-multi-kmer 0 --threads 10 --compressed 0 -v 3 \n",
      "\n",
      "kmermatcher process/process_redundant_db tmp/8656781693095139340/linclust/9215862896326619405/pref --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' --alph-size aa:13,nucl:5 --min-seq-id 0.9 --kmer-per-seq 21 --spaced-kmer-mode 1 --kmer-per-seq-scale aa:0.000,nucl:0.200 --adjust-kmer-len 0 --mask 0 --mask-prob 0.9 --mask-lower-case 0 --cov-mode 1 -k 0 -c 0.5 --max-seq-len 65535 --hash-shift 67 --split-memory-limit 0 --include-only-extendable 0 --ignore-multi-kmer 0 --threads 10 --compressed 0 -v 3 \n",
      "\n",
      "Database size: 573293 type: Aminoacid\n",
      "Reduced amino acid alphabet: (A S T) (C) (D B N) (E Q Z) (F Y) (G) (H) (I V) (K R) (L J M) (P) (W) (X) \n",
      "\n",
      "Generate k-mers list for 1 split\n",
      "[=================================================================] 573.29K 1s 100ms\n",
      "Sort kmer 0h 0m 0s 251ms\n",
      "Sort by rep. sequence 0h 0m 0s 120ms\n",
      "Time for fill: 0h 0m 0s 140ms\n",
      "Time for merging to pref: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 1s 886ms\n",
      "rescorediagonal process/process_redundant_db process/process_redundant_db tmp/8656781693095139340/linclust/9215862896326619405/pref tmp/8656781693095139340/linclust/9215862896326619405/pref_rescore1 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' --rescore-mode 0 --wrapped-scoring 0 --filter-hits 0 -e 0.001 -c 0.5 -a 0 --cov-mode 1 --min-seq-id 0.9 --min-aln-len 0 --seq-id-mode 0 --add-self-matches 0 --sort-results 0 --db-load-mode 0 --threads 10 --compressed 0 -v 3 \n",
      "\n",
      "[=================================================================] 573.29K 0s 260ms\n",
      "Time for merging to pref_rescore1: 0h 0m 0s 477ms\n",
      "Time for processing: 0h 0m 0s 921ms\n",
      "clust process/process_redundant_db tmp/8656781693095139340/linclust/9215862896326619405/pref_rescore1 tmp/8656781693095139340/linclust/9215862896326619405/pre_clust --cluster-mode 3 --max-iterations 1000 --similarity-type 2 --threads 10 --compressed 0 -v 3 \n",
      "\n",
      "Clustering mode: Greedy Low Mem\n",
      "Total time: 0h 0m 0s 117ms\n",
      "\n",
      "Size of the sequence database: 573293\n",
      "Size of the alignment database: 573293\n",
      "Number of clusters: 326620\n",
      "\n",
      "Writing results 0h 0m 0s 84ms\n",
      "Time for merging to pre_clust: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 426ms\n",
      "createsubdb tmp/8656781693095139340/linclust/9215862896326619405/order_redundancy process/process_redundant_db tmp/8656781693095139340/linclust/9215862896326619405/input_step_redundancy -v 3 --subdb-mode 1 \n",
      "\n",
      "Time for merging to input_step_redundancy: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 118ms\n",
      "createsubdb tmp/8656781693095139340/linclust/9215862896326619405/order_redundancy tmp/8656781693095139340/linclust/9215862896326619405/pref tmp/8656781693095139340/linclust/9215862896326619405/pref_filter1 -v 3 --subdb-mode 1 \n",
      "\n",
      "Time for merging to pref_filter1: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 123ms\n",
      "filterdb tmp/8656781693095139340/linclust/9215862896326619405/pref_filter1 tmp/8656781693095139340/linclust/9215862896326619405/pref_filter2 --filter-file tmp/8656781693095139340/linclust/9215862896326619405/order_redundancy --threads 10 --compressed 0 -v 3 \n",
      "\n",
      "Filtering using file(s)\n",
      "[=================================================================] 326.62K 0s 105ms\n",
      "Time for merging to pref_filter2: 0h 0m 0s 172ms\n",
      "Time for processing: 0h 0m 0s 421ms\n",
      "rescorediagonal tmp/8656781693095139340/linclust/9215862896326619405/input_step_redundancy tmp/8656781693095139340/linclust/9215862896326619405/input_step_redundancy tmp/8656781693095139340/linclust/9215862896326619405/pref_filter2 tmp/8656781693095139340/linclust/9215862896326619405/pref_rescore2 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' --rescore-mode 1 --wrapped-scoring 0 --filter-hits 1 -e 0.001 -c 0.5 -a 0 --cov-mode 1 --min-seq-id 0.9 --min-aln-len 0 --seq-id-mode 0 --add-self-matches 0 --sort-results 0 --db-load-mode 0 --threads 10 --compressed 0 -v 3 \n",
      "\n",
      "[=================================================================] 326.62K 0s 142ms\n",
      "Time for merging to pref_rescore2: 0h 0m 0s 152ms\n",
      "Time for processing: 0h 0m 0s 447ms\n",
      "align tmp/8656781693095139340/linclust/9215862896326619405/input_step_redundancy tmp/8656781693095139340/linclust/9215862896326619405/input_step_redundancy tmp/8656781693095139340/linclust/9215862896326619405/pref_rescore2 tmp/8656781693095139340/linclust/9215862896326619405/aln --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' -a 0 --alignment-mode 3 --alignment-output-mode 0 --wrapped-scoring 0 -e 0.001 --min-seq-id 0.9 --min-aln-len 0 --seq-id-mode 0 --alt-ali 0 -c 0.5 --cov-mode 1 --max-seq-len 65535 --comp-bias-corr 0 --comp-bias-corr-scale 1 --max-rejected 2147483647 --max-accept 2147483647 --add-self-matches 0 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --score-bias 0 --realign 0 --realign-score-bias -0.2 --realign-max-seqs 2147483647 --corr-score-weight 0 --gap-open aa:11,nucl:5 --gap-extend aa:1,nucl:2 --zdrop 40 --threads 10 --compressed 0 -v 3 \n",
      "\n",
      "Compute score, coverage and sequence identity\n",
      "Query database size: 326620 type: Aminoacid\n",
      "Target database size: 326620 type: Aminoacid\n",
      "Calculation of alignments\n",
      "[=================================================================] 326.62K 20s 268ms\n",
      "Time for merging to aln: 0h 0m 0s 99ms\n",
      "671644 alignments calculated\n",
      "380954 sequence pairs passed the thresholds (0.567196 of overall calculated)\n",
      "1.166352 hits per query sequence\n",
      "Time for processing: 0h 0m 20s 486ms\n",
      "clust tmp/8656781693095139340/linclust/9215862896326619405/input_step_redundancy tmp/8656781693095139340/linclust/9215862896326619405/aln tmp/8656781693095139340/linclust/9215862896326619405/clust --cluster-mode 3 --max-iterations 1000 --similarity-type 2 --threads 10 --compressed 0 -v 3 \n",
      "\n",
      "Clustering mode: Greedy Low Mem\n",
      "Total time: 0h 0m 0s 30ms\n",
      "\n",
      "Size of the sequence database: 326620\n",
      "Size of the alignment database: 326620\n",
      "Number of clusters: 287208\n",
      "\n",
      "Writing results 0h 0m 0s 92ms\n",
      "Time for merging to clust: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 164ms\n",
      "mergeclusters process/process_redundant_db tmp/8656781693095139340/clu_redundancy tmp/8656781693095139340/linclust/9215862896326619405/pre_clust tmp/8656781693095139340/linclust/9215862896326619405/clust --threads 10 --compressed 0 -v 3 \n",
      "\n",
      "Clustering step 1\n",
      "[=================================================================] 326.62K 0s 35ms\n",
      "Clustering step 2\n",
      "[=================================================================] 287.21K 0s 71ms\n",
      "Write merged clustering\n",
      "[=================================================================] 573.29K 0s 124ms\n",
      "Time for merging to clu_redundancy: 0h 0m 0s 162ms\n",
      "Time for processing: 0h 0m 0s 408ms\n",
      "createsubdb tmp/8656781693095139340/clu_redundancy process/process_redundant_db tmp/8656781693095139340/input_step_redundancy -v 3 --subdb-mode 1 \n",
      "\n",
      "Time for merging to input_step_redundancy: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 109ms\n",
      "prefilter tmp/8656781693095139340/input_step_redundancy tmp/8656781693095139340/input_step_redundancy tmp/8656781693095139340/pref_step0 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' --seed-sub-mat 'aa:VTML80.out,nucl:nucleotide.out' -s 1 -k 0 --k-score seq:2147483647,prof:2147483647 --alph-size aa:21,nucl:5 --max-seq-len 65535 --max-seqs 20 --split 0 --split-mode 2 --split-memory-limit 0 -c 0.5 --cov-mode 1 --comp-bias-corr 0 --comp-bias-corr-scale 1 --diag-score 0 --exact-kmer-matching 0 --mask 1 --mask-prob 0.9 --mask-lower-case 0 --min-ungapped-score 0 --add-self-matches 0 --spaced-kmer-mode 1 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --threads 10 --compressed 0 -v 3 \n",
      "\n",
      "Query database size: 287208 type: Aminoacid\n",
      "Estimated memory consumption: 1G\n",
      "Target database size: 287208 type: Aminoacid\n",
      "Index table k-mer threshold: 154 at k-mer size 6 \n",
      "Index table: counting k-mers\n",
      "[=================================================================] 287.21K 1s 825ms\n",
      "Index table: Masked residues: 2414592\n",
      "Index table: fill\n",
      "[=================================================================] 287.21K 1s 311ms\n",
      "Index statistics\n",
      "Entries:          57624967\n",
      "DB size:          818 MB\n",
      "Avg k-mer size:   0.900390\n",
      "Top 10 k-mers\n",
      "    FAGTGK\t2408\n",
      "    AYQGKT\t2199\n",
      "    GQGGYT\t927\n",
      "    YGTSTY\t897\n",
      "    SMNGTN\t757\n",
      "    GQGGFT\t723\n",
      "    HIYDLT\t573\n",
      "    HVYDLT\t568\n",
      "    KNGCCW\t461\n",
      "    LGGFVY\t432\n",
      "Time for index table init: 0h 0m 3s 550ms\n",
      "Process prefiltering step 1 of 1\n",
      "\n",
      "k-mer similarity threshold: 154\n",
      "Starting prefiltering scores calculation (step 1 of 1)\n",
      "Query db start 1 to 287208\n",
      "Target db start 1 to 287208\n",
      "[=================================================================] 287.21K 5s 300ms\n",
      "\n",
      "1.961322 k-mers per position\n",
      "2357 DB matches per sequence\n",
      "0 overflows\n",
      "0 queries produce too many hits (truncated result)\n",
      "17 sequences passed prefiltering per query sequence\n",
      "20 median result list length\n",
      "0 sequences with 0 size result lists\n",
      "Time for merging to pref_step0: 0h 0m 0s 94ms\n",
      "Time for processing: 0h 0m 9s 310ms\n",
      "align tmp/8656781693095139340/input_step_redundancy tmp/8656781693095139340/input_step_redundancy tmp/8656781693095139340/pref_step0 tmp/8656781693095139340/aln_step0 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' -a 0 --alignment-mode 3 --alignment-output-mode 0 --wrapped-scoring 0 -e 0.001 --min-seq-id 0.9 --min-aln-len 0 --seq-id-mode 0 --alt-ali 0 -c 0.5 --cov-mode 1 --max-seq-len 65535 --comp-bias-corr 0 --comp-bias-corr-scale 1 --max-rejected 2147483647 --max-accept 2147483647 --add-self-matches 0 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --score-bias 0 --realign 0 --realign-score-bias -0.2 --realign-max-seqs 2147483647 --corr-score-weight 0 --gap-open aa:11,nucl:5 --gap-extend aa:1,nucl:2 --zdrop 40 --threads 10 --compressed 0 -v 3 \n",
      "\n",
      "Compute score, coverage and sequence identity\n",
      "Query database size: 287208 type: Aminoacid\n",
      "Target database size: 287208 type: Aminoacid\n",
      "Calculation of alignments\n",
      "[=================================================================] 287.21K 5m 58s 490ms\n",
      "Time for merging to aln_step0: 0h 0m 0s 168ms\n",
      "4452585 alignments calculated\n",
      "310352 sequence pairs passed the thresholds (0.069702 of overall calculated)\n",
      "1.080583 hits per query sequence\n",
      "Time for processing: 0h 5m 58s 830ms\n",
      "clust tmp/8656781693095139340/input_step_redundancy tmp/8656781693095139340/aln_step0 tmp/8656781693095139340/clu_step0 --cluster-mode 3 --max-iterations 1000 --similarity-type 2 --threads 10 --compressed 0 -v 3 \n",
      "\n",
      "Clustering mode: Greedy Low Mem\n",
      "Total time: 0h 0m 0s 31ms\n",
      "\n",
      "Size of the sequence database: 287208\n",
      "Size of the alignment database: 287208\n",
      "Number of clusters: 279340\n",
      "\n",
      "Writing results 0h 0m 0s 69ms\n",
      "Time for merging to clu_step0: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 134ms\n",
      "mergeclusters process/process_redundant_db process/process_clusters_redundant tmp/8656781693095139340/clu_redundancy tmp/8656781693095139340/clu_step0 --threads 10 --compressed 0 -v 3 \n",
      "\n",
      "Clustering step 1\n",
      "[=================================================================] 287.21K 0s 26ms\n",
      "Clustering step 2\n",
      "[=================================================================] 279.34K 0s 133ms\n",
      "Write merged clustering\n",
      "[=================================================================] 573.29K 0s 244ms\n",
      "Time for merging to process_clusters_redundant: 0h 0m 0s 291ms\n",
      "Time for processing: 0h 0m 0s 858ms\n",
      "Creating TSV files...\n",
      "createtsv function/function_redundant_db function/function_redundant_db function/function_clusters_redundant function/function_clusters_redundant.tsv \n",
      "\n",
      "MMseqs Version:                 \t14.7e284\n",
      "First sequence as representative\tfalse\n",
      "Target column                   \t1\n",
      "Add full header                 \tfalse\n",
      "Sequence source                 \t0\n",
      "Database output                 \tfalse\n",
      "Threads                         \t10\n",
      "Compressed                      \t0\n",
      "Verbosity                       \t3\n",
      "\n",
      "Time for merging to function_clusters_redundant.tsv: 0h 0m 0s 321ms\n",
      "Time for processing: 0h 0m 0s 720ms\n",
      "createtsv component/component_redundant_db component/component_redundant_db component/component_clusters_redundant component/component_clusters_redundant.tsv \n",
      "\n",
      "MMseqs Version:                 \t14.7e284\n",
      "First sequence as representative\tfalse\n",
      "Target column                   \t1\n",
      "Add full header                 \tfalse\n",
      "Sequence source                 \t0\n",
      "Database output                 \tfalse\n",
      "Threads                         \t10\n",
      "Compressed                      \t0\n",
      "Verbosity                       \t3\n",
      "\n",
      "Time for merging to component_clusters_redundant.tsv: 0h 0m 0s 103ms\n",
      "Time for processing: 0h 0m 0s 313ms\n",
      "createtsv process/process_redundant_db process/process_redundant_db process/process_clusters_redundant process/process_clusters_redundant.tsv \n",
      "\n",
      "MMseqs Version:                 \t14.7e284\n",
      "First sequence as representative\tfalse\n",
      "Target column                   \t1\n",
      "Add full header                 \tfalse\n",
      "Sequence source                 \t0\n",
      "Database output                 \tfalse\n",
      "Threads                         \t10\n",
      "Compressed                      \t0\n",
      "Verbosity                       \t3\n",
      "\n",
      "Time for merging to process_clusters_redundant.tsv: 0h 0m 0s 100ms\n",
      "Time for processing: 0h 0m 0s 257ms\n",
      "All tasks completed successfully.\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "CompletedProcess(args=['./mmseq_cluster.sh', '/Users/harsh/Documents/PFP_PUBLISH/raw_data_from_uniprot/reduced_90', '/Users/harsh/Documents/PFP_PUBLISH/raw_data_from_uniprot/total_SAR_function_for_reduce.fasta', '/Users/harsh/Documents/PFP_PUBLISH/raw_data_from_uniprot/total_SAR_component_for_reduce.fasta', '/Users/harsh/Documents/PFP_PUBLISH/raw_data_from_uniprot/total_SAR_process_for_reduce.fasta', '0.9'], returncode=0)"
      ]
     },
     "execution_count": 20,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "subprocess.run([\"./mmseq_cluster.sh\", \"reduced_90\", \"total_SAR_function_for_reduce.fasta\", \"total_SAR_component_for_reduce.fasta\", \"total_SAR_process_for_reduce.fasta\", \"0.9\"])"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Pick cluster representatives from reduced dataset (0.9)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(716735, 2)\n",
      "(573293, 2)\n",
      "(630944, 2)\n"
     ]
    }
   ],
   "source": [
    "SAR_function_clusters = pd.read_csv('reduced_90/function/function_clusters_redundant.tsv', sep='\\t', header=None)\n",
    "SAR_function_clusters.columns = [\"Cluster\", \"Entry\"]\n",
    "SAR_process_clusters = pd.read_csv('reduced_90/process/process_clusters_redundant.tsv', sep='\\t', header=None)\n",
    "SAR_process_clusters.columns = [\"Cluster\", \"Entry\"]\n",
    "SAR_component_clusters = pd.read_csv('reduced_90/component/component_clusters_redundant.tsv', sep='\\t', header=None)\n",
    "SAR_component_clusters.columns = [\"Cluster\", \"Entry\"]\n",
    "\n",
    "print(SAR_function_clusters.shape)\n",
    "print(SAR_process_clusters.shape)\n",
    "print(SAR_component_clusters.shape)\n",
    "\n",
    "assert SAR_function_clusters[\"Entry\"].isin(total_SAR_function[\"Entry\"]).all()\n",
    "assert SAR_process_clusters[\"Entry\"].isin(total_SAR_process[\"Entry\"]).all()\n",
    "assert SAR_component_clusters[\"Entry\"].isin(total_SAR_component[\"Entry\"]).all()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(352585, 11)\n",
      "(279340, 11)\n",
      "(306559, 11)\n"
     ]
    }
   ],
   "source": [
    "total_SAR_function_merged = pd.merge(total_SAR_function, SAR_function_clusters, on=\"Entry\") \n",
    "total_SAR_process_merged = pd.merge(total_SAR_process, SAR_process_clusters, on=\"Entry\")\n",
    "total_SAR_component_merged = pd.merge(total_SAR_component, SAR_component_clusters, on=\"Entry\")\n",
    "\n",
    "total_SAR_function_merged_grouped = total_SAR_function_merged.groupby(\"Cluster\")\n",
    "total_SAR_process_merged_grouped = total_SAR_process_merged.groupby(\"Cluster\")\n",
    "total_SAR_component_merged_grouped = total_SAR_component_merged.groupby(\"Cluster\")\n",
    "\n",
    "total_SAR_function_GO = total_SAR_function_merged_grouped.apply(lambda x: x.loc[x['Gene Ontology (molecular function)'].str.findall(r\"GO:\\d+\").str.len().idxmax()])\n",
    "total_SAR_process_GO = total_SAR_process_merged_grouped.apply(lambda x: x.loc[x['Gene Ontology (biological process)'].str.findall(r\"GO:\\d+\").str.len().idxmax()])\n",
    "total_SAR_component_GO = total_SAR_component_merged_grouped.apply(lambda x: x.loc[x['Gene Ontology (cellular component)'].str.findall(r\"GO:\\d+\").str.len().idxmax()])\n",
    "\n",
    "print(total_SAR_function_GO.shape)\n",
    "print(total_SAR_process_GO.shape)\n",
    "print(total_SAR_component_GO.shape)\n",
    "\n",
    "total_SAR_function_GO_terms = total_SAR_function_GO[\"Gene Ontology (molecular function)\"].str.findall(r\"GO:\\d+\")\n",
    "total_SAR_process_GO_terms = total_SAR_process_GO[\"Gene Ontology (biological process)\"].str.findall(r\"GO:\\d+\")\n",
    "total_SAR_component_GO_terms = total_SAR_component_GO[\"Gene Ontology (cellular component)\"].str.findall(r\"GO:\\d+\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "save "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [],
   "source": [
    "#create fasta records for total_SAR_function_GO, total_SAR_process_GO, total_SAR_component_GO\n",
    "seq_records_SAR_function_GO = create_seq_records(total_SAR_function_GO)\n",
    "seq_records_SAR_process_GO = create_seq_records(total_SAR_process_GO)\n",
    "seq_records_SAR_component_GO = create_seq_records(total_SAR_component_GO)\n",
    "\n",
    "assert len(seq_records_SAR_function_GO) == total_SAR_function_GO.shape[0]\n",
    "assert len(seq_records_SAR_process_GO) == total_SAR_process_GO.shape[0]\n",
    "assert len(seq_records_SAR_component_GO) == total_SAR_component_GO.shape[0]\n",
    "\n",
    "#write \n",
    "function_path = \"reduced_90_function.fasta\"\n",
    "process_path = \"reduced_90_process.fasta\"\n",
    "component_path = \"reduced_90_component.fasta\"\n",
    "\n",
    "SeqIO.write(seq_records_SAR_function_GO, function_path, \"fasta\")\n",
    "SeqIO.write(seq_records_SAR_process_GO, process_path, \"fasta\")\n",
    "SeqIO.write(seq_records_SAR_component_GO, component_path, \"fasta\")\n",
    "\n",
    "#write tsv files\n",
    "total_SAR_function_GO.to_csv(\"reduced_90_function.tsv\", sep=\"\\t\", index=False)\n",
    "total_SAR_process_GO.to_csv(\"reduced_90_process.tsv\", sep=\"\\t\", index=False)\n",
    "total_SAR_component_GO.to_csv(\"reduced_90_component.tsv\", sep=\"\\t\", index=False)\n",
    "\n",
    "#print out the shapes into a file \n",
    "with open('reduced_90_expected_shapes.txt', 'w') as f:\n",
    "    print(\"Function:\", total_SAR_function_GO.shape, file=f)\n",
    "    print(\"Process:\", total_SAR_process_GO.shape, file=f)\n",
    "    print(\"Component:\", total_SAR_component_GO.shape, file=f)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Propogate"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [],
   "source": [
    "total_SAR_function_GO_raw_terms = total_SAR_function_GO['Gene Ontology (molecular function)'].str.findall(r\"GO:\\d+\")\n",
    "total_SAR_process_GO_raw_terms = total_SAR_process_GO['Gene Ontology (biological process)'].str.findall(r\"GO:\\d+\")\n",
    "total_SAR_component_GO_raw_terms = total_SAR_component_GO['Gene Ontology (cellular component)'].str.findall(r\"GO:\\d+\")\n",
    "\n",
    "total_SAR_function_GO['Raw GO terms'] = total_SAR_function_GO_raw_terms\n",
    "total_SAR_process_GO['Raw GO terms'] = total_SAR_process_GO_raw_terms\n",
    "total_SAR_component_GO['Raw GO terms'] = total_SAR_component_GO_raw_terms"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Unique GO terms in function: 2608\n",
      "Unique GO terms in process: 2291\n",
      "Unique GO terms in component: 810\n"
     ]
    }
   ],
   "source": [
    "#get unique terms \n",
    "def get_unique_go_terms(go_terms):\n",
    "    return set([term for terms in go_terms for term in terms])\n",
    "\n",
    "unique_go_terms_function = get_unique_go_terms(total_SAR_function_GO_raw_terms)\n",
    "unique_go_terms_process = get_unique_go_terms(total_SAR_process_GO_raw_terms)\n",
    "unique_go_terms_component = get_unique_go_terms(total_SAR_component_GO_raw_terms)\n",
    "\n",
    "print(\"Unique GO terms in function:\", len(unique_go_terms_function))\n",
    "print(\"Unique GO terms in process:\", len(unique_go_terms_process))\n",
    "print(\"Unique GO terms in component:\", len(unique_go_terms_component))\n",
    "\n",
    "#conver to lists \n",
    "unique_go_terms_function = list(unique_go_terms_function)\n",
    "unique_go_terms_process = list(unique_go_terms_process)\n",
    "unique_go_terms_component = list(unique_go_terms_component)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 57,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "100%|██████████| 53/53 [00:43<00:00,  1.22it/s]\n",
      "100%|██████████| 46/46 [00:38<00:00,  1.18it/s]\n",
      "100%|██████████| 17/17 [00:13<00:00,  1.26it/s]\n"
     ]
    }
   ],
   "source": [
    "base_url = \"https://www.ebi.ac.uk/QuickGO/services/ontology/go/terms\"\n",
    "\n",
    "# Function to get ancestors for a batch of GO terms and store the JSON response\n",
    "def get_ancestors(go_terms_batch):\n",
    "    terms = \"%2C\".join(go_terms_batch)  # Join the GO terms with URL-encoded commas\n",
    "    request_url = f\"{base_url}/{terms}/ancestors?relations=is_a\"\n",
    "    \n",
    "    try:\n",
    "        response = requests.get(request_url, headers={\"Accept\": \"application/json\"})\n",
    "        response.raise_for_status()  # Raise exception if request failed\n",
    "\n",
    "        data = response.json()\n",
    "        \n",
    "        # Build the ancestors dictionary for the batch\n",
    "        ancestors_dict_batch = {}\n",
    "        json_dict_batch = {}\n",
    "        for result in data.get(\"results\", []):\n",
    "            term = result.get(\"id\")\n",
    "            ancestors = result.get(\"ancestors\", [])\n",
    "            ancestors_dict_batch[term] = ancestors if ancestors else None\n",
    "            json_dict_batch[term] = result  # Store only the specific JSON data for each term\n",
    "\n",
    "        return json_dict_batch, ancestors_dict_batch\n",
    "\n",
    "    except requests.exceptions.RequestException as e:\n",
    "        print(f\"Error fetching ancestors for {go_terms_batch}: {e}\")\n",
    "        return None, None\n",
    "\n",
    "def process_go_terms_in_batches(go_terms, batch_size=20):\n",
    "    function_ancestors_dict = {}      \n",
    "    function_json_dict = {}           \n",
    "\n",
    "    # Split the GO terms into batches\n",
    "    for i in tqdm(range(0, len(go_terms), batch_size)):\n",
    "        batch = go_terms[i:i + batch_size]  # Get the current batch of GO terms\n",
    "        json_data_batch, ancestors_dict_batch = get_ancestors(batch)\n",
    "\n",
    "        # Update the dictionaries with the batch results\n",
    "        if json_data_batch and ancestors_dict_batch:\n",
    "            function_json_dict.update(json_data_batch)  # Store only the relevant JSON for each term\n",
    "            function_ancestors_dict.update(ancestors_dict_batch)  # Store the ancestors\n",
    "\n",
    "    return function_json_dict, function_ancestors_dict\n",
    "\n",
    "batch_size = 50  \n",
    "\n",
    "function_json_dict, function_ancestors_dict = process_go_terms_in_batches(unique_go_terms_function, batch_size=batch_size)\n",
    "\n",
    "with open('function_ALL_data_ancestors_dict.json', 'w') as f:\n",
    "    json.dump(function_ancestors_dict, f, indent=4)\n",
    "with open('function_ALL_data_json_dict.json', 'w') as f:\n",
    "    json.dump(function_json_dict, f, indent=4)\n",
    "\n",
    "\n",
    "process_json_dict, process_ancestors_dict = process_go_terms_in_batches(unique_go_terms_process, batch_size=batch_size)\n",
    "with open('process_ALL_data_ancestors_dict.json', 'w') as f:\n",
    "    json.dump(process_ancestors_dict, f, indent=4)\n",
    "with open('process_ALL_data_json_dict.json', 'w') as f:\n",
    "    json.dump(process_json_dict, f, indent=4)\n",
    "\n",
    "assert len(process_json_dict) == len(unique_go_terms_process)\n",
    "assert len(process_ancestors_dict) == len(unique_go_terms_process)\n",
    "\n",
    "component_json_dict, component_ancestors_dict = process_go_terms_in_batches(unique_go_terms_component, batch_size=batch_size)\n",
    "with open('component_ALL_data_ancestors_dict.json', 'w') as f:\n",
    "    json.dump(component_ancestors_dict, f, indent=4)\n",
    "with open('component_ALL_data_json_dict.json', 'w') as f:\n",
    "    json.dump(component_json_dict, f, indent=4)\n",
    "\n",
    "assert len(component_json_dict) == len(unique_go_terms_component)\n",
    "assert len(component_ancestors_dict) == len(unique_go_terms_component)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 211,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Function Obselete: 53\n",
      "Process Obselete: 10\n",
      "Component Obselete: 12\n",
      "Total Obselete terms: 75\n"
     ]
    }
   ],
   "source": [
    "function_OBSELETE = [term for term in function_json_dict if function_json_dict[term].get(\"isObsolete\")]\n",
    "process_OBSELETE = [term for term in process_json_dict if process_json_dict[term].get(\"isObsolete\")]\n",
    "component_OBSELETE = [term for term in component_json_dict if component_json_dict[term].get(\"isObsolete\")]\n",
    "\n",
    "print(\"Function Obselete:\", len(function_OBSELETE))\n",
    "print(\"Process Obselete:\", len(process_OBSELETE))\n",
    "print(\"Component Obselete:\", len(component_OBSELETE))\n",
    "\n",
    "#combine and set \n",
    "ALL_OBSELETE_TERMS = set(function_OBSELETE + process_OBSELETE + component_OBSELETE)\n",
    "\n",
    "print(\"Total Obselete terms:\", len(ALL_OBSELETE_TERMS))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 72,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Adding ancestors: 100%|██████████| 352585/352585 [00:00<00:00, 413121.88it/s]\n",
      "Adding ancestors: 100%|██████████| 279340/279340 [00:00<00:00, 355595.35it/s]\n",
      "Adding ancestors: 100%|██████████| 306559/306559 [00:00<00:00, 619003.57it/s]\n"
     ]
    }
   ],
   "source": [
    "def add_ancestors(go_terms_list, ancestors_dict):\n",
    "    all_ancestors = []\n",
    "    \n",
    "    for term_list in tqdm(go_terms_list, desc=\"Adding ancestors\"):\n",
    "        terms_return = set() \n",
    "        for t in term_list:\n",
    "            # Check if the term exists in the dictionary and has valid ancestors\n",
    "            if t in ancestors_dict and ancestors_dict[t] is not None:\n",
    "                # Add ancestors, excluding the original term itself\n",
    "                terms_return.update(a for a in ancestors_dict[t] if a != t)\n",
    "        \n",
    "        # Append the filtered ancestors found for this protein (as a list)\n",
    "        all_ancestors.append(list(terms_return))\n",
    "\n",
    "    return all_ancestors\n",
    "\n",
    "total_SAR_function_GO_raw_terms_propogated = add_ancestors(list(total_SAR_function_GO_raw_terms), function_ancestors_dict)\n",
    "total_SAR_process_GO_raw_terms_propogated = add_ancestors(list(total_SAR_process_GO_raw_terms), process_ancestors_dict)\n",
    "total_SAR_component_GO_raw_terms_propogated = add_ancestors(list(total_SAR_component_GO_raw_terms), component_ancestors_dict)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 76,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Non-propogated function terms: 86\n",
      "Non-propogated process terms: 260\n",
      "Non-propogated component terms: 1937\n"
     ]
    }
   ],
   "source": [
    "#get the indicies for non-propogated terms (aka the len is 0)\n",
    "non_propogated_function_terms = [i for i, x in enumerate(total_SAR_function_GO_raw_terms_propogated) if len(x) == 0]\n",
    "non_propogated_process_terms = [i for i, x in enumerate(total_SAR_process_GO_raw_terms_propogated) if len(x) == 0]\n",
    "non_propogated_component_terms = [i for i, x in enumerate(total_SAR_component_GO_raw_terms_propogated) if len(x) == 0]\n",
    "\n",
    "print(\"Non-propogated function terms:\", len(non_propogated_function_terms))\n",
    "print(\"Non-propogated process terms:\", len(non_propogated_process_terms))\n",
    "print(\"Non-propogated component terms:\", len(non_propogated_component_terms))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 74,
   "metadata": {},
   "outputs": [],
   "source": [
    "total_SAR_function_GO['Raw propagated GO terms'] = total_SAR_function_GO_raw_terms_propogated\n",
    "total_SAR_process_GO['Raw propagated GO terms'] = total_SAR_process_GO_raw_terms_propogated\n",
    "total_SAR_component_GO['Raw propagated GO terms'] = total_SAR_component_GO_raw_terms_propogated"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 80,
   "metadata": {},
   "outputs": [],
   "source": [
    "#filter the dataframes to only include rows where the length of the Raw propagated GO terms is not zero \n",
    "total_SAR_function_GO = total_SAR_function_GO[total_SAR_function_GO['Raw propagated GO terms'].str.len() > 0]\n",
    "total_SAR_process_GO = total_SAR_process_GO[total_SAR_process_GO['Raw propagated GO terms'].str.len() > 0]\n",
    "total_SAR_component_GO = total_SAR_component_GO[total_SAR_component_GO['Raw propagated GO terms'].str.len() > 0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 82,
   "metadata": {},
   "outputs": [],
   "source": [
    "seq_records_SAR_function_GO_propagated = create_seq_records(total_SAR_function_GO)\n",
    "seq_records_SAR_process_GO_propagated = create_seq_records(total_SAR_process_GO)\n",
    "seq_records_SAR_component_GO_propagated = create_seq_records(total_SAR_component_GO)\n",
    "\n",
    "assert len(seq_records_SAR_function_GO_propagated) == total_SAR_function_GO.shape[0]\n",
    "assert len(seq_records_SAR_process_GO_propagated) == total_SAR_process_GO.shape[0]\n",
    "assert len(seq_records_SAR_component_GO_propagated) == total_SAR_component_GO.shape[0]\n",
    "\n",
    "function_path = \"reduced_90_function_propagated.fasta\"\n",
    "process_path = \"reduced_90_process_propagated.fasta\"\n",
    "component_path = \"reduced_90_component_propagated.fasta\"\n",
    "\n",
    "SeqIO.write(seq_records_SAR_function_GO_propagated, function_path, \"fasta\")\n",
    "SeqIO.write(seq_records_SAR_process_GO_propagated, process_path, \"fasta\")\n",
    "SeqIO.write(seq_records_SAR_component_GO_propagated, component_path, \"fasta\")\n",
    "\n",
    "total_SAR_function_GO.to_csv(\"reduced_90_function_propagated.tsv\", sep=\"\\t\", index=False)\n",
    "total_SAR_process_GO.to_csv(\"reduced_90_process_propagated.tsv\", sep=\"\\t\", index=False)\n",
    "total_SAR_component_GO.to_csv(\"reduced_90_component_propagated.tsv\", sep=\"\\t\", index=False)\n",
    "\n",
    "with open('reduced_90_expected_shapes_propagated.txt', 'w') as f:\n",
    "    print(\"Function:\", total_SAR_function_GO.shape, file=f)\n",
    "    print(\"Process:\", total_SAR_process_GO.shape, file=f)\n",
    "    print(\"Component:\", total_SAR_component_GO.shape, file=f)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Now we can preform the sequence similarity splits for the train, dev, and test sets"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 83,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Creating databases...\n",
      "createdb /Users/harsh/Documents/PFP_PUBLISH/raw_data_from_uniprot/reduced_90_function_propagated.fasta function/function_redundant_db \n",
      "\n",
      "MMseqs Version:       \t14.7e284\n",
      "Database type         \t0\n",
      "Shuffle input database\ttrue\n",
      "Createdb mode         \t0\n",
      "Write lookup file     \t1\n",
      "Offset of numeric ids \t0\n",
      "Compressed            \t0\n",
      "Verbosity             \t3\n",
      "\n",
      "Converting sequences\n",
      "[===================================\n",
      "Time for merging to function_redundant_db_h: 0h 0m 0s 62ms\n",
      "Time for merging to function_redundant_db: 0h 0m 0s 108ms\n",
      "Database type: Aminoacid\n",
      "Time for processing: 0h 0m 1s 237ms\n",
      "createdb /Users/harsh/Documents/PFP_PUBLISH/raw_data_from_uniprot/reduced_90_component_propagated.fasta component/component_redundant_db \n",
      "\n",
      "MMseqs Version:       \t14.7e284\n",
      "Database type         \t0\n",
      "Shuffle input database\ttrue\n",
      "Createdb mode         \t0\n",
      "Write lookup file     \t1\n",
      "Offset of numeric ids \t0\n",
      "Compressed            \t0\n",
      "Verbosity             \t3\n",
      "\n",
      "Converting sequences\n",
      "[==============================\n",
      "Time for merging to component_redundant_db_h: 0h 0m 0s 60ms\n",
      "Time for merging to component_redundant_db: 0h 0m 0s 117ms\n",
      "Database type: Aminoacid\n",
      "Time for processing: 0h 0m 1s 37ms\n",
      "createdb /Users/harsh/Documents/PFP_PUBLISH/raw_data_from_uniprot/reduced_90_process_propagated.fasta process/process_redundant_db \n",
      "\n",
      "MMseqs Version:       \t14.7e284\n",
      "Database type         \t0\n",
      "Shuffle input database\ttrue\n",
      "Createdb mode         \t0\n",
      "Write lookup file     \t1\n",
      "Offset of numeric ids \t0\n",
      "Compressed            \t0\n",
      "Verbosity             \t3\n",
      "\n",
      "Converting sequences\n",
      "[===========================\n",
      "Time for merging to process_redundant_db_h: 0h 0m 0s 51ms\n",
      "Time for merging to process_redundant_db: 0h 0m 0s 113ms\n",
      "Database type: Aminoacid\n",
      "Time for processing: 0h 0m 0s 879ms\n",
      "Clustering sequences...\n",
      "Create directory tmp\n",
      "cluster function/function_redundant_db function/function_clusters_redundant tmp --min-seq-id 0.3 -c 0.5 --cov-mode 1 \n",
      "\n",
      "MMseqs Version:                     \t14.7e284\n",
      "Substitution matrix                 \taa:blosum62.out,nucl:nucleotide.out\n",
      "Seed substitution matrix            \taa:VTML80.out,nucl:nucleotide.out\n",
      "Sensitivity                         \t4\n",
      "k-mer length                        \t0\n",
      "k-score                             \tseq:2147483647,prof:2147483647\n",
      "Alphabet size                       \taa:21,nucl:5\n",
      "Max sequence length                 \t65535\n",
      "Max results per query               \t20\n",
      "Split database                      \t0\n",
      "Split mode                          \t2\n",
      "Split memory limit                  \t0\n",
      "Coverage threshold                  \t0.5\n",
      "Coverage mode                       \t1\n",
      "Compositional bias                  \t1\n",
      "Compositional bias                  \t1\n",
      "Diagonal scoring                    \ttrue\n",
      "Exact k-mer matching                \t0\n",
      "Mask residues                       \t1\n",
      "Mask residues probability           \t0.9\n",
      "Mask lower case residues            \t0\n",
      "Minimum diagonal score              \t15\n",
      "Selected taxa                       \t\n",
      "Include identical seq. id.          \tfalse\n",
      "Spaced k-mers                       \t1\n",
      "Preload mode                        \t0\n",
      "Pseudo count a                      \tsubstitution:1.100,context:1.400\n",
      "Pseudo count b                      \tsubstitution:4.100,context:5.800\n",
      "Spaced k-mer pattern                \t\n",
      "Local temporary path                \t\n",
      "Threads                             \t10\n",
      "Compressed                          \t0\n",
      "Verbosity                           \t3\n",
      "Add backtrace                       \tfalse\n",
      "Alignment mode                      \t3\n",
      "Alignment mode                      \t0\n",
      "Allow wrapped scoring               \tfalse\n",
      "E-value threshold                   \t0.001\n",
      "Seq. id. threshold                  \t0.3\n",
      "Min alignment length                \t0\n",
      "Seq. id. mode                       \t0\n",
      "Alternative alignments              \t0\n",
      "Max reject                          \t2147483647\n",
      "Max accept                          \t2147483647\n",
      "Score bias                          \t0\n",
      "Realign hits                        \tfalse\n",
      "Realign score bias                  \t-0.2\n",
      "Realign max seqs                    \t2147483647\n",
      "Correlation score weight            \t0\n",
      "Gap open cost                       \taa:11,nucl:5\n",
      "Gap extension cost                  \taa:1,nucl:2\n",
      "Zdrop                               \t40\n",
      "Rescore mode                        \t0\n",
      "Remove hits by seq. id. and coverage\tfalse\n",
      "Sort results                        \t0\n",
      "Cluster mode                        \t0\n",
      "Max connected component depth       \t1000\n",
      "Similarity type                     \t2\n",
      "Single step clustering              \tfalse\n",
      "Cascaded clustering steps           \t3\n",
      "Cluster reassign                    \tfalse\n",
      "Remove temporary files              \tfalse\n",
      "Force restart with latest tmp       \tfalse\n",
      "MPI runner                          \t\n",
      "k-mers per sequence                 \t21\n",
      "Scale k-mers per sequence           \taa:0.000,nucl:0.200\n",
      "Adjust k-mer length                 \tfalse\n",
      "Shift hash                          \t67\n",
      "Include only extendable             \tfalse\n",
      "Skip repeating k-mers               \tfalse\n",
      "\n",
      "Set cluster sensitivity to -s 5.000000\n",
      "Set cluster mode GREEDY MEM\n",
      "Set cluster iterations to 3\n",
      "linclust function/function_redundant_db tmp/7310995555670121402/clu_redundancy tmp/7310995555670121402/linclust --cluster-mode 3 --max-iterations 1000 --similarity-type 2 --threads 10 --compressed 0 -v 3 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' -a 0 --alignment-mode 3 --alignment-output-mode 0 --wrapped-scoring 0 -e 0.001 --min-seq-id 0.3 --min-aln-len 0 --seq-id-mode 0 --alt-ali 0 -c 0.5 --cov-mode 1 --max-seq-len 65535 --comp-bias-corr 1 --comp-bias-corr-scale 1 --max-rejected 2147483647 --max-accept 2147483647 --add-self-matches 0 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --score-bias 0 --realign 0 --realign-score-bias -0.2 --realign-max-seqs 2147483647 --corr-score-weight 0 --gap-open aa:11,nucl:5 --gap-extend aa:1,nucl:2 --zdrop 40 --alph-size aa:13,nucl:5 --kmer-per-seq 21 --spaced-kmer-mode 1 --kmer-per-seq-scale aa:0.000,nucl:0.200 --adjust-kmer-len 0 --mask 0 --mask-prob 0.9 --mask-lower-case 0 -k 0 --hash-shift 67 --split-memory-limit 0 --include-only-extendable 0 --ignore-multi-kmer 0 --rescore-mode 0 --filter-hits 0 --sort-results 0 --remove-tmp-files 0 --force-reuse 0 \n",
      "\n",
      "kmermatcher function/function_redundant_db tmp/7310995555670121402/linclust/11900443378622354739/pref --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' --alph-size aa:13,nucl:5 --min-seq-id 0.3 --kmer-per-seq 21 --spaced-kmer-mode 1 --kmer-per-seq-scale aa:0.000,nucl:0.200 --adjust-kmer-len 0 --mask 0 --mask-prob 0.9 --mask-lower-case 0 --cov-mode 1 -k 0 -c 0.5 --max-seq-len 65535 --hash-shift 67 --split-memory-limit 0 --include-only-extendable 0 --ignore-multi-kmer 0 --threads 10 --compressed 0 -v 3 \n",
      "\n",
      "kmermatcher function/function_redundant_db tmp/7310995555670121402/linclust/11900443378622354739/pref --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' --alph-size aa:13,nucl:5 --min-seq-id 0.3 --kmer-per-seq 21 --spaced-kmer-mode 1 --kmer-per-seq-scale aa:0.000,nucl:0.200 --adjust-kmer-len 0 --mask 0 --mask-prob 0.9 --mask-lower-case 0 --cov-mode 1 -k 0 -c 0.5 --max-seq-len 65535 --hash-shift 67 --split-memory-limit 0 --include-only-extendable 0 --ignore-multi-kmer 0 --threads 10 --compressed 0 -v 3 \n",
      "\n",
      "Database size: 352499 type: Aminoacid\n",
      "Reduced amino acid alphabet: (A S T) (C) (D B N) (E Q Z) (F Y) (G) (H) (I V) (K R) (L J M) (P) (W) (X) \n",
      "\n",
      "Generate k-mers list for 1 split\n",
      "[=================================================================] 352.50K 0s 680ms\n",
      "Sort kmer 0h 0m 0s 160ms\n",
      "Sort by rep. sequence 0h 0m 0s 46ms\n",
      "Time for fill: 0h 0m 0s 95ms\n",
      "Time for merging to pref: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 1s 174ms\n",
      "rescorediagonal function/function_redundant_db function/function_redundant_db tmp/7310995555670121402/linclust/11900443378622354739/pref tmp/7310995555670121402/linclust/11900443378622354739/pref_rescore1 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' --rescore-mode 0 --wrapped-scoring 0 --filter-hits 0 -e 0.001 -c 0.5 -a 0 --cov-mode 1 --min-seq-id 0.5 --min-aln-len 0 --seq-id-mode 0 --add-self-matches 0 --sort-results 0 --db-load-mode 0 --threads 10 --compressed 0 -v 3 \n",
      "\n",
      "[=================================================================] 352.50K 0s 134ms\n",
      "Time for merging to pref_rescore1: 0h 0m 0s 145ms\n",
      "Time for processing: 0h 0m 0s 375ms\n",
      "clust function/function_redundant_db tmp/7310995555670121402/linclust/11900443378622354739/pref_rescore1 tmp/7310995555670121402/linclust/11900443378622354739/pre_clust --cluster-mode 3 --max-iterations 1000 --similarity-type 2 --threads 10 --compressed 0 -v 3 \n",
      "\n",
      "Clustering mode: Greedy Low Mem\n",
      "Total time: 0h 0m 0s 43ms\n",
      "\n",
      "Size of the sequence database: 352499\n",
      "Size of the alignment database: 352499\n",
      "Number of clusters: 240467\n",
      "\n",
      "Writing results 0h 0m 0s 62ms\n",
      "Time for merging to pre_clust: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 144ms\n",
      "createsubdb tmp/7310995555670121402/linclust/11900443378622354739/order_redundancy function/function_redundant_db tmp/7310995555670121402/linclust/11900443378622354739/input_step_redundancy -v 3 --subdb-mode 1 \n",
      "\n",
      "Time for merging to input_step_redundancy: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 80ms\n",
      "createsubdb tmp/7310995555670121402/linclust/11900443378622354739/order_redundancy tmp/7310995555670121402/linclust/11900443378622354739/pref tmp/7310995555670121402/linclust/11900443378622354739/pref_filter1 -v 3 --subdb-mode 1 \n",
      "\n",
      "Time for merging to pref_filter1: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 87ms\n",
      "filterdb tmp/7310995555670121402/linclust/11900443378622354739/pref_filter1 tmp/7310995555670121402/linclust/11900443378622354739/pref_filter2 --filter-file tmp/7310995555670121402/linclust/11900443378622354739/order_redundancy --threads 10 --compressed 0 -v 3 \n",
      "\n",
      "Filtering using file(s)\n",
      "[=================================================================] 240.47K 0s 64ms\n",
      "Time for merging to pref_filter2: 0h 0m 0s 141ms\n",
      "Time for processing: 0h 0m 0s 318ms\n",
      "rescorediagonal tmp/7310995555670121402/linclust/11900443378622354739/input_step_redundancy tmp/7310995555670121402/linclust/11900443378622354739/input_step_redundancy tmp/7310995555670121402/linclust/11900443378622354739/pref_filter2 tmp/7310995555670121402/linclust/11900443378622354739/pref_rescore2 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' --rescore-mode 1 --wrapped-scoring 0 --filter-hits 1 -e 0.001 -c 0.5 -a 0 --cov-mode 1 --min-seq-id 0.3 --min-aln-len 0 --seq-id-mode 0 --add-self-matches 0 --sort-results 0 --db-load-mode 0 --threads 10 --compressed 0 -v 3 \n",
      "\n",
      "[=================================================================] 240.47K 0s 128ms\n",
      "Time for merging to pref_rescore2: 0h 0m 0s 130ms\n",
      "Time for processing: 0h 0m 0s 349ms\n",
      "align tmp/7310995555670121402/linclust/11900443378622354739/input_step_redundancy tmp/7310995555670121402/linclust/11900443378622354739/input_step_redundancy tmp/7310995555670121402/linclust/11900443378622354739/pref_rescore2 tmp/7310995555670121402/linclust/11900443378622354739/aln --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' -a 0 --alignment-mode 3 --alignment-output-mode 0 --wrapped-scoring 0 -e 0.001 --min-seq-id 0.3 --min-aln-len 0 --seq-id-mode 0 --alt-ali 0 -c 0.5 --cov-mode 1 --max-seq-len 65535 --comp-bias-corr 1 --comp-bias-corr-scale 1 --max-rejected 2147483647 --max-accept 2147483647 --add-self-matches 0 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --score-bias 0 --realign 0 --realign-score-bias -0.2 --realign-max-seqs 2147483647 --corr-score-weight 0 --gap-open aa:11,nucl:5 --gap-extend aa:1,nucl:2 --zdrop 40 --threads 10 --compressed 0 -v 3 \n",
      "\n",
      "Compute score, coverage and sequence identity\n",
      "Query database size: 240467 type: Aminoacid\n",
      "Target database size: 240467 type: Aminoacid\n",
      "Calculation of alignments\n",
      "[=================================================================] 240.47K 35s 835ms\n",
      "Time for merging to aln: 0h 0m 0s 203ms\n",
      "587088 alignments calculated\n",
      "532946 sequence pairs passed the thresholds (0.907779 of overall calculated)\n",
      "2.216296 hits per query sequence\n",
      "Time for processing: 0h 0m 36s 179ms\n",
      "clust tmp/7310995555670121402/linclust/11900443378622354739/input_step_redundancy tmp/7310995555670121402/linclust/11900443378622354739/aln tmp/7310995555670121402/linclust/11900443378622354739/clust --cluster-mode 3 --max-iterations 1000 --similarity-type 2 --threads 10 --compressed 0 -v 3 \n",
      "\n",
      "Clustering mode: Greedy Low Mem\n",
      "Total time: 0h 0m 0s 91ms\n",
      "\n",
      "Size of the sequence database: 240467\n",
      "Size of the alignment database: 240467\n",
      "Number of clusters: 129277\n",
      "\n",
      "Writing results 0h 0m 0s 35ms\n",
      "Time for merging to clust: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 198ms\n",
      "mergeclusters function/function_redundant_db tmp/7310995555670121402/clu_redundancy tmp/7310995555670121402/linclust/11900443378622354739/pre_clust tmp/7310995555670121402/linclust/11900443378622354739/clust --threads 10 --compressed 0 -v 3 \n",
      "\n",
      "Clustering step 1\n",
      "[=================================================================] 240.47K 0s 22ms\n",
      "Clustering step 2\n",
      "[=================================================================] 129.28K 0s 44ms\n",
      "Write merged clustering\n",
      "[=================================================================] 352.50K 0s 98ms\n",
      "Time for merging to clu_redundancy: 0h 0m 0s 106ms\n",
      "Time for processing: 0h 0m 0s 287ms\n",
      "createsubdb tmp/7310995555670121402/clu_redundancy function/function_redundant_db tmp/7310995555670121402/input_step_redundancy -v 3 --subdb-mode 1 \n",
      "\n",
      "Time for merging to input_step_redundancy: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 56ms\n",
      "prefilter tmp/7310995555670121402/input_step_redundancy tmp/7310995555670121402/input_step_redundancy tmp/7310995555670121402/pref_step0 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' --seed-sub-mat 'aa:VTML80.out,nucl:nucleotide.out' -s 1 -k 0 --k-score seq:2147483647,prof:2147483647 --alph-size aa:21,nucl:5 --max-seq-len 65535 --max-seqs 20 --split 0 --split-mode 2 --split-memory-limit 0 -c 0.5 --cov-mode 1 --comp-bias-corr 0 --comp-bias-corr-scale 1 --diag-score 0 --exact-kmer-matching 0 --mask 1 --mask-prob 0.9 --mask-lower-case 0 --min-ungapped-score 0 --add-self-matches 0 --spaced-kmer-mode 1 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --threads 10 --compressed 0 -v 3 \n",
      "\n",
      "Query database size: 129277 type: Aminoacid\n",
      "Estimated memory consumption: 1G\n",
      "Target database size: 129277 type: Aminoacid\n",
      "Index table k-mer threshold: 154 at k-mer size 6 \n",
      "Index table: counting k-mers\n",
      "[=================================================================] 129.28K 0s 906ms\n",
      "Index table: Masked residues: 1472848\n",
      "Index table: fill\n",
      "[=================================================================] 129.28K 0s 663ms\n",
      "Index statistics\n",
      "Entries:          28536038\n",
      "DB size:          651 MB\n",
      "Avg k-mer size:   0.445876\n",
      "Top 10 k-mers\n",
      "    FAGTGK\t753\n",
      "    AYQGKT\t706\n",
      "    HRLPIL\t384\n",
      "    LGGFVY\t327\n",
      "    GQGGYT\t301\n",
      "    GPGGTL\t277\n",
      "    YGTSTY\t273\n",
      "    VHDKNL\t243\n",
      "    VHDKNV\t233\n",
      "    GLNGCY\t219\n",
      "Time for index table init: 0h 0m 1s 877ms\n",
      "Process prefiltering step 1 of 1\n",
      "\n",
      "k-mer similarity threshold: 154\n",
      "Starting prefiltering scores calculation (step 1 of 1)\n",
      "Query db start 1 to 129277\n",
      "Target db start 1 to 129277\n",
      "[=================================================================] 129.28K 2s 257ms\n",
      "\n",
      "2.099023 k-mers per position\n",
      "977 DB matches per sequence\n",
      "0 overflows\n",
      "0 queries produce too many hits (truncated result)\n",
      "11 sequences passed prefiltering per query sequence\n",
      "11 median result list length\n",
      "0 sequences with 0 size result lists\n",
      "Time for merging to pref_step0: 0h 0m 0s 47ms\n",
      "Time for processing: 0h 0m 4s 512ms\n",
      "align tmp/7310995555670121402/input_step_redundancy tmp/7310995555670121402/input_step_redundancy tmp/7310995555670121402/pref_step0 tmp/7310995555670121402/aln_step0 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' -a 0 --alignment-mode 3 --alignment-output-mode 0 --wrapped-scoring 0 -e 0.001 --min-seq-id 0.3 --min-aln-len 0 --seq-id-mode 0 --alt-ali 0 -c 0.5 --cov-mode 1 --max-seq-len 65535 --comp-bias-corr 0 --comp-bias-corr-scale 1 --max-rejected 2147483647 --max-accept 2147483647 --add-self-matches 0 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --score-bias 0 --realign 0 --realign-score-bias -0.2 --realign-max-seqs 2147483647 --corr-score-weight 0 --gap-open aa:11,nucl:5 --gap-extend aa:1,nucl:2 --zdrop 40 --threads 10 --compressed 0 -v 3 \n",
      "\n",
      "Compute score, coverage and sequence identity\n",
      "Query database size: 129277 type: Aminoacid\n",
      "Target database size: 129277 type: Aminoacid\n",
      "Calculation of alignments\n",
      "[=================================================================] 129.28K 1m 53s 438ms\n",
      "Time for merging to aln_step0: 0h 0m 0s 255ms\n",
      "1318669 alignments calculated\n",
      "755619 sequence pairs passed the thresholds (0.573016 of overall calculated)\n",
      "5.844961 hits per query sequence\n",
      "Time for processing: 0h 1m 53s 809ms\n",
      "clust tmp/7310995555670121402/input_step_redundancy tmp/7310995555670121402/aln_step0 tmp/7310995555670121402/clu_step0 --cluster-mode 3 --max-iterations 1000 --similarity-type 2 --threads 10 --compressed 0 -v 3 \n",
      "\n",
      "Clustering mode: Greedy Low Mem\n",
      "Total time: 0h 0m 0s 310ms\n",
      "\n",
      "Size of the sequence database: 129277\n",
      "Size of the alignment database: 129277\n",
      "Number of clusters: 63992\n",
      "\n",
      "Writing results 0h 0m 0s 18ms\n",
      "Time for merging to clu_step0: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 602ms\n",
      "createsubdb tmp/7310995555670121402/clu_step0 tmp/7310995555670121402/input_step_redundancy tmp/7310995555670121402/input_step1 -v 3 --subdb-mode 1 \n",
      "\n",
      "Time for merging to input_step1: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 29ms\n",
      "prefilter tmp/7310995555670121402/input_step1 tmp/7310995555670121402/input_step1 tmp/7310995555670121402/pref_step1 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' --seed-sub-mat 'aa:VTML80.out,nucl:nucleotide.out' -s 3 -k 0 --k-score seq:2147483647,prof:2147483647 --alph-size aa:21,nucl:5 --max-seq-len 65535 --max-seqs 20 --split 0 --split-mode 2 --split-memory-limit 0 -c 0.5 --cov-mode 1 --comp-bias-corr 1 --comp-bias-corr-scale 1 --diag-score 1 --exact-kmer-matching 0 --mask 1 --mask-prob 0.9 --mask-lower-case 0 --min-ungapped-score 15 --add-self-matches 0 --spaced-kmer-mode 1 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --threads 10 --compressed 0 -v 3 \n",
      "\n",
      "Query database size: 63992 type: Aminoacid\n",
      "Estimated memory consumption: 1G\n",
      "Target database size: 63992 type: Aminoacid\n",
      "Index table k-mer threshold: 136 at k-mer size 6 \n",
      "Index table: counting k-mers\n",
      "[=================================================================] 63.99K 0s 600ms\n",
      "Index table: Masked residues: 987445\n",
      "Index table: fill\n",
      "[=================================================================] 63.99K 0s 614ms\n",
      "Index statistics\n",
      "Entries:          30542156\n",
      "DB size:          663 MB\n",
      "Avg k-mer size:   0.477221\n",
      "Top 10 k-mers\n",
      "    IHDKNI\t292\n",
      "    FAGTGK\t279\n",
      "    AYQGKT\t276\n",
      "    QTSKTM\t218\n",
      "    KLLDGS\t144\n",
      "    LGGFVY\t141\n",
      "    SGTALP\t121\n",
      "    GLNGCY\t119\n",
      "    YGTSTY\t118\n",
      "    IGGFVY\t104\n",
      "Time for index table init: 0h 0m 1s 600ms\n",
      "Process prefiltering step 1 of 1\n",
      "\n",
      "k-mer similarity threshold: 136\n",
      "Starting prefiltering scores calculation (step 1 of 1)\n",
      "Query db start 1 to 63992\n",
      "Target db start 1 to 63992\n",
      "[=================================================================] 63.99K 5s 947ms\n",
      "\n",
      "20.735735 k-mers per position\n",
      "4322 DB matches per sequence\n",
      "0 overflows\n",
      "0 queries produce too many hits (truncated result)\n",
      "16 sequences passed prefiltering per query sequence\n",
      "20 median result list length\n",
      "0 sequences with 0 size result lists\n",
      "Time for merging to pref_step1: 0h 0m 0s 42ms\n",
      "Time for processing: 0h 0m 7s 944ms\n",
      "align tmp/7310995555670121402/input_step1 tmp/7310995555670121402/input_step1 tmp/7310995555670121402/pref_step1 tmp/7310995555670121402/aln_step1 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' -a 0 --alignment-mode 3 --alignment-output-mode 0 --wrapped-scoring 0 -e 0.001 --min-seq-id 0.3 --min-aln-len 0 --seq-id-mode 0 --alt-ali 0 -c 0.5 --cov-mode 1 --max-seq-len 65535 --comp-bias-corr 1 --comp-bias-corr-scale 1 --max-rejected 2147483647 --max-accept 2147483647 --add-self-matches 0 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --score-bias 0 --realign 0 --realign-score-bias -0.2 --realign-max-seqs 2147483647 --corr-score-weight 0 --gap-open aa:11,nucl:5 --gap-extend aa:1,nucl:2 --zdrop 40 --threads 10 --compressed 0 -v 3 \n",
      "\n",
      "Compute score, coverage and sequence identity\n",
      "Query database size: 63992 type: Aminoacid\n",
      "Target database size: 63992 type: Aminoacid\n",
      "Calculation of alignments\n",
      "[=================================================================] 63.99K 41s 969ms\n",
      "Time for merging to aln_step1: 0h 0m 0s 123ms\n",
      "868675 alignments calculated\n",
      "180523 sequence pairs passed the thresholds (0.207814 of overall calculated)\n",
      "2.821024 hits per query sequence\n",
      "Time for processing: 0h 0m 42s 198ms\n",
      "clust tmp/7310995555670121402/input_step1 tmp/7310995555670121402/aln_step1 tmp/7310995555670121402/clu_step1 --cluster-mode 3 --max-iterations 1000 --similarity-type 2 --threads 10 --compressed 0 -v 3 \n",
      "\n",
      "Clustering mode: Greedy Low Mem\n",
      "Total time: 0h 0m 0s 151ms\n",
      "\n",
      "Size of the sequence database: 63992\n",
      "Size of the alignment database: 63992\n",
      "Number of clusters: 42907\n",
      "\n",
      "Writing results 0h 0m 0s 11ms\n",
      "Time for merging to clu_step1: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 282ms\n",
      "createsubdb tmp/7310995555670121402/clu_step1 tmp/7310995555670121402/input_step1 tmp/7310995555670121402/input_step2 -v 3 --subdb-mode 1 \n",
      "\n",
      "Time for merging to input_step2: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 63ms\n",
      "prefilter tmp/7310995555670121402/input_step2 tmp/7310995555670121402/input_step2 tmp/7310995555670121402/pref_step2 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' --seed-sub-mat 'aa:VTML80.out,nucl:nucleotide.out' -s 5 -k 0 --k-score seq:2147483647,prof:2147483647 --alph-size aa:21,nucl:5 --max-seq-len 65535 --max-seqs 20 --split 0 --split-mode 2 --split-memory-limit 0 -c 0.5 --cov-mode 1 --comp-bias-corr 1 --comp-bias-corr-scale 1 --diag-score 1 --exact-kmer-matching 0 --mask 1 --mask-prob 0.9 --mask-lower-case 0 --min-ungapped-score 15 --add-self-matches 0 --spaced-kmer-mode 1 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --threads 10 --compressed 0 -v 3 \n",
      "\n",
      "Query database size: 42907 type: Aminoacid\n",
      "Estimated memory consumption: 1G\n",
      "Target database size: 42907 type: Aminoacid\n",
      "Index table k-mer threshold: 118 at k-mer size 6 \n",
      "Index table: counting k-mers\n",
      "[=================================================================] 42.91K 0s 500ms\n",
      "Index table: Masked residues: 812882\n",
      "Index table: fill\n",
      "[=================================================================] 42.91K 0s 488ms\n",
      "Index statistics\n",
      "Entries:          23484526\n",
      "DB size:          622 MB\n",
      "Avg k-mer size:   0.366946\n",
      "Top 10 k-mers\n",
      "    AYQGKT\t203\n",
      "    IHDKNI\t194\n",
      "    YRSLLL\t159\n",
      "    QTSKTM\t156\n",
      "    RDKELL\t116\n",
      "    KLRLSL\t115\n",
      "    SSSSSS\t106\n",
      "    GQGGYT\t99\n",
      "    GLNGCY\t91\n",
      "    LGGFVY\t89\n",
      "Time for index table init: 0h 0m 1s 317ms\n",
      "Process prefiltering step 1 of 1\n",
      "\n",
      "k-mer similarity threshold: 118\n",
      "Starting prefiltering scores calculation (step 1 of 1)\n",
      "Query db start 1 to 42907\n",
      "Target db start 1 to 42907\n",
      "[=================================================================] 42.91K 19s 136ms\n",
      "\n",
      "152.989696 k-mers per position\n",
      "27203 DB matches per sequence\n",
      "0 overflows\n",
      "0 queries produce too many hits (truncated result)\n",
      "19 sequences passed prefiltering per query sequence\n",
      "20 median result list length\n",
      "0 sequences with 0 size result lists\n",
      "Time for merging to pref_step2: 0h 0m 0s 166ms\n",
      "Time for processing: 0h 0m 20s 918ms\n",
      "align tmp/7310995555670121402/input_step2 tmp/7310995555670121402/input_step2 tmp/7310995555670121402/pref_step2 tmp/7310995555670121402/aln_step2 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' -a 0 --alignment-mode 3 --alignment-output-mode 0 --wrapped-scoring 0 -e 0.001 --min-seq-id 0.3 --min-aln-len 0 --seq-id-mode 0 --alt-ali 0 -c 0.5 --cov-mode 1 --max-seq-len 65535 --comp-bias-corr 1 --comp-bias-corr-scale 1 --max-rejected 2147483647 --max-accept 2147483647 --add-self-matches 0 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --score-bias 0 --realign 0 --realign-score-bias -0.2 --realign-max-seqs 2147483647 --corr-score-weight 0 --gap-open aa:11,nucl:5 --gap-extend aa:1,nucl:2 --zdrop 40 --threads 10 --compressed 0 -v 3 \n",
      "\n",
      "Compute score, coverage and sequence identity\n",
      "Query database size: 42907 type: Aminoacid\n",
      "Target database size: 42907 type: Aminoacid\n",
      "Calculation of alignments\n",
      "[=================================================================] 42.91K 24s 325ms\n",
      "Time for merging to aln_step2: 0h 0m 0s 22ms\n",
      "641657 alignments calculated\n",
      "65787 sequence pairs passed the thresholds (0.102527 of overall calculated)\n",
      "1.533246 hits per query sequence\n",
      "Time for processing: 0h 0m 24s 511ms\n",
      "clust tmp/7310995555670121402/input_step2 tmp/7310995555670121402/aln_step2 tmp/7310995555670121402/clu_step2 --cluster-mode 3 --max-iterations 1000 --similarity-type 2 --threads 10 --compressed 0 -v 3 \n",
      "\n",
      "Clustering mode: Greedy Low Mem\n",
      "Total time: 0h 0m 0s 19ms\n",
      "\n",
      "Size of the sequence database: 42907\n",
      "Size of the alignment database: 42907\n",
      "Number of clusters: 36398\n",
      "\n",
      "Writing results 0h 0m 0s 9ms\n",
      "Time for merging to clu_step2: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 35ms\n",
      "mergeclusters function/function_redundant_db function/function_clusters_redundant tmp/7310995555670121402/clu_redundancy tmp/7310995555670121402/clu_step0 tmp/7310995555670121402/clu_step1 tmp/7310995555670121402/clu_step2 --threads 10 --compressed 0 -v 3 \n",
      "\n",
      "Clustering step 1\n",
      "[=================================================================] 129.28K 0s 17ms\n",
      "Clustering step 2\n",
      "[=================================================================] 63.99K 0s 26ms\n",
      "Clustering step 3\n",
      "[=================================================================] 42.91K 0s 33ms\n",
      "Clustering step 4\n",
      "[=================================================================] 36.40K 0s 38ms\n",
      "Write merged clustering\n",
      "[=================================================================] 352.50K 0s 63ms\n",
      "Time for merging to function_clusters_redundant: 0h 0m 0s 49ms\n",
      "Time for processing: 0h 0m 0s 213ms\n",
      "cluster component/component_redundant_db component/component_clusters_redundant tmp --min-seq-id 0.3 -c 0.5 --cov-mode 1 \n",
      "\n",
      "MMseqs Version:                     \t14.7e284\n",
      "Substitution matrix                 \taa:blosum62.out,nucl:nucleotide.out\n",
      "Seed substitution matrix            \taa:VTML80.out,nucl:nucleotide.out\n",
      "Sensitivity                         \t4\n",
      "k-mer length                        \t0\n",
      "k-score                             \tseq:2147483647,prof:2147483647\n",
      "Alphabet size                       \taa:21,nucl:5\n",
      "Max sequence length                 \t65535\n",
      "Max results per query               \t20\n",
      "Split database                      \t0\n",
      "Split mode                          \t2\n",
      "Split memory limit                  \t0\n",
      "Coverage threshold                  \t0.5\n",
      "Coverage mode                       \t1\n",
      "Compositional bias                  \t1\n",
      "Compositional bias                  \t1\n",
      "Diagonal scoring                    \ttrue\n",
      "Exact k-mer matching                \t0\n",
      "Mask residues                       \t1\n",
      "Mask residues probability           \t0.9\n",
      "Mask lower case residues            \t0\n",
      "Minimum diagonal score              \t15\n",
      "Selected taxa                       \t\n",
      "Include identical seq. id.          \tfalse\n",
      "Spaced k-mers                       \t1\n",
      "Preload mode                        \t0\n",
      "Pseudo count a                      \tsubstitution:1.100,context:1.400\n",
      "Pseudo count b                      \tsubstitution:4.100,context:5.800\n",
      "Spaced k-mer pattern                \t\n",
      "Local temporary path                \t\n",
      "Threads                             \t10\n",
      "Compressed                          \t0\n",
      "Verbosity                           \t3\n",
      "Add backtrace                       \tfalse\n",
      "Alignment mode                      \t3\n",
      "Alignment mode                      \t0\n",
      "Allow wrapped scoring               \tfalse\n",
      "E-value threshold                   \t0.001\n",
      "Seq. id. threshold                  \t0.3\n",
      "Min alignment length                \t0\n",
      "Seq. id. mode                       \t0\n",
      "Alternative alignments              \t0\n",
      "Max reject                          \t2147483647\n",
      "Max accept                          \t2147483647\n",
      "Score bias                          \t0\n",
      "Realign hits                        \tfalse\n",
      "Realign score bias                  \t-0.2\n",
      "Realign max seqs                    \t2147483647\n",
      "Correlation score weight            \t0\n",
      "Gap open cost                       \taa:11,nucl:5\n",
      "Gap extension cost                  \taa:1,nucl:2\n",
      "Zdrop                               \t40\n",
      "Rescore mode                        \t0\n",
      "Remove hits by seq. id. and coverage\tfalse\n",
      "Sort results                        \t0\n",
      "Cluster mode                        \t0\n",
      "Max connected component depth       \t1000\n",
      "Similarity type                     \t2\n",
      "Single step clustering              \tfalse\n",
      "Cascaded clustering steps           \t3\n",
      "Cluster reassign                    \tfalse\n",
      "Remove temporary files              \tfalse\n",
      "Force restart with latest tmp       \tfalse\n",
      "MPI runner                          \t\n",
      "k-mers per sequence                 \t21\n",
      "Scale k-mers per sequence           \taa:0.000,nucl:0.200\n",
      "Adjust k-mer length                 \tfalse\n",
      "Shift hash                          \t67\n",
      "Include only extendable             \tfalse\n",
      "Skip repeating k-mers               \tfalse\n",
      "\n",
      "Set cluster sensitivity to -s 5.000000\n",
      "Set cluster mode GREEDY MEM\n",
      "Set cluster iterations to 3\n",
      "linclust component/component_redundant_db tmp/6254332096599620672/clu_redundancy tmp/6254332096599620672/linclust --cluster-mode 3 --max-iterations 1000 --similarity-type 2 --threads 10 --compressed 0 -v 3 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' -a 0 --alignment-mode 3 --alignment-output-mode 0 --wrapped-scoring 0 -e 0.001 --min-seq-id 0.3 --min-aln-len 0 --seq-id-mode 0 --alt-ali 0 -c 0.5 --cov-mode 1 --max-seq-len 65535 --comp-bias-corr 1 --comp-bias-corr-scale 1 --max-rejected 2147483647 --max-accept 2147483647 --add-self-matches 0 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --score-bias 0 --realign 0 --realign-score-bias -0.2 --realign-max-seqs 2147483647 --corr-score-weight 0 --gap-open aa:11,nucl:5 --gap-extend aa:1,nucl:2 --zdrop 40 --alph-size aa:13,nucl:5 --kmer-per-seq 21 --spaced-kmer-mode 1 --kmer-per-seq-scale aa:0.000,nucl:0.200 --adjust-kmer-len 0 --mask 0 --mask-prob 0.9 --mask-lower-case 0 -k 0 --hash-shift 67 --split-memory-limit 0 --include-only-extendable 0 --ignore-multi-kmer 0 --rescore-mode 0 --filter-hits 0 --sort-results 0 --remove-tmp-files 0 --force-reuse 0 \n",
      "\n",
      "kmermatcher component/component_redundant_db tmp/6254332096599620672/linclust/16851860129876914997/pref --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' --alph-size aa:13,nucl:5 --min-seq-id 0.3 --kmer-per-seq 21 --spaced-kmer-mode 1 --kmer-per-seq-scale aa:0.000,nucl:0.200 --adjust-kmer-len 0 --mask 0 --mask-prob 0.9 --mask-lower-case 0 --cov-mode 1 -k 0 -c 0.5 --max-seq-len 65535 --hash-shift 67 --split-memory-limit 0 --include-only-extendable 0 --ignore-multi-kmer 0 --threads 10 --compressed 0 -v 3 \n",
      "\n",
      "kmermatcher component/component_redundant_db tmp/6254332096599620672/linclust/16851860129876914997/pref --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' --alph-size aa:13,nucl:5 --min-seq-id 0.3 --kmer-per-seq 21 --spaced-kmer-mode 1 --kmer-per-seq-scale aa:0.000,nucl:0.200 --adjust-kmer-len 0 --mask 0 --mask-prob 0.9 --mask-lower-case 0 --cov-mode 1 -k 0 -c 0.5 --max-seq-len 65535 --hash-shift 67 --split-memory-limit 0 --include-only-extendable 0 --ignore-multi-kmer 0 --threads 10 --compressed 0 -v 3 \n",
      "\n",
      "Database size: 304622 type: Aminoacid\n",
      "Reduced amino acid alphabet: (A S T) (C) (D B N) (E Q Z) (F Y) (G) (H) (I V) (K R) (L J M) (P) (W) (X) \n",
      "\n",
      "Generate k-mers list for 1 split\n",
      "[=================================================================] 304.62K 0s 560ms\n",
      "Sort kmer 0h 0m 0s 132ms\n",
      "Sort by rep. sequence 0h 0m 0s 41ms\n",
      "Time for fill: 0h 0m 0s 79ms\n",
      "Time for merging to pref: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 1s 7ms\n",
      "rescorediagonal component/component_redundant_db component/component_redundant_db tmp/6254332096599620672/linclust/16851860129876914997/pref tmp/6254332096599620672/linclust/16851860129876914997/pref_rescore1 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' --rescore-mode 0 --wrapped-scoring 0 --filter-hits 0 -e 0.001 -c 0.5 -a 0 --cov-mode 1 --min-seq-id 0.5 --min-aln-len 0 --seq-id-mode 0 --add-self-matches 0 --sort-results 0 --db-load-mode 0 --threads 10 --compressed 0 -v 3 \n",
      "\n",
      "[=================================================================] 304.62K 0s 110ms\n",
      "Time for merging to pref_rescore1: 0h 0m 0s 141ms\n",
      "Time for processing: 0h 0m 0s 376ms\n",
      "clust component/component_redundant_db tmp/6254332096599620672/linclust/16851860129876914997/pref_rescore1 tmp/6254332096599620672/linclust/16851860129876914997/pre_clust --cluster-mode 3 --max-iterations 1000 --similarity-type 2 --threads 10 --compressed 0 -v 3 \n",
      "\n",
      "Clustering mode: Greedy Low Mem\n",
      "Total time: 0h 0m 0s 34ms\n",
      "\n",
      "Size of the sequence database: 304622\n",
      "Size of the alignment database: 304622\n",
      "Number of clusters: 210629\n",
      "\n",
      "Writing results 0h 0m 0s 54ms\n",
      "Time for merging to pre_clust: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 136ms\n",
      "createsubdb tmp/6254332096599620672/linclust/16851860129876914997/order_redundancy component/component_redundant_db tmp/6254332096599620672/linclust/16851860129876914997/input_step_redundancy -v 3 --subdb-mode 1 \n",
      "\n",
      "Time for merging to input_step_redundancy: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 72ms\n",
      "createsubdb tmp/6254332096599620672/linclust/16851860129876914997/order_redundancy tmp/6254332096599620672/linclust/16851860129876914997/pref tmp/6254332096599620672/linclust/16851860129876914997/pref_filter1 -v 3 --subdb-mode 1 \n",
      "\n",
      "Time for merging to pref_filter1: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 107ms\n",
      "filterdb tmp/6254332096599620672/linclust/16851860129876914997/pref_filter1 tmp/6254332096599620672/linclust/16851860129876914997/pref_filter2 --filter-file tmp/6254332096599620672/linclust/16851860129876914997/order_redundancy --threads 10 --compressed 0 -v 3 \n",
      "\n",
      "Filtering using file(s)\n",
      "[=================================================================] 210.63K 0s 62ms\n",
      "Time for merging to pref_filter2: 0h 0m 0s 207ms\n",
      "Time for processing: 0h 0m 0s 477ms\n",
      "rescorediagonal tmp/6254332096599620672/linclust/16851860129876914997/input_step_redundancy tmp/6254332096599620672/linclust/16851860129876914997/input_step_redundancy tmp/6254332096599620672/linclust/16851860129876914997/pref_filter2 tmp/6254332096599620672/linclust/16851860129876914997/pref_rescore2 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' --rescore-mode 1 --wrapped-scoring 0 --filter-hits 1 -e 0.001 -c 0.5 -a 0 --cov-mode 1 --min-seq-id 0.3 --min-aln-len 0 --seq-id-mode 0 --add-self-matches 0 --sort-results 0 --db-load-mode 0 --threads 10 --compressed 0 -v 3 \n",
      "\n",
      "[=================================================================] 210.63K 0s 111ms\n",
      "Time for merging to pref_rescore2: 0h 0m 0s 292ms\n",
      "Time for processing: 0h 0m 0s 537ms\n",
      "align tmp/6254332096599620672/linclust/16851860129876914997/input_step_redundancy tmp/6254332096599620672/linclust/16851860129876914997/input_step_redundancy tmp/6254332096599620672/linclust/16851860129876914997/pref_rescore2 tmp/6254332096599620672/linclust/16851860129876914997/aln --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' -a 0 --alignment-mode 3 --alignment-output-mode 0 --wrapped-scoring 0 -e 0.001 --min-seq-id 0.3 --min-aln-len 0 --seq-id-mode 0 --alt-ali 0 -c 0.5 --cov-mode 1 --max-seq-len 65535 --comp-bias-corr 1 --comp-bias-corr-scale 1 --max-rejected 2147483647 --max-accept 2147483647 --add-self-matches 0 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --score-bias 0 --realign 0 --realign-score-bias -0.2 --realign-max-seqs 2147483647 --corr-score-weight 0 --gap-open aa:11,nucl:5 --gap-extend aa:1,nucl:2 --zdrop 40 --threads 10 --compressed 0 -v 3 \n",
      "\n",
      "Compute score, coverage and sequence identity\n",
      "Query database size: 210629 type: Aminoacid\n",
      "Target database size: 210629 type: Aminoacid\n",
      "Calculation of alignments\n",
      "[=================================================================] 210.63K 27s 956ms\n",
      "Time for merging to aln: 0h 0m 0s 69ms\n",
      "488068 alignments calculated\n",
      "443474 sequence pairs passed the thresholds (0.908632 of overall calculated)\n",
      "2.105474 hits per query sequence\n",
      "Time for processing: 0h 0m 28s 102ms\n",
      "clust tmp/6254332096599620672/linclust/16851860129876914997/input_step_redundancy tmp/6254332096599620672/linclust/16851860129876914997/aln tmp/6254332096599620672/linclust/16851860129876914997/clust --cluster-mode 3 --max-iterations 1000 --similarity-type 2 --threads 10 --compressed 0 -v 3 \n",
      "\n",
      "Clustering mode: Greedy Low Mem\n",
      "Total time: 0h 0m 0s 26ms\n",
      "\n",
      "Size of the sequence database: 210629\n",
      "Size of the alignment database: 210629\n",
      "Number of clusters: 118803\n",
      "\n",
      "Writing results 0h 0m 0s 32ms\n",
      "Time for merging to clust: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 82ms\n",
      "mergeclusters component/component_redundant_db tmp/6254332096599620672/clu_redundancy tmp/6254332096599620672/linclust/16851860129876914997/pre_clust tmp/6254332096599620672/linclust/16851860129876914997/clust --threads 10 --compressed 0 -v 3 \n",
      "\n",
      "Clustering step 1\n",
      "[=================================================================] 210.63K 0s 19ms\n",
      "Clustering step 2\n",
      "[=================================================================] 118.80K 0s 37ms\n",
      "Write merged clustering\n",
      "[=================================================================] 304.62K 0s 62ms\n",
      "Time for merging to clu_redundancy: 0h 0m 0s 252ms\n",
      "Time for processing: 0h 0m 0s 423ms\n",
      "createsubdb tmp/6254332096599620672/clu_redundancy component/component_redundant_db tmp/6254332096599620672/input_step_redundancy -v 3 --subdb-mode 1 \n",
      "\n",
      "Time for merging to input_step_redundancy: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 49ms\n",
      "prefilter tmp/6254332096599620672/input_step_redundancy tmp/6254332096599620672/input_step_redundancy tmp/6254332096599620672/pref_step0 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' --seed-sub-mat 'aa:VTML80.out,nucl:nucleotide.out' -s 1 -k 0 --k-score seq:2147483647,prof:2147483647 --alph-size aa:21,nucl:5 --max-seq-len 65535 --max-seqs 20 --split 0 --split-mode 2 --split-memory-limit 0 -c 0.5 --cov-mode 1 --comp-bias-corr 0 --comp-bias-corr-scale 1 --diag-score 0 --exact-kmer-matching 0 --mask 1 --mask-prob 0.9 --mask-lower-case 0 --min-ungapped-score 0 --add-self-matches 0 --spaced-kmer-mode 1 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --threads 10 --compressed 0 -v 3 \n",
      "\n",
      "Query database size: 118803 type: Aminoacid\n",
      "Estimated memory consumption: 1G\n",
      "Target database size: 118803 type: Aminoacid\n",
      "Index table k-mer threshold: 154 at k-mer size 6 \n",
      "Index table: counting k-mers\n",
      "[=================================================================] 118.80K 0s 872ms\n",
      "Index table: Masked residues: 1372997\n",
      "Index table: fill\n",
      "[=================================================================] 118.80K 0s 566ms\n",
      "Index statistics\n",
      "Entries:          24791509\n",
      "DB size:          630 MB\n",
      "Avg k-mer size:   0.387367\n",
      "Top 10 k-mers\n",
      "    FAGTGK\t633\n",
      "    AYQGKT\t619\n",
      "    VHDKNI\t328\n",
      "    GQGGYT\t276\n",
      "    YGTSTY\t252\n",
      "    LGGFVY\t209\n",
      "    PYDKRL\t182\n",
      "    HVYDLT\t170\n",
      "    HHPICV\t157\n",
      "    VHDKNV\t154\n",
      "Time for index table init: 0h 0m 1s 801ms\n",
      "Process prefiltering step 1 of 1\n",
      "\n",
      "k-mer similarity threshold: 154\n",
      "Starting prefiltering scores calculation (step 1 of 1)\n",
      "Query db start 1 to 118803\n",
      "Target db start 1 to 118803\n",
      "[=================================================================] 118.80K 2s 1ms\n",
      "\n",
      "2.048060 k-mers per position\n",
      "817 DB matches per sequence\n",
      "0 overflows\n",
      "0 queries produce too many hits (truncated result)\n",
      "10 sequences passed prefiltering per query sequence\n",
      "8 median result list length\n",
      "0 sequences with 0 size result lists\n",
      "Time for merging to pref_step0: 0h 0m 0s 49ms\n",
      "Time for processing: 0h 0m 4s 155ms\n",
      "align tmp/6254332096599620672/input_step_redundancy tmp/6254332096599620672/input_step_redundancy tmp/6254332096599620672/pref_step0 tmp/6254332096599620672/aln_step0 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' -a 0 --alignment-mode 3 --alignment-output-mode 0 --wrapped-scoring 0 -e 0.001 --min-seq-id 0.3 --min-aln-len 0 --seq-id-mode 0 --alt-ali 0 -c 0.5 --cov-mode 1 --max-seq-len 65535 --comp-bias-corr 0 --comp-bias-corr-scale 1 --max-rejected 2147483647 --max-accept 2147483647 --add-self-matches 0 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --score-bias 0 --realign 0 --realign-score-bias -0.2 --realign-max-seqs 2147483647 --corr-score-weight 0 --gap-open aa:11,nucl:5 --gap-extend aa:1,nucl:2 --zdrop 40 --threads 10 --compressed 0 -v 3 \n",
      "\n",
      "Compute score, coverage and sequence identity\n",
      "Query database size: 118803 type: Aminoacid\n",
      "Target database size: 118803 type: Aminoacid\n",
      "Calculation of alignments\n",
      "[=================================================================] 118.80K 1m 32s 287ms\n",
      "Time for merging to aln_step0: 0h 0m 0s 236ms\n",
      "1072112 alignments calculated\n",
      "619046 sequence pairs passed the thresholds (0.577408 of overall calculated)\n",
      "5.210693 hits per query sequence\n",
      "Time for processing: 0h 1m 32s 666ms\n",
      "clust tmp/6254332096599620672/input_step_redundancy tmp/6254332096599620672/aln_step0 tmp/6254332096599620672/clu_step0 --cluster-mode 3 --max-iterations 1000 --similarity-type 2 --threads 10 --compressed 0 -v 3 \n",
      "\n",
      "Clustering mode: Greedy Low Mem\n",
      "Total time: 0h 0m 0s 42ms\n",
      "\n",
      "Size of the sequence database: 118803\n",
      "Size of the alignment database: 118803\n",
      "Number of clusters: 62469\n",
      "\n",
      "Writing results 0h 0m 0s 17ms\n",
      "Time for merging to clu_step0: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 224ms\n",
      "createsubdb tmp/6254332096599620672/clu_step0 tmp/6254332096599620672/input_step_redundancy tmp/6254332096599620672/input_step1 -v 3 --subdb-mode 1 \n",
      "\n",
      "Time for merging to input_step1: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 26ms\n",
      "prefilter tmp/6254332096599620672/input_step1 tmp/6254332096599620672/input_step1 tmp/6254332096599620672/pref_step1 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' --seed-sub-mat 'aa:VTML80.out,nucl:nucleotide.out' -s 3 -k 0 --k-score seq:2147483647,prof:2147483647 --alph-size aa:21,nucl:5 --max-seq-len 65535 --max-seqs 20 --split 0 --split-mode 2 --split-memory-limit 0 -c 0.5 --cov-mode 1 --comp-bias-corr 1 --comp-bias-corr-scale 1 --diag-score 1 --exact-kmer-matching 0 --mask 1 --mask-prob 0.9 --mask-lower-case 0 --min-ungapped-score 15 --add-self-matches 0 --spaced-kmer-mode 1 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --threads 10 --compressed 0 -v 3 \n",
      "\n",
      "Query database size: 62469 type: Aminoacid\n",
      "Estimated memory consumption: 1G\n",
      "Target database size: 62469 type: Aminoacid\n",
      "Index table k-mer threshold: 136 at k-mer size 6 \n",
      "Index table: counting k-mers\n",
      "[=================================================================] 62.47K 0s 588ms\n",
      "Index table: Masked residues: 917307\n",
      "Index table: fill\n",
      "[=================================================================] 62.47K 0s 562ms\n",
      "Index statistics\n",
      "Entries:          28162202\n",
      "DB size:          649 MB\n",
      "Avg k-mer size:   0.440034\n",
      "Top 10 k-mers\n",
      "    AYQGKT\t244\n",
      "    YRSLLL\t213\n",
      "    QTSKTM\t178\n",
      "    RDKELL\t136\n",
      "    KLLDGS\t135\n",
      "    GQGGYT\t116\n",
      "    LGGFVY\t103\n",
      "    HVYDLT\t84\n",
      "    VHDKNV\t76\n",
      "    IVRLEN\t68\n",
      "Time for index table init: 0h 0m 1s 492ms\n",
      "Process prefiltering step 1 of 1\n",
      "\n",
      "k-mer similarity threshold: 136\n",
      "Starting prefiltering scores calculation (step 1 of 1)\n",
      "Query db start 1 to 62469\n",
      "Target db start 1 to 62469\n",
      "[=================================================================] 62.47K 5s 427ms\n",
      "\n",
      "19.568138 k-mers per position\n",
      "3546 DB matches per sequence\n",
      "0 overflows\n",
      "0 queries produce too many hits (truncated result)\n",
      "15 sequences passed prefiltering per query sequence\n",
      "20 median result list length\n",
      "0 sequences with 0 size result lists\n",
      "Time for merging to pref_step1: 0h 0m 0s 208ms\n",
      "Time for processing: 0h 0m 7s 502ms\n",
      "align tmp/6254332096599620672/input_step1 tmp/6254332096599620672/input_step1 tmp/6254332096599620672/pref_step1 tmp/6254332096599620672/aln_step1 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' -a 0 --alignment-mode 3 --alignment-output-mode 0 --wrapped-scoring 0 -e 0.001 --min-seq-id 0.3 --min-aln-len 0 --seq-id-mode 0 --alt-ali 0 -c 0.5 --cov-mode 1 --max-seq-len 65535 --comp-bias-corr 1 --comp-bias-corr-scale 1 --max-rejected 2147483647 --max-accept 2147483647 --add-self-matches 0 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --score-bias 0 --realign 0 --realign-score-bias -0.2 --realign-max-seqs 2147483647 --corr-score-weight 0 --gap-open aa:11,nucl:5 --gap-extend aa:1,nucl:2 --zdrop 40 --threads 10 --compressed 0 -v 3 \n",
      "\n",
      "Compute score, coverage and sequence identity\n",
      "Query database size: 62469 type: Aminoacid\n",
      "Target database size: 62469 type: Aminoacid\n",
      "Calculation of alignments\n",
      "[=================================================================] 62.47K 35s 613ms\n",
      "Time for merging to aln_step1: 0h 0m 0s 21ms\n",
      "786913 alignments calculated\n",
      "161894 sequence pairs passed the thresholds (0.205733 of overall calculated)\n",
      "2.591589 hits per query sequence\n",
      "Time for processing: 0h 0m 35s 905ms\n",
      "clust tmp/6254332096599620672/input_step1 tmp/6254332096599620672/aln_step1 tmp/6254332096599620672/clu_step1 --cluster-mode 3 --max-iterations 1000 --similarity-type 2 --threads 10 --compressed 0 -v 3 \n",
      "\n",
      "Clustering mode: Greedy Low Mem\n",
      "Total time: 0h 0m 0s 14ms\n",
      "\n",
      "Size of the sequence database: 62469\n",
      "Size of the alignment database: 62469\n",
      "Number of clusters: 43236\n",
      "\n",
      "Writing results 0h 0m 0s 11ms\n",
      "Time for merging to clu_step1: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 121ms\n",
      "createsubdb tmp/6254332096599620672/clu_step1 tmp/6254332096599620672/input_step1 tmp/6254332096599620672/input_step2 -v 3 --subdb-mode 1 \n",
      "\n",
      "Time for merging to input_step2: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 51ms\n",
      "prefilter tmp/6254332096599620672/input_step2 tmp/6254332096599620672/input_step2 tmp/6254332096599620672/pref_step2 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' --seed-sub-mat 'aa:VTML80.out,nucl:nucleotide.out' -s 5 -k 0 --k-score seq:2147483647,prof:2147483647 --alph-size aa:21,nucl:5 --max-seq-len 65535 --max-seqs 20 --split 0 --split-mode 2 --split-memory-limit 0 -c 0.5 --cov-mode 1 --comp-bias-corr 1 --comp-bias-corr-scale 1 --diag-score 1 --exact-kmer-matching 0 --mask 1 --mask-prob 0.9 --mask-lower-case 0 --min-ungapped-score 15 --add-self-matches 0 --spaced-kmer-mode 1 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --threads 10 --compressed 0 -v 3 \n",
      "\n",
      "Query database size: 43236 type: Aminoacid\n",
      "Estimated memory consumption: 1G\n",
      "Target database size: 43236 type: Aminoacid\n",
      "Index table k-mer threshold: 118 at k-mer size 6 \n",
      "Index table: counting k-mers\n",
      "[=================================================================] 43.24K 0s 420ms\n",
      "Index table: Masked residues: 760587\n",
      "Index table: fill\n",
      "[=================================================================] 43.24K 0s 501ms\n",
      "Index statistics\n",
      "Entries:          22300983\n",
      "DB size:          615 MB\n",
      "Avg k-mer size:   0.348453\n",
      "Top 10 k-mers\n",
      "    AAAAAA\t194\n",
      "    SFRLYG\t182\n",
      "    AYQGKT\t178\n",
      "    YRSLLL\t148\n",
      "    QTSKTM\t133\n",
      "    SSSSSS\t110\n",
      "    AAAAAL\t103\n",
      "    VPRSTR\t96\n",
      "    KLLDGS\t92\n",
      "    GQGGYT\t91\n",
      "Time for index table init: 0h 0m 1s 215ms\n",
      "Process prefiltering step 1 of 1\n",
      "\n",
      "k-mer similarity threshold: 118\n",
      "Starting prefiltering scores calculation (step 1 of 1)\n",
      "Query db start 1 to 43236\n",
      "Target db start 1 to 43236\n",
      "[=================================================================] 43.24K 17s 745ms\n",
      "\n",
      "145.913965 k-mers per position\n",
      "23101 DB matches per sequence\n",
      "0 overflows\n",
      "0 queries produce too many hits (truncated result)\n",
      "19 sequences passed prefiltering per query sequence\n",
      "20 median result list length\n",
      "0 sequences with 0 size result lists\n",
      "Time for merging to pref_step2: 0h 0m 0s 39ms\n",
      "Time for processing: 0h 0m 19s 318ms\n",
      "align tmp/6254332096599620672/input_step2 tmp/6254332096599620672/input_step2 tmp/6254332096599620672/pref_step2 tmp/6254332096599620672/aln_step2 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' -a 0 --alignment-mode 3 --alignment-output-mode 0 --wrapped-scoring 0 -e 0.001 --min-seq-id 0.3 --min-aln-len 0 --seq-id-mode 0 --alt-ali 0 -c 0.5 --cov-mode 1 --max-seq-len 65535 --comp-bias-corr 1 --comp-bias-corr-scale 1 --max-rejected 2147483647 --max-accept 2147483647 --add-self-matches 0 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --score-bias 0 --realign 0 --realign-score-bias -0.2 --realign-max-seqs 2147483647 --corr-score-weight 0 --gap-open aa:11,nucl:5 --gap-extend aa:1,nucl:2 --zdrop 40 --threads 10 --compressed 0 -v 3 \n",
      "\n",
      "Compute score, coverage and sequence identity\n",
      "Query database size: 43236 type: Aminoacid\n",
      "Target database size: 43236 type: Aminoacid\n",
      "Calculation of alignments\n",
      "[=================================================================] 43.24K 21s 664ms\n",
      "Time for merging to aln_step2: 0h 0m 0s 229ms\n",
      "629428 alignments calculated\n",
      "66007 sequence pairs passed the thresholds (0.104868 of overall calculated)\n",
      "1.526668 hits per query sequence\n",
      "Time for processing: 0h 0m 22s 62ms\n",
      "clust tmp/6254332096599620672/input_step2 tmp/6254332096599620672/aln_step2 tmp/6254332096599620672/clu_step2 --cluster-mode 3 --max-iterations 1000 --similarity-type 2 --threads 10 --compressed 0 -v 3 \n",
      "\n",
      "Clustering mode: Greedy Low Mem\n",
      "Total time: 0h 0m 0s 278ms\n",
      "\n",
      "Size of the sequence database: 43236\n",
      "Size of the alignment database: 43236\n",
      "Number of clusters: 36571\n",
      "\n",
      "Writing results 0h 0m 0s 9ms\n",
      "Time for merging to clu_step2: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 550ms\n",
      "mergeclusters component/component_redundant_db component/component_clusters_redundant tmp/6254332096599620672/clu_redundancy tmp/6254332096599620672/clu_step0 tmp/6254332096599620672/clu_step1 tmp/6254332096599620672/clu_step2 --threads 10 --compressed 0 -v 3 \n",
      "\n",
      "Clustering step 1\n",
      "[=================================================================] 118.80K 0s 18ms\n",
      "Clustering step 2\n",
      "[=================================================================] 62.47K 0s 84ms\n",
      "Clustering step 3\n",
      "[=================================================================] 43.24K 0s 167ms\n",
      "Clustering step 4\n",
      "[=================================================================] 36.57K 0s 248ms\n",
      "Write merged clustering\n",
      "[=================================================================] 304.62K 0s 310ms\n",
      "Time for merging to component_clusters_redundant: 0h 0m 0s 198ms\n",
      "Time for processing: 0h 0m 0s 861ms\n",
      "cluster process/process_redundant_db process/process_clusters_redundant tmp --min-seq-id 0.3 -c 0.5 --cov-mode 1 \n",
      "\n",
      "MMseqs Version:                     \t14.7e284\n",
      "Substitution matrix                 \taa:blosum62.out,nucl:nucleotide.out\n",
      "Seed substitution matrix            \taa:VTML80.out,nucl:nucleotide.out\n",
      "Sensitivity                         \t4\n",
      "k-mer length                        \t0\n",
      "k-score                             \tseq:2147483647,prof:2147483647\n",
      "Alphabet size                       \taa:21,nucl:5\n",
      "Max sequence length                 \t65535\n",
      "Max results per query               \t20\n",
      "Split database                      \t0\n",
      "Split mode                          \t2\n",
      "Split memory limit                  \t0\n",
      "Coverage threshold                  \t0.5\n",
      "Coverage mode                       \t1\n",
      "Compositional bias                  \t1\n",
      "Compositional bias                  \t1\n",
      "Diagonal scoring                    \ttrue\n",
      "Exact k-mer matching                \t0\n",
      "Mask residues                       \t1\n",
      "Mask residues probability           \t0.9\n",
      "Mask lower case residues            \t0\n",
      "Minimum diagonal score              \t15\n",
      "Selected taxa                       \t\n",
      "Include identical seq. id.          \tfalse\n",
      "Spaced k-mers                       \t1\n",
      "Preload mode                        \t0\n",
      "Pseudo count a                      \tsubstitution:1.100,context:1.400\n",
      "Pseudo count b                      \tsubstitution:4.100,context:5.800\n",
      "Spaced k-mer pattern                \t\n",
      "Local temporary path                \t\n",
      "Threads                             \t10\n",
      "Compressed                          \t0\n",
      "Verbosity                           \t3\n",
      "Add backtrace                       \tfalse\n",
      "Alignment mode                      \t3\n",
      "Alignment mode                      \t0\n",
      "Allow wrapped scoring               \tfalse\n",
      "E-value threshold                   \t0.001\n",
      "Seq. id. threshold                  \t0.3\n",
      "Min alignment length                \t0\n",
      "Seq. id. mode                       \t0\n",
      "Alternative alignments              \t0\n",
      "Max reject                          \t2147483647\n",
      "Max accept                          \t2147483647\n",
      "Score bias                          \t0\n",
      "Realign hits                        \tfalse\n",
      "Realign score bias                  \t-0.2\n",
      "Realign max seqs                    \t2147483647\n",
      "Correlation score weight            \t0\n",
      "Gap open cost                       \taa:11,nucl:5\n",
      "Gap extension cost                  \taa:1,nucl:2\n",
      "Zdrop                               \t40\n",
      "Rescore mode                        \t0\n",
      "Remove hits by seq. id. and coverage\tfalse\n",
      "Sort results                        \t0\n",
      "Cluster mode                        \t0\n",
      "Max connected component depth       \t1000\n",
      "Similarity type                     \t2\n",
      "Single step clustering              \tfalse\n",
      "Cascaded clustering steps           \t3\n",
      "Cluster reassign                    \tfalse\n",
      "Remove temporary files              \tfalse\n",
      "Force restart with latest tmp       \tfalse\n",
      "MPI runner                          \t\n",
      "k-mers per sequence                 \t21\n",
      "Scale k-mers per sequence           \taa:0.000,nucl:0.200\n",
      "Adjust k-mer length                 \tfalse\n",
      "Shift hash                          \t67\n",
      "Include only extendable             \tfalse\n",
      "Skip repeating k-mers               \tfalse\n",
      "\n",
      "Set cluster sensitivity to -s 5.000000\n",
      "Set cluster mode GREEDY MEM\n",
      "Set cluster iterations to 3\n",
      "linclust process/process_redundant_db tmp/16526474843887299237/clu_redundancy tmp/16526474843887299237/linclust --cluster-mode 3 --max-iterations 1000 --similarity-type 2 --threads 10 --compressed 0 -v 3 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' -a 0 --alignment-mode 3 --alignment-output-mode 0 --wrapped-scoring 0 -e 0.001 --min-seq-id 0.3 --min-aln-len 0 --seq-id-mode 0 --alt-ali 0 -c 0.5 --cov-mode 1 --max-seq-len 65535 --comp-bias-corr 1 --comp-bias-corr-scale 1 --max-rejected 2147483647 --max-accept 2147483647 --add-self-matches 0 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --score-bias 0 --realign 0 --realign-score-bias -0.2 --realign-max-seqs 2147483647 --corr-score-weight 0 --gap-open aa:11,nucl:5 --gap-extend aa:1,nucl:2 --zdrop 40 --alph-size aa:13,nucl:5 --kmer-per-seq 21 --spaced-kmer-mode 1 --kmer-per-seq-scale aa:0.000,nucl:0.200 --adjust-kmer-len 0 --mask 0 --mask-prob 0.9 --mask-lower-case 0 -k 0 --hash-shift 67 --split-memory-limit 0 --include-only-extendable 0 --ignore-multi-kmer 0 --rescore-mode 0 --filter-hits 0 --sort-results 0 --remove-tmp-files 0 --force-reuse 0 \n",
      "\n",
      "kmermatcher process/process_redundant_db tmp/16526474843887299237/linclust/6893248161584287490/pref --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' --alph-size aa:13,nucl:5 --min-seq-id 0.3 --kmer-per-seq 21 --spaced-kmer-mode 1 --kmer-per-seq-scale aa:0.000,nucl:0.200 --adjust-kmer-len 0 --mask 0 --mask-prob 0.9 --mask-lower-case 0 --cov-mode 1 -k 0 -c 0.5 --max-seq-len 65535 --hash-shift 67 --split-memory-limit 0 --include-only-extendable 0 --ignore-multi-kmer 0 --threads 10 --compressed 0 -v 3 \n",
      "\n",
      "kmermatcher process/process_redundant_db tmp/16526474843887299237/linclust/6893248161584287490/pref --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' --alph-size aa:13,nucl:5 --min-seq-id 0.3 --kmer-per-seq 21 --spaced-kmer-mode 1 --kmer-per-seq-scale aa:0.000,nucl:0.200 --adjust-kmer-len 0 --mask 0 --mask-prob 0.9 --mask-lower-case 0 --cov-mode 1 -k 0 -c 0.5 --max-seq-len 65535 --hash-shift 67 --split-memory-limit 0 --include-only-extendable 0 --ignore-multi-kmer 0 --threads 10 --compressed 0 -v 3 \n",
      "\n",
      "Database size: 279080 type: Aminoacid\n",
      "Reduced amino acid alphabet: (A S T) (C) (D B N) (E Q Z) (F Y) (G) (H) (I V) (K R) (L J M) (P) (W) (X) \n",
      "\n",
      "Generate k-mers list for 1 split\n",
      "[=================================================================] 279.08K 0s 515ms\n",
      "Sort kmer 0h 0m 0s 149ms\n",
      "Sort by rep. sequence 0h 0m 0s 34ms\n",
      "Time for fill: 0h 0m 0s 74ms\n",
      "Time for merging to pref: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 942ms\n",
      "rescorediagonal process/process_redundant_db process/process_redundant_db tmp/16526474843887299237/linclust/6893248161584287490/pref tmp/16526474843887299237/linclust/6893248161584287490/pref_rescore1 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' --rescore-mode 0 --wrapped-scoring 0 --filter-hits 0 -e 0.001 -c 0.5 -a 0 --cov-mode 1 --min-seq-id 0.5 --min-aln-len 0 --seq-id-mode 0 --add-self-matches 0 --sort-results 0 --db-load-mode 0 --threads 10 --compressed 0 -v 3 \n",
      "\n",
      "[=================================================================] 279.08K 0s 104ms\n",
      "Time for merging to pref_rescore1: 0h 0m 0s 171ms\n",
      "Time for processing: 0h 0m 0s 377ms\n",
      "clust process/process_redundant_db tmp/16526474843887299237/linclust/6893248161584287490/pref_rescore1 tmp/16526474843887299237/linclust/6893248161584287490/pre_clust --cluster-mode 3 --max-iterations 1000 --similarity-type 2 --threads 10 --compressed 0 -v 3 \n",
      "\n",
      "Clustering mode: Greedy Low Mem\n",
      "Total time: 0h 0m 0s 29ms\n",
      "\n",
      "Size of the sequence database: 279080\n",
      "Size of the alignment database: 279080\n",
      "Number of clusters: 190102\n",
      "\n",
      "Writing results 0h 0m 0s 49ms\n",
      "Time for merging to pre_clust: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 108ms\n",
      "createsubdb tmp/16526474843887299237/linclust/6893248161584287490/order_redundancy process/process_redundant_db tmp/16526474843887299237/linclust/6893248161584287490/input_step_redundancy -v 3 --subdb-mode 1 \n",
      "\n",
      "Time for merging to input_step_redundancy: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 66ms\n",
      "createsubdb tmp/16526474843887299237/linclust/6893248161584287490/order_redundancy tmp/16526474843887299237/linclust/6893248161584287490/pref tmp/16526474843887299237/linclust/6893248161584287490/pref_filter1 -v 3 --subdb-mode 1 \n",
      "\n",
      "Time for merging to pref_filter1: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 70ms\n",
      "filterdb tmp/16526474843887299237/linclust/6893248161584287490/pref_filter1 tmp/16526474843887299237/linclust/6893248161584287490/pref_filter2 --filter-file tmp/16526474843887299237/linclust/6893248161584287490/order_redundancy --threads 10 --compressed 0 -v 3 \n",
      "\n",
      "Filtering using file(s)\n",
      "[=================================================================] 190.10K 0s 49ms\n",
      "Time for merging to pref_filter2: 0h 0m 0s 148ms\n",
      "Time for processing: 0h 0m 0s 313ms\n",
      "rescorediagonal tmp/16526474843887299237/linclust/6893248161584287490/input_step_redundancy tmp/16526474843887299237/linclust/6893248161584287490/input_step_redundancy tmp/16526474843887299237/linclust/6893248161584287490/pref_filter2 tmp/16526474843887299237/linclust/6893248161584287490/pref_rescore2 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' --rescore-mode 1 --wrapped-scoring 0 --filter-hits 1 -e 0.001 -c 0.5 -a 0 --cov-mode 1 --min-seq-id 0.3 --min-aln-len 0 --seq-id-mode 0 --add-self-matches 0 --sort-results 0 --db-load-mode 0 --threads 10 --compressed 0 -v 3 \n",
      "\n",
      "[=================================================================] 190.10K 0s 93ms\n",
      "Time for merging to pref_rescore2: 0h 0m 0s 139ms\n",
      "Time for processing: 0h 0m 0s 310ms\n",
      "align tmp/16526474843887299237/linclust/6893248161584287490/input_step_redundancy tmp/16526474843887299237/linclust/6893248161584287490/input_step_redundancy tmp/16526474843887299237/linclust/6893248161584287490/pref_rescore2 tmp/16526474843887299237/linclust/6893248161584287490/aln --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' -a 0 --alignment-mode 3 --alignment-output-mode 0 --wrapped-scoring 0 -e 0.001 --min-seq-id 0.3 --min-aln-len 0 --seq-id-mode 0 --alt-ali 0 -c 0.5 --cov-mode 1 --max-seq-len 65535 --comp-bias-corr 1 --comp-bias-corr-scale 1 --max-rejected 2147483647 --max-accept 2147483647 --add-self-matches 0 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --score-bias 0 --realign 0 --realign-score-bias -0.2 --realign-max-seqs 2147483647 --corr-score-weight 0 --gap-open aa:11,nucl:5 --gap-extend aa:1,nucl:2 --zdrop 40 --threads 10 --compressed 0 -v 3 \n",
      "\n",
      "Compute score, coverage and sequence identity\n",
      "Query database size: 190102 type: Aminoacid\n",
      "Target database size: 190102 type: Aminoacid\n",
      "Calculation of alignments\n",
      "[=================================================================] 190.10K 28s 592ms\n",
      "Time for merging to aln: 0h 0m 0s 148ms\n",
      "440484 alignments calculated\n",
      "404630 sequence pairs passed the thresholds (0.918603 of overall calculated)\n",
      "2.128489 hits per query sequence\n",
      "Time for processing: 0h 0m 28s 886ms\n",
      "clust tmp/16526474843887299237/linclust/6893248161584287490/input_step_redundancy tmp/16526474843887299237/linclust/6893248161584287490/aln tmp/16526474843887299237/linclust/6893248161584287490/clust --cluster-mode 3 --max-iterations 1000 --similarity-type 2 --threads 10 --compressed 0 -v 3 \n",
      "\n",
      "Clustering mode: Greedy Low Mem\n",
      "Total time: 0h 0m 0s 26ms\n",
      "\n",
      "Size of the sequence database: 190102\n",
      "Size of the alignment database: 190102\n",
      "Number of clusters: 104262\n",
      "\n",
      "Writing results 0h 0m 0s 29ms\n",
      "Time for merging to clust: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 80ms\n",
      "mergeclusters process/process_redundant_db tmp/16526474843887299237/clu_redundancy tmp/16526474843887299237/linclust/6893248161584287490/pre_clust tmp/16526474843887299237/linclust/6893248161584287490/clust --threads 10 --compressed 0 -v 3 \n",
      "\n",
      "Clustering step 1\n",
      "[=================================================================] 190.10K 0s 18ms\n",
      "Clustering step 2\n",
      "[=================================================================] 104.26K 0s 34ms\n",
      "Write merged clustering\n",
      "[=================================================================] 279.08K 0s 57ms\n",
      "Time for merging to clu_redundancy: 0h 0m 0s 35ms\n",
      "Time for processing: 0h 0m 0s 144ms\n",
      "createsubdb tmp/16526474843887299237/clu_redundancy process/process_redundant_db tmp/16526474843887299237/input_step_redundancy -v 3 --subdb-mode 1 \n",
      "\n",
      "Time for merging to input_step_redundancy: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 44ms\n",
      "prefilter tmp/16526474843887299237/input_step_redundancy tmp/16526474843887299237/input_step_redundancy tmp/16526474843887299237/pref_step0 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' --seed-sub-mat 'aa:VTML80.out,nucl:nucleotide.out' -s 1 -k 0 --k-score seq:2147483647,prof:2147483647 --alph-size aa:21,nucl:5 --max-seq-len 65535 --max-seqs 20 --split 0 --split-mode 2 --split-memory-limit 0 -c 0.5 --cov-mode 1 --comp-bias-corr 0 --comp-bias-corr-scale 1 --diag-score 0 --exact-kmer-matching 0 --mask 1 --mask-prob 0.9 --mask-lower-case 0 --min-ungapped-score 0 --add-self-matches 0 --spaced-kmer-mode 1 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --threads 10 --compressed 0 -v 3 \n",
      "\n",
      "Query database size: 104262 type: Aminoacid\n",
      "Estimated memory consumption: 1G\n",
      "Target database size: 104262 type: Aminoacid\n",
      "Index table k-mer threshold: 154 at k-mer size 6 \n",
      "Index table: counting k-mers\n",
      "[=================================================================] 104.26K 0s 794ms\n",
      "Index table: Masked residues: 1185923\n",
      "Index table: fill\n",
      "[=================================================================] 104.26K 0s 559ms\n",
      "Index statistics\n",
      "Entries:          22555911\n",
      "DB size:          617 MB\n",
      "Avg k-mer size:   0.352436\n",
      "Top 10 k-mers\n",
      "    FAGTGK\t750\n",
      "    AYQGKT\t703\n",
      "    GQGGYT\t304\n",
      "    YGTSTY\t276\n",
      "    HVYDLT\t195\n",
      "    HIYDLT\t177\n",
      "    LGGFVY\t171\n",
      "    NTFNLQ\t140\n",
      "    PGGTAR\t137\n",
      "    EKDWGV\t133\n",
      "Time for index table init: 0h 0m 1s 661ms\n",
      "Process prefiltering step 1 of 1\n",
      "\n",
      "k-mer similarity threshold: 154\n",
      "Starting prefiltering scores calculation (step 1 of 1)\n",
      "Query db start 1 to 104262\n",
      "Target db start 1 to 104262\n",
      "[=================================================================] 104.26K 1s 784ms\n",
      "\n",
      "2.026928 k-mers per position\n",
      "795 DB matches per sequence\n",
      "0 overflows\n",
      "0 queries produce too many hits (truncated result)\n",
      "10 sequences passed prefiltering per query sequence\n",
      "9 median result list length\n",
      "0 sequences with 0 size result lists\n",
      "Time for merging to pref_step0: 0h 0m 0s 33ms\n",
      "Time for processing: 0h 0m 3s 851ms\n",
      "align tmp/16526474843887299237/input_step_redundancy tmp/16526474843887299237/input_step_redundancy tmp/16526474843887299237/pref_step0 tmp/16526474843887299237/aln_step0 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' -a 0 --alignment-mode 3 --alignment-output-mode 0 --wrapped-scoring 0 -e 0.001 --min-seq-id 0.3 --min-aln-len 0 --seq-id-mode 0 --alt-ali 0 -c 0.5 --cov-mode 1 --max-seq-len 65535 --comp-bias-corr 0 --comp-bias-corr-scale 1 --max-rejected 2147483647 --max-accept 2147483647 --add-self-matches 0 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --score-bias 0 --realign 0 --realign-score-bias -0.2 --realign-max-seqs 2147483647 --corr-score-weight 0 --gap-open aa:11,nucl:5 --gap-extend aa:1,nucl:2 --zdrop 40 --threads 10 --compressed 0 -v 3 \n",
      "\n",
      "Compute score, coverage and sequence identity\n",
      "Query database size: 104262 type: Aminoacid\n",
      "Target database size: 104262 type: Aminoacid\n",
      "Calculation of alignments\n",
      "[=================================================================] 104.26K 1m 34s 591ms\n",
      "Time for merging to aln_step0: 0h 0m 0s 229ms\n",
      "982420 alignments calculated\n",
      "583679 sequence pairs passed the thresholds (0.594124 of overall calculated)\n",
      "5.598195 hits per query sequence\n",
      "Time for processing: 0h 1m 34s 940ms\n",
      "clust tmp/16526474843887299237/input_step_redundancy tmp/16526474843887299237/aln_step0 tmp/16526474843887299237/clu_step0 --cluster-mode 3 --max-iterations 1000 --similarity-type 2 --threads 10 --compressed 0 -v 3 \n",
      "\n",
      "Clustering mode: Greedy Low Mem\n",
      "Total time: 0h 0m 0s 273ms\n",
      "\n",
      "Size of the sequence database: 104262\n",
      "Size of the alignment database: 104262\n",
      "Number of clusters: 52791\n",
      "\n",
      "Writing results 0h 0m 0s 14ms\n",
      "Time for merging to clu_step0: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 525ms\n",
      "createsubdb tmp/16526474843887299237/clu_step0 tmp/16526474843887299237/input_step_redundancy tmp/16526474843887299237/input_step1 -v 3 --subdb-mode 1 \n",
      "\n",
      "Time for merging to input_step1: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 22ms\n",
      "prefilter tmp/16526474843887299237/input_step1 tmp/16526474843887299237/input_step1 tmp/16526474843887299237/pref_step1 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' --seed-sub-mat 'aa:VTML80.out,nucl:nucleotide.out' -s 3 -k 0 --k-score seq:2147483647,prof:2147483647 --alph-size aa:21,nucl:5 --max-seq-len 65535 --max-seqs 20 --split 0 --split-mode 2 --split-memory-limit 0 -c 0.5 --cov-mode 1 --comp-bias-corr 1 --comp-bias-corr-scale 1 --diag-score 1 --exact-kmer-matching 0 --mask 1 --mask-prob 0.9 --mask-lower-case 0 --min-ungapped-score 15 --add-self-matches 0 --spaced-kmer-mode 1 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --threads 10 --compressed 0 -v 3 \n",
      "\n",
      "Query database size: 52791 type: Aminoacid\n",
      "Estimated memory consumption: 1G\n",
      "Target database size: 52791 type: Aminoacid\n",
      "Index table k-mer threshold: 136 at k-mer size 6 \n",
      "Index table: counting k-mers\n",
      "[=================================================================] 52.79K 0s 481ms\n",
      "Index table: Masked residues: 765478\n",
      "Index table: fill\n",
      "[=================================================================] 52.79K 0s 498ms\n",
      "Index statistics\n",
      "Entries:          24758851\n",
      "DB size:          629 MB\n",
      "Avg k-mer size:   0.386857\n",
      "Top 10 k-mers\n",
      "    FAGTGK\t282\n",
      "    AYQGKT\t269\n",
      "    QTSKTM\t218\n",
      "    VPRSTR\t152\n",
      "    KLLDGS\t139\n",
      "    GQGGYT\t134\n",
      "    GLNGCY\t121\n",
      "    YGTSTY\t117\n",
      "    LGGFVY\t84\n",
      "    LASRKT\t68\n",
      "Time for index table init: 0h 0m 1s 322ms\n",
      "Process prefiltering step 1 of 1\n",
      "\n",
      "k-mer similarity threshold: 136\n",
      "Starting prefiltering scores calculation (step 1 of 1)\n",
      "Query db start 1 to 52791\n",
      "Target db start 1 to 52791\n",
      "[=================================================================] 52.79K 4s 456ms\n",
      "\n",
      "20.052189 k-mers per position\n",
      "3378 DB matches per sequence\n",
      "0 overflows\n",
      "0 queries produce too many hits (truncated result)\n",
      "15 sequences passed prefiltering per query sequence\n",
      "20 median result list length\n",
      "0 sequences with 0 size result lists\n",
      "Time for merging to pref_step1: 0h 0m 0s 38ms\n",
      "Time for processing: 0h 0m 6s 159ms\n",
      "align tmp/16526474843887299237/input_step1 tmp/16526474843887299237/input_step1 tmp/16526474843887299237/pref_step1 tmp/16526474843887299237/aln_step1 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' -a 0 --alignment-mode 3 --alignment-output-mode 0 --wrapped-scoring 0 -e 0.001 --min-seq-id 0.3 --min-aln-len 0 --seq-id-mode 0 --alt-ali 0 -c 0.5 --cov-mode 1 --max-seq-len 65535 --comp-bias-corr 1 --comp-bias-corr-scale 1 --max-rejected 2147483647 --max-accept 2147483647 --add-self-matches 0 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --score-bias 0 --realign 0 --realign-score-bias -0.2 --realign-max-seqs 2147483647 --corr-score-weight 0 --gap-open aa:11,nucl:5 --gap-extend aa:1,nucl:2 --zdrop 40 --threads 10 --compressed 0 -v 3 \n",
      "\n",
      "Compute score, coverage and sequence identity\n",
      "Query database size: 52791 type: Aminoacid\n",
      "Target database size: 52791 type: Aminoacid\n",
      "Calculation of alignments\n",
      "[=================================================================] 52.79K 33s 14ms\n",
      "Time for merging to aln_step1: 0h 0m 0s 121ms\n",
      "666817 alignments calculated\n",
      "143032 sequence pairs passed the thresholds (0.214500 of overall calculated)\n",
      "2.709401 hits per query sequence\n",
      "Time for processing: 0h 0m 33s 184ms\n",
      "clust tmp/16526474843887299237/input_step1 tmp/16526474843887299237/aln_step1 tmp/16526474843887299237/clu_step1 --cluster-mode 3 --max-iterations 1000 --similarity-type 2 --threads 10 --compressed 0 -v 3 \n",
      "\n",
      "Clustering mode: Greedy Low Mem\n",
      "Total time: 0h 0m 0s 9ms\n",
      "\n",
      "Size of the sequence database: 52791\n",
      "Size of the alignment database: 52791\n",
      "Number of clusters: 35835\n",
      "\n",
      "Writing results 0h 0m 0s 9ms\n",
      "Time for merging to clu_step1: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 93ms\n",
      "createsubdb tmp/16526474843887299237/clu_step1 tmp/16526474843887299237/input_step1 tmp/16526474843887299237/input_step2 -v 3 --subdb-mode 1 \n",
      "\n",
      "Time for merging to input_step2: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 16ms\n",
      "prefilter tmp/16526474843887299237/input_step2 tmp/16526474843887299237/input_step2 tmp/16526474843887299237/pref_step2 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' --seed-sub-mat 'aa:VTML80.out,nucl:nucleotide.out' -s 5 -k 0 --k-score seq:2147483647,prof:2147483647 --alph-size aa:21,nucl:5 --max-seq-len 65535 --max-seqs 20 --split 0 --split-mode 2 --split-memory-limit 0 -c 0.5 --cov-mode 1 --comp-bias-corr 1 --comp-bias-corr-scale 1 --diag-score 1 --exact-kmer-matching 0 --mask 1 --mask-prob 0.9 --mask-lower-case 0 --min-ungapped-score 15 --add-self-matches 0 --spaced-kmer-mode 1 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --threads 10 --compressed 0 -v 3 \n",
      "\n",
      "Query database size: 35835 type: Aminoacid\n",
      "Estimated memory consumption: 1G\n",
      "Target database size: 35835 type: Aminoacid\n",
      "Index table k-mer threshold: 118 at k-mer size 6 \n",
      "Index table: counting k-mers\n",
      "[=================================================================] 35.83K 0s 362ms\n",
      "Index table: Masked residues: 620035\n",
      "Index table: fill\n",
      "[=================================================================] 35.83K 0s 401ms\n",
      "Index statistics\n",
      "Entries:          19328955\n",
      "DB size:          598 MB\n",
      "Avg k-mer size:   0.302015\n",
      "Top 10 k-mers\n",
      "    FAGTGK\t203\n",
      "    AYQGKT\t201\n",
      "    QTSKTM\t153\n",
      "    VPRSTR\t108\n",
      "    GQGGYT\t101\n",
      "    GLNGCY\t91\n",
      "    YGTSTY\t83\n",
      "    AAAAAS\t66\n",
      "    AAAAAV\t65\n",
      "    LLLLLL\t58\n",
      "Time for index table init: 0h 0m 1s 78ms\n",
      "Process prefiltering step 1 of 1\n",
      "\n",
      "k-mer similarity threshold: 118\n",
      "Starting prefiltering scores calculation (step 1 of 1)\n",
      "Query db start 1 to 35835\n",
      "Target db start 1 to 35835\n",
      "[=================================================================] 35.83K 15s 18ms\n",
      "\n",
      "149.262353 k-mers per position\n",
      "21501 DB matches per sequence\n",
      "0 overflows\n",
      "0 queries produce too many hits (truncated result)\n",
      "19 sequences passed prefiltering per query sequence\n",
      "20 median result list length\n",
      "0 sequences with 0 size result lists\n",
      "Time for merging to pref_step2: 0h 0m 0s 44ms\n",
      "Time for processing: 0h 0m 16s 484ms\n",
      "align tmp/16526474843887299237/input_step2 tmp/16526474843887299237/input_step2 tmp/16526474843887299237/pref_step2 tmp/16526474843887299237/aln_step2 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' -a 0 --alignment-mode 3 --alignment-output-mode 0 --wrapped-scoring 0 -e 0.001 --min-seq-id 0.3 --min-aln-len 0 --seq-id-mode 0 --alt-ali 0 -c 0.5 --cov-mode 1 --max-seq-len 65535 --comp-bias-corr 1 --comp-bias-corr-scale 1 --max-rejected 2147483647 --max-accept 2147483647 --add-self-matches 0 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --score-bias 0 --realign 0 --realign-score-bias -0.2 --realign-max-seqs 2147483647 --corr-score-weight 0 --gap-open aa:11,nucl:5 --gap-extend aa:1,nucl:2 --zdrop 40 --threads 10 --compressed 0 -v 3 \n",
      "\n",
      "Compute score, coverage and sequence identity\n",
      "Query database size: 35835 type: Aminoacid\n",
      "Target database size: 35835 type: Aminoacid\n",
      "Calculation of alignments\n",
      "[=================================================================] 35.83K 19s 674ms\n",
      "Time for merging to aln_step2: 0h 0m 0s 220ms\n",
      "527415 alignments calculated\n",
      "55482 sequence pairs passed the thresholds (0.105196 of overall calculated)\n",
      "1.548263 hits per query sequence\n",
      "Time for processing: 0h 0m 20s 28ms\n",
      "clust tmp/16526474843887299237/input_step2 tmp/16526474843887299237/aln_step2 tmp/16526474843887299237/clu_step2 --cluster-mode 3 --max-iterations 1000 --similarity-type 2 --threads 10 --compressed 0 -v 3 \n",
      "\n",
      "Clustering mode: Greedy Low Mem\n",
      "Total time: 0h 0m 0s 131ms\n",
      "\n",
      "Size of the sequence database: 35835\n",
      "Size of the alignment database: 35835\n",
      "Number of clusters: 30239\n",
      "\n",
      "Writing results 0h 0m 0s 7ms\n",
      "Time for merging to clu_step2: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 168ms\n",
      "mergeclusters process/process_redundant_db process/process_clusters_redundant tmp/16526474843887299237/clu_redundancy tmp/16526474843887299237/clu_step0 tmp/16526474843887299237/clu_step1 tmp/16526474843887299237/clu_step2 --threads 10 --compressed 0 -v 3 \n",
      "\n",
      "Clustering step 1\n",
      "[=================================================================] 104.26K 0s 12ms\n",
      "Clustering step 2\n",
      "[=================================================================] 52.79K 0s 22ms\n",
      "Clustering step 3\n",
      "[=================================================================] 35.83K 0s 27ms\n",
      "Clustering step 4\n",
      "[=================================================================] 30.24K 0s 32ms\n",
      "Write merged clustering\n",
      "[=================================================================] 279.08K 0s 56ms\n",
      "Time for merging to process_clusters_redundant: 0h 0m 0s 11ms\n",
      "Time for processing: 0h 0m 0s 110ms\n",
      "Creating TSV files...\n",
      "createtsv function/function_redundant_db function/function_redundant_db function/function_clusters_redundant function/function_clusters_redundant.tsv \n",
      "\n",
      "MMseqs Version:                 \t14.7e284\n",
      "First sequence as representative\tfalse\n",
      "Target column                   \t1\n",
      "Add full header                 \tfalse\n",
      "Sequence source                 \t0\n",
      "Database output                 \tfalse\n",
      "Threads                         \t10\n",
      "Compressed                      \t0\n",
      "Verbosity                       \t3\n",
      "\n",
      "Time for merging to function_clusters_redundant.tsv: 0h 0m 0s 34ms\n",
      "Time for processing: 0h 0m 0s 90ms\n",
      "createtsv component/component_redundant_db component/component_redundant_db component/component_clusters_redundant component/component_clusters_redundant.tsv \n",
      "\n",
      "MMseqs Version:                 \t14.7e284\n",
      "First sequence as representative\tfalse\n",
      "Target column                   \t1\n",
      "Add full header                 \tfalse\n",
      "Sequence source                 \t0\n",
      "Database output                 \tfalse\n",
      "Threads                         \t10\n",
      "Compressed                      \t0\n",
      "Verbosity                       \t3\n",
      "\n",
      "Time for merging to component_clusters_redundant.tsv: 0h 0m 0s 16ms\n",
      "Time for processing: 0h 0m 0s 94ms\n",
      "createtsv process/process_redundant_db process/process_redundant_db process/process_clusters_redundant process/process_clusters_redundant.tsv \n",
      "\n",
      "MMseqs Version:                 \t14.7e284\n",
      "First sequence as representative\tfalse\n",
      "Target column                   \t1\n",
      "Add full header                 \tfalse\n",
      "Sequence source                 \t0\n",
      "Database output                 \tfalse\n",
      "Threads                         \t10\n",
      "Compressed                      \t0\n",
      "Verbosity                       \t3\n",
      "\n",
      "Time for merging to process_clusters_redundant.tsv: 0h 0m 0s 15ms\n",
      "Time for processing: 0h 0m 0s 64ms\n",
      "All tasks completed successfully.\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "CompletedProcess(args=['./mmseq_cluster.sh', '/Users/harsh/Documents/PFP_PUBLISH/raw_data_from_uniprot/reduced_90_similarity_30', '/Users/harsh/Documents/PFP_PUBLISH/raw_data_from_uniprot/reduced_90_function_propagated.fasta', '/Users/harsh/Documents/PFP_PUBLISH/raw_data_from_uniprot/reduced_90_component_propagated.fasta', '/Users/harsh/Documents/PFP_PUBLISH/raw_data_from_uniprot/reduced_90_process_propagated.fasta', '0.3'], returncode=0)"
      ]
     },
     "execution_count": 83,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#subprocess.run([\"./mmseq_cluster.sh\", \"/Users/harsh/Documents/PFP_PUBLISH/raw_data_from_uniprot/reduced_90\", \"/Users/harsh/Documents/PFP_PUBLISH/raw_data_from_uniprot/total_SAR_function_for_reduce.fasta\", \"/Users/harsh/Documents/PFP_PUBLISH/raw_data_from_uniprot/total_SAR_component_for_reduce.fasta\", \"/Users/harsh/Documents/PFP_PUBLISH/raw_data_from_uniprot/total_SAR_process_for_reduce.fasta\", \"0.9\"])\n",
    "\n",
    "subprocess.run([\"./mmseq_cluster.sh\", \"reduced_90_similarity_30\", \"reduced_90_function_propagated.fasta\", \"reduced_90_component_propagated.fasta\", \"reduced_90_process_propagated.fasta\", \"0.3\"])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 84,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Creating databases...\n",
      "createdb /Users/harsh/Documents/PFP_PUBLISH/raw_data_from_uniprot/reduced_90_function_propagated.fasta function/function_redundant_db \n",
      "\n",
      "MMseqs Version:       \t14.7e284\n",
      "Database type         \t0\n",
      "Shuffle input database\ttrue\n",
      "Createdb mode         \t0\n",
      "Write lookup file     \t1\n",
      "Offset of numeric ids \t0\n",
      "Compressed            \t0\n",
      "Verbosity             \t3\n",
      "\n",
      "Converting sequences\n",
      "[===================================\n",
      "Time for merging to function_redundant_db_h: 0h 0m 0s 64ms\n",
      "Time for merging to function_redundant_db: 0h 0m 0s 138ms\n",
      "Database type: Aminoacid\n",
      "Time for processing: 0h 0m 1s 72ms\n",
      "createdb /Users/harsh/Documents/PFP_PUBLISH/raw_data_from_uniprot/reduced_90_component_propagated.fasta component/component_redundant_db \n",
      "\n",
      "MMseqs Version:       \t14.7e284\n",
      "Database type         \t0\n",
      "Shuffle input database\ttrue\n",
      "Createdb mode         \t0\n",
      "Write lookup file     \t1\n",
      "Offset of numeric ids \t0\n",
      "Compressed            \t0\n",
      "Verbosity             \t3\n",
      "\n",
      "Converting sequences\n",
      "[==============================\n",
      "Time for merging to component_redundant_db_h: 0h 0m 0s 58ms\n",
      "Time for merging to component_redundant_db: 0h 0m 0s 123ms\n",
      "Database type: Aminoacid\n",
      "Time for processing: 0h 0m 0s 920ms\n",
      "createdb /Users/harsh/Documents/PFP_PUBLISH/raw_data_from_uniprot/reduced_90_process_propagated.fasta process/process_redundant_db \n",
      "\n",
      "MMseqs Version:       \t14.7e284\n",
      "Database type         \t0\n",
      "Shuffle input database\ttrue\n",
      "Createdb mode         \t0\n",
      "Write lookup file     \t1\n",
      "Offset of numeric ids \t0\n",
      "Compressed            \t0\n",
      "Verbosity             \t3\n",
      "\n",
      "Converting sequences\n",
      "[===========================\n",
      "Time for merging to process_redundant_db_h: 0h 0m 0s 55ms\n",
      "Time for merging to process_redundant_db: 0h 0m 0s 119ms\n",
      "Database type: Aminoacid\n",
      "Time for processing: 0h 0m 0s 881ms\n",
      "Clustering sequences...\n",
      "Create directory tmp\n",
      "cluster function/function_redundant_db function/function_clusters_redundant tmp --min-seq-id 0.1 -c 0.5 --cov-mode 1 \n",
      "\n",
      "MMseqs Version:                     \t14.7e284\n",
      "Substitution matrix                 \taa:blosum62.out,nucl:nucleotide.out\n",
      "Seed substitution matrix            \taa:VTML80.out,nucl:nucleotide.out\n",
      "Sensitivity                         \t4\n",
      "k-mer length                        \t0\n",
      "k-score                             \tseq:2147483647,prof:2147483647\n",
      "Alphabet size                       \taa:21,nucl:5\n",
      "Max sequence length                 \t65535\n",
      "Max results per query               \t20\n",
      "Split database                      \t0\n",
      "Split mode                          \t2\n",
      "Split memory limit                  \t0\n",
      "Coverage threshold                  \t0.5\n",
      "Coverage mode                       \t1\n",
      "Compositional bias                  \t1\n",
      "Compositional bias                  \t1\n",
      "Diagonal scoring                    \ttrue\n",
      "Exact k-mer matching                \t0\n",
      "Mask residues                       \t1\n",
      "Mask residues probability           \t0.9\n",
      "Mask lower case residues            \t0\n",
      "Minimum diagonal score              \t15\n",
      "Selected taxa                       \t\n",
      "Include identical seq. id.          \tfalse\n",
      "Spaced k-mers                       \t1\n",
      "Preload mode                        \t0\n",
      "Pseudo count a                      \tsubstitution:1.100,context:1.400\n",
      "Pseudo count b                      \tsubstitution:4.100,context:5.800\n",
      "Spaced k-mer pattern                \t\n",
      "Local temporary path                \t\n",
      "Threads                             \t10\n",
      "Compressed                          \t0\n",
      "Verbosity                           \t3\n",
      "Add backtrace                       \tfalse\n",
      "Alignment mode                      \t3\n",
      "Alignment mode                      \t0\n",
      "Allow wrapped scoring               \tfalse\n",
      "E-value threshold                   \t0.001\n",
      "Seq. id. threshold                  \t0.1\n",
      "Min alignment length                \t0\n",
      "Seq. id. mode                       \t0\n",
      "Alternative alignments              \t0\n",
      "Max reject                          \t2147483647\n",
      "Max accept                          \t2147483647\n",
      "Score bias                          \t0\n",
      "Realign hits                        \tfalse\n",
      "Realign score bias                  \t-0.2\n",
      "Realign max seqs                    \t2147483647\n",
      "Correlation score weight            \t0\n",
      "Gap open cost                       \taa:11,nucl:5\n",
      "Gap extension cost                  \taa:1,nucl:2\n",
      "Zdrop                               \t40\n",
      "Rescore mode                        \t0\n",
      "Remove hits by seq. id. and coverage\tfalse\n",
      "Sort results                        \t0\n",
      "Cluster mode                        \t0\n",
      "Max connected component depth       \t1000\n",
      "Similarity type                     \t2\n",
      "Single step clustering              \tfalse\n",
      "Cascaded clustering steps           \t3\n",
      "Cluster reassign                    \tfalse\n",
      "Remove temporary files              \tfalse\n",
      "Force restart with latest tmp       \tfalse\n",
      "MPI runner                          \t\n",
      "k-mers per sequence                 \t21\n",
      "Scale k-mers per sequence           \taa:0.000,nucl:0.200\n",
      "Adjust k-mer length                 \tfalse\n",
      "Shift hash                          \t67\n",
      "Include only extendable             \tfalse\n",
      "Skip repeating k-mers               \tfalse\n",
      "\n",
      "Set cluster sensitivity to -s 6.000000\n",
      "Set cluster mode GREEDY MEM\n",
      "Set cluster iterations to 3\n",
      "linclust function/function_redundant_db tmp/4508048661228596920/clu_redundancy tmp/4508048661228596920/linclust --cluster-mode 3 --max-iterations 1000 --similarity-type 2 --threads 10 --compressed 0 -v 3 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' -a 0 --alignment-mode 3 --alignment-output-mode 0 --wrapped-scoring 0 -e 0.001 --min-seq-id 0.1 --min-aln-len 0 --seq-id-mode 0 --alt-ali 0 -c 0.5 --cov-mode 1 --max-seq-len 65535 --comp-bias-corr 1 --comp-bias-corr-scale 1 --max-rejected 2147483647 --max-accept 2147483647 --add-self-matches 0 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --score-bias 0 --realign 0 --realign-score-bias -0.2 --realign-max-seqs 2147483647 --corr-score-weight 0 --gap-open aa:11,nucl:5 --gap-extend aa:1,nucl:2 --zdrop 40 --alph-size aa:13,nucl:5 --kmer-per-seq 21 --spaced-kmer-mode 1 --kmer-per-seq-scale aa:0.000,nucl:0.200 --adjust-kmer-len 0 --mask 0 --mask-prob 0.9 --mask-lower-case 0 -k 0 --hash-shift 67 --split-memory-limit 0 --include-only-extendable 0 --ignore-multi-kmer 0 --rescore-mode 0 --filter-hits 0 --sort-results 0 --remove-tmp-files 0 --force-reuse 0 \n",
      "\n",
      "kmermatcher function/function_redundant_db tmp/4508048661228596920/linclust/2783909104000407252/pref --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' --alph-size aa:13,nucl:5 --min-seq-id 0.1 --kmer-per-seq 21 --spaced-kmer-mode 1 --kmer-per-seq-scale aa:0.000,nucl:0.200 --adjust-kmer-len 0 --mask 0 --mask-prob 0.9 --mask-lower-case 0 --cov-mode 1 -k 0 -c 0.5 --max-seq-len 65535 --hash-shift 67 --split-memory-limit 0 --include-only-extendable 0 --ignore-multi-kmer 0 --threads 10 --compressed 0 -v 3 \n",
      "\n",
      "kmermatcher function/function_redundant_db tmp/4508048661228596920/linclust/2783909104000407252/pref --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' --alph-size aa:13,nucl:5 --min-seq-id 0.1 --kmer-per-seq 21 --spaced-kmer-mode 1 --kmer-per-seq-scale aa:0.000,nucl:0.200 --adjust-kmer-len 0 --mask 0 --mask-prob 0.9 --mask-lower-case 0 --cov-mode 1 -k 0 -c 0.5 --max-seq-len 65535 --hash-shift 67 --split-memory-limit 0 --include-only-extendable 0 --ignore-multi-kmer 0 --threads 10 --compressed 0 -v 3 \n",
      "\n",
      "Database size: 352499 type: Aminoacid\n",
      "Reduced amino acid alphabet: (A S T) (C) (D B N) (E Q Z) (F Y) (G) (H) (I V) (K R) (L J M) (P) (W) (X) \n",
      "\n",
      "Generate k-mers list for 1 split\n",
      "[=================================================================] 352.50K 0s 867ms\n",
      "Sort kmer 0h 0m 0s 147ms\n",
      "Sort by rep. sequence 0h 0m 0s 84ms\n",
      "Time for fill: 0h 0m 0s 95ms\n",
      "Time for merging to pref: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 1s 362ms\n",
      "rescorediagonal function/function_redundant_db function/function_redundant_db tmp/4508048661228596920/linclust/2783909104000407252/pref tmp/4508048661228596920/linclust/2783909104000407252/pref_rescore1 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' --rescore-mode 0 --wrapped-scoring 0 --filter-hits 0 -e 0.001 -c 0.5 -a 0 --cov-mode 1 --min-seq-id 0.5 --min-aln-len 0 --seq-id-mode 0 --add-self-matches 0 --sort-results 0 --db-load-mode 0 --threads 10 --compressed 0 -v 3 \n",
      "\n",
      "[=================================================================] 352.50K 0s 142ms\n",
      "Time for merging to pref_rescore1: 0h 0m 0s 280ms\n",
      "Time for processing: 0h 0m 0s 518ms\n",
      "clust function/function_redundant_db tmp/4508048661228596920/linclust/2783909104000407252/pref_rescore1 tmp/4508048661228596920/linclust/2783909104000407252/pre_clust --cluster-mode 3 --max-iterations 1000 --similarity-type 2 --threads 10 --compressed 0 -v 3 \n",
      "\n",
      "Clustering mode: Greedy Low Mem\n",
      "Total time: 0h 0m 0s 48ms\n",
      "\n",
      "Size of the sequence database: 352499\n",
      "Size of the alignment database: 352499\n",
      "Number of clusters: 240467\n",
      "\n",
      "Writing results 0h 0m 0s 62ms\n",
      "Time for merging to pre_clust: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 149ms\n",
      "createsubdb tmp/4508048661228596920/linclust/2783909104000407252/order_redundancy function/function_redundant_db tmp/4508048661228596920/linclust/2783909104000407252/input_step_redundancy -v 3 --subdb-mode 1 \n",
      "\n",
      "Time for merging to input_step_redundancy: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 83ms\n",
      "createsubdb tmp/4508048661228596920/linclust/2783909104000407252/order_redundancy tmp/4508048661228596920/linclust/2783909104000407252/pref tmp/4508048661228596920/linclust/2783909104000407252/pref_filter1 -v 3 --subdb-mode 1 \n",
      "\n",
      "Time for merging to pref_filter1: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 90ms\n",
      "filterdb tmp/4508048661228596920/linclust/2783909104000407252/pref_filter1 tmp/4508048661228596920/linclust/2783909104000407252/pref_filter2 --filter-file tmp/4508048661228596920/linclust/2783909104000407252/order_redundancy --threads 10 --compressed 0 -v 3 \n",
      "\n",
      "Filtering using file(s)\n",
      "[=================================================================] 240.47K 0s 68ms\n",
      "Time for merging to pref_filter2: 0h 0m 0s 137ms\n",
      "Time for processing: 0h 0m 0s 308ms\n",
      "rescorediagonal tmp/4508048661228596920/linclust/2783909104000407252/input_step_redundancy tmp/4508048661228596920/linclust/2783909104000407252/input_step_redundancy tmp/4508048661228596920/linclust/2783909104000407252/pref_filter2 tmp/4508048661228596920/linclust/2783909104000407252/pref_rescore2 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' --rescore-mode 1 --wrapped-scoring 0 --filter-hits 1 -e 0.001 -c 0.5 -a 0 --cov-mode 1 --min-seq-id 0.1 --min-aln-len 0 --seq-id-mode 0 --add-self-matches 0 --sort-results 0 --db-load-mode 0 --threads 10 --compressed 0 -v 3 \n",
      "\n",
      "Can not find any score per column for coverage 0.500000 and sequence identity 0.100000. No hit will be filtered.\n",
      "[=================================================================] 240.47K 0s 130ms\n",
      "Time for merging to pref_rescore2: 0h 0m 0s 133ms\n",
      "Time for processing: 0h 0m 0s 375ms\n",
      "align tmp/4508048661228596920/linclust/2783909104000407252/input_step_redundancy tmp/4508048661228596920/linclust/2783909104000407252/input_step_redundancy tmp/4508048661228596920/linclust/2783909104000407252/pref_rescore2 tmp/4508048661228596920/linclust/2783909104000407252/aln --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' -a 0 --alignment-mode 3 --alignment-output-mode 0 --wrapped-scoring 0 -e 0.001 --min-seq-id 0.1 --min-aln-len 0 --seq-id-mode 0 --alt-ali 0 -c 0.5 --cov-mode 1 --max-seq-len 65535 --comp-bias-corr 1 --comp-bias-corr-scale 1 --max-rejected 2147483647 --max-accept 2147483647 --add-self-matches 0 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --score-bias 0 --realign 0 --realign-score-bias -0.2 --realign-max-seqs 2147483647 --corr-score-weight 0 --gap-open aa:11,nucl:5 --gap-extend aa:1,nucl:2 --zdrop 40 --threads 10 --compressed 0 -v 3 \n",
      "\n",
      "Compute score, coverage and sequence identity\n",
      "Query database size: 240467 type: Aminoacid\n",
      "Target database size: 240467 type: Aminoacid\n",
      "Calculation of alignments\n",
      "[=================================================================] 240.47K 39s 870ms\n",
      "Time for merging to aln: 0h 0m 0s 266ms\n",
      "699781 alignments calculated\n",
      "549536 sequence pairs passed the thresholds (0.785297 of overall calculated)\n",
      "2.285286 hits per query sequence\n",
      "Time for processing: 0h 0m 40s 282ms\n",
      "clust tmp/4508048661228596920/linclust/2783909104000407252/input_step_redundancy tmp/4508048661228596920/linclust/2783909104000407252/aln tmp/4508048661228596920/linclust/2783909104000407252/clust --cluster-mode 3 --max-iterations 1000 --similarity-type 2 --threads 10 --compressed 0 -v 3 \n",
      "\n",
      "Clustering mode: Greedy Low Mem\n",
      "Total time: 0h 0m 0s 108ms\n",
      "\n",
      "Size of the sequence database: 240467\n",
      "Size of the alignment database: 240467\n",
      "Number of clusters: 125597\n",
      "\n",
      "Writing results 0h 0m 0s 34ms\n",
      "Time for merging to clust: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 204ms\n",
      "mergeclusters function/function_redundant_db tmp/4508048661228596920/clu_redundancy tmp/4508048661228596920/linclust/2783909104000407252/pre_clust tmp/4508048661228596920/linclust/2783909104000407252/clust --threads 10 --compressed 0 -v 3 \n",
      "\n",
      "Clustering step 1\n",
      "[=================================================================] 240.47K 0s 29ms\n",
      "Clustering step 2\n",
      "[=================================================================] 125.60K 0s 97ms\n",
      "Write merged clustering\n",
      "[=================================================================] 352.50K 0s 131ms\n",
      "Time for merging to clu_redundancy: 0h 0m 0s 99ms\n",
      "Time for processing: 0h 0m 0s 326ms\n",
      "createsubdb tmp/4508048661228596920/clu_redundancy function/function_redundant_db tmp/4508048661228596920/input_step_redundancy -v 3 --subdb-mode 1 \n",
      "\n",
      "Time for merging to input_step_redundancy: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 55ms\n",
      "prefilter tmp/4508048661228596920/input_step_redundancy tmp/4508048661228596920/input_step_redundancy tmp/4508048661228596920/pref_step0 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' --seed-sub-mat 'aa:VTML80.out,nucl:nucleotide.out' -s 1 -k 0 --k-score seq:2147483647,prof:2147483647 --alph-size aa:21,nucl:5 --max-seq-len 65535 --max-seqs 20 --split 0 --split-mode 2 --split-memory-limit 0 -c 0.5 --cov-mode 1 --comp-bias-corr 0 --comp-bias-corr-scale 1 --diag-score 0 --exact-kmer-matching 0 --mask 1 --mask-prob 0.9 --mask-lower-case 0 --min-ungapped-score 0 --add-self-matches 0 --spaced-kmer-mode 1 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --threads 10 --compressed 0 -v 3 \n",
      "\n",
      "Query database size: 125597 type: Aminoacid\n",
      "Estimated memory consumption: 1G\n",
      "Target database size: 125597 type: Aminoacid\n",
      "Index table k-mer threshold: 154 at k-mer size 6 \n",
      "Index table: counting k-mers\n",
      "[=================================================================] 125.60K 0s 959ms\n",
      "Index table: Masked residues: 1446711\n",
      "Index table: fill\n",
      "[=================================================================] 125.60K 0s 791ms\n",
      "Index statistics\n",
      "Entries:          27640375\n",
      "DB size:          646 MB\n",
      "Avg k-mer size:   0.431881\n",
      "Top 10 k-mers\n",
      "    FAGTGK\t714\n",
      "    AYQGKT\t654\n",
      "    HRLPIL\t346\n",
      "    GQGGHT\t306\n",
      "    LGGFVY\t299\n",
      "    HRIPIL\t251\n",
      "    IPRSTR\t246\n",
      "    YGTSTY\t246\n",
      "    GQGGFT\t211\n",
      "    VHDKNV\t210\n",
      "Time for index table init: 0h 0m 2s 44ms\n",
      "Process prefiltering step 1 of 1\n",
      "\n",
      "k-mer similarity threshold: 154\n",
      "Starting prefiltering scores calculation (step 1 of 1)\n",
      "Query db start 1 to 125597\n",
      "Target db start 1 to 125597\n",
      "[=================================================================] 125.60K 2s 386ms\n",
      "\n",
      "2.101738 k-mers per position\n",
      "945 DB matches per sequence\n",
      "0 overflows\n",
      "0 queries produce too many hits (truncated result)\n",
      "11 sequences passed prefiltering per query sequence\n",
      "10 median result list length\n",
      "0 sequences with 0 size result lists\n",
      "Time for merging to pref_step0: 0h 0m 0s 43ms\n",
      "Time for processing: 0h 0m 4s 777ms\n",
      "align tmp/4508048661228596920/input_step_redundancy tmp/4508048661228596920/input_step_redundancy tmp/4508048661228596920/pref_step0 tmp/4508048661228596920/aln_step0 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' -a 0 --alignment-mode 3 --alignment-output-mode 0 --wrapped-scoring 0 -e 0.001 --min-seq-id 0.1 --min-aln-len 0 --seq-id-mode 0 --alt-ali 0 -c 0.5 --cov-mode 1 --max-seq-len 65535 --comp-bias-corr 0 --comp-bias-corr-scale 1 --max-rejected 2147483647 --max-accept 2147483647 --add-self-matches 0 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --score-bias 0 --realign 0 --realign-score-bias -0.2 --realign-max-seqs 2147483647 --corr-score-weight 0 --gap-open aa:11,nucl:5 --gap-extend aa:1,nucl:2 --zdrop 40 --threads 10 --compressed 0 -v 3 \n",
      "\n",
      "Compute score, coverage and sequence identity\n",
      "Query database size: 125597 type: Aminoacid\n",
      "Target database size: 125597 type: Aminoacid\n",
      "Calculation of alignments\n",
      "[=================================================================] 125.60K 1m 45s 671ms\n",
      "Time for merging to aln_step0: 0h 0m 0s 263ms\n",
      "1254803 alignments calculated\n",
      "849941 sequence pairs passed the thresholds (0.677350 of overall calculated)\n",
      "6.767208 hits per query sequence\n",
      "Time for processing: 0h 1m 46s 28ms\n",
      "clust tmp/4508048661228596920/input_step_redundancy tmp/4508048661228596920/aln_step0 tmp/4508048661228596920/clu_step0 --cluster-mode 3 --max-iterations 1000 --similarity-type 2 --threads 10 --compressed 0 -v 3 \n",
      "\n",
      "Clustering mode: Greedy Low Mem\n",
      "Total time: 0h 0m 0s 262ms\n",
      "\n",
      "Size of the sequence database: 125597\n",
      "Size of the alignment database: 125597\n",
      "Number of clusters: 58417\n",
      "\n",
      "Writing results 0h 0m 0s 17ms\n",
      "Time for merging to clu_step0: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 520ms\n",
      "createsubdb tmp/4508048661228596920/clu_step0 tmp/4508048661228596920/input_step_redundancy tmp/4508048661228596920/input_step1 -v 3 --subdb-mode 1 \n",
      "\n",
      "Time for merging to input_step1: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 27ms\n",
      "prefilter tmp/4508048661228596920/input_step1 tmp/4508048661228596920/input_step1 tmp/4508048661228596920/pref_step1 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' --seed-sub-mat 'aa:VTML80.out,nucl:nucleotide.out' -s 3.5 -k 0 --k-score seq:2147483647,prof:2147483647 --alph-size aa:21,nucl:5 --max-seq-len 65535 --max-seqs 20 --split 0 --split-mode 2 --split-memory-limit 0 -c 0.5 --cov-mode 1 --comp-bias-corr 1 --comp-bias-corr-scale 1 --diag-score 1 --exact-kmer-matching 0 --mask 1 --mask-prob 0.9 --mask-lower-case 0 --min-ungapped-score 15 --add-self-matches 0 --spaced-kmer-mode 1 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --threads 10 --compressed 0 -v 3 \n",
      "\n",
      "Query database size: 58417 type: Aminoacid\n",
      "Estimated memory consumption: 1G\n",
      "Target database size: 58417 type: Aminoacid\n",
      "Index table k-mer threshold: 131 at k-mer size 6 \n",
      "Index table: counting k-mers\n",
      "[=================================================================] 58.42K 0s 566ms\n",
      "Index table: Masked residues: 926134\n",
      "Index table: fill\n",
      "[=================================================================] 58.42K 0s 592ms\n",
      "Index statistics\n",
      "Entries:          28827102\n",
      "DB size:          653 MB\n",
      "Avg k-mer size:   0.450423\n",
      "Top 10 k-mers\n",
      "    FAGTGK\t227\n",
      "    AYQGKT\t222\n",
      "    QTSKTM\t170\n",
      "    KLRLSL\t133\n",
      "    GPGGTL\t128\n",
      "    VPRSTR\t121\n",
      "    KLLDGS\t114\n",
      "    LGGFVY\t110\n",
      "    SKTLDS\t94\n",
      "    VHDKNV\t94\n",
      "Time for index table init: 0h 0m 1s 509ms\n",
      "Process prefiltering step 1 of 1\n",
      "\n",
      "k-mer similarity threshold: 131\n",
      "Starting prefiltering scores calculation (step 1 of 1)\n",
      "Query db start 1 to 58417\n",
      "Target db start 1 to 58417\n",
      "[=================================================================] 58.42K 8s 284ms\n",
      "\n",
      "36.740094 k-mers per position\n",
      "7072 DB matches per sequence\n",
      "0 overflows\n",
      "0 queries produce too many hits (truncated result)\n",
      "18 sequences passed prefiltering per query sequence\n",
      "20 median result list length\n",
      "0 sequences with 0 size result lists\n",
      "Time for merging to pref_step1: 0h 0m 0s 29ms\n",
      "Time for processing: 0h 0m 10s 105ms\n",
      "align tmp/4508048661228596920/input_step1 tmp/4508048661228596920/input_step1 tmp/4508048661228596920/pref_step1 tmp/4508048661228596920/aln_step1 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' -a 0 --alignment-mode 3 --alignment-output-mode 0 --wrapped-scoring 0 -e 0.001 --min-seq-id 0.1 --min-aln-len 0 --seq-id-mode 0 --alt-ali 0 -c 0.5 --cov-mode 1 --max-seq-len 65535 --comp-bias-corr 1 --comp-bias-corr-scale 1 --max-rejected 2147483647 --max-accept 2147483647 --add-self-matches 0 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --score-bias 0 --realign 0 --realign-score-bias -0.2 --realign-max-seqs 2147483647 --corr-score-weight 0 --gap-open aa:11,nucl:5 --gap-extend aa:1,nucl:2 --zdrop 40 --threads 10 --compressed 0 -v 3 \n",
      "\n",
      "Compute score, coverage and sequence identity\n",
      "Query database size: 58417 type: Aminoacid\n",
      "Target database size: 58417 type: Aminoacid\n",
      "Calculation of alignments\n",
      "[=================================================================] 58.42K 36s 928ms\n",
      "Time for merging to aln_step1: 0h 0m 0s 211ms\n",
      "830220 alignments calculated\n",
      "241064 sequence pairs passed the thresholds (0.290362 of overall calculated)\n",
      "4.126607 hits per query sequence\n",
      "Time for processing: 0h 0m 37s 240ms\n",
      "clust tmp/4508048661228596920/input_step1 tmp/4508048661228596920/aln_step1 tmp/4508048661228596920/clu_step1 --cluster-mode 3 --max-iterations 1000 --similarity-type 2 --threads 10 --compressed 0 -v 3 \n",
      "\n",
      "Clustering mode: Greedy Low Mem\n",
      "Total time: 0h 0m 0s 218ms\n",
      "\n",
      "Size of the sequence database: 58417\n",
      "Size of the alignment database: 58417\n",
      "Number of clusters: 33583\n",
      "\n",
      "Writing results 0h 0m 0s 9ms\n",
      "Time for merging to clu_step1: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 410ms\n",
      "createsubdb tmp/4508048661228596920/clu_step1 tmp/4508048661228596920/input_step1 tmp/4508048661228596920/input_step2 -v 3 --subdb-mode 1 \n",
      "\n",
      "Time for merging to input_step2: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 18ms\n",
      "prefilter tmp/4508048661228596920/input_step2 tmp/4508048661228596920/input_step2 tmp/4508048661228596920/pref_step2 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' --seed-sub-mat 'aa:VTML80.out,nucl:nucleotide.out' -s 6 -k 0 --k-score seq:2147483647,prof:2147483647 --alph-size aa:21,nucl:5 --max-seq-len 65535 --max-seqs 20 --split 0 --split-mode 2 --split-memory-limit 0 -c 0.5 --cov-mode 1 --comp-bias-corr 1 --comp-bias-corr-scale 1 --diag-score 1 --exact-kmer-matching 0 --mask 1 --mask-prob 0.9 --mask-lower-case 0 --min-ungapped-score 15 --add-self-matches 0 --spaced-kmer-mode 1 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --threads 10 --compressed 0 -v 3 \n",
      "\n",
      "Query database size: 33583 type: Aminoacid\n",
      "Estimated memory consumption: 1G\n",
      "Target database size: 33583 type: Aminoacid\n",
      "Index table k-mer threshold: 109 at k-mer size 6 \n",
      "Index table: counting k-mers\n",
      "[=================================================================] 33.58K 0s 409ms\n",
      "Index table: Masked residues: 702668\n",
      "Index table: fill\n",
      "[=================================================================] 33.58K 0s 424ms\n",
      "Index statistics\n",
      "Entries:          18863990\n",
      "DB size:          596 MB\n",
      "Avg k-mer size:   0.294750\n",
      "Top 10 k-mers\n",
      "    FAGTGK\t174\n",
      "    AYQGKT\t168\n",
      "    YRSLLL\t138\n",
      "    QTSKTM\t124\n",
      "    KLRLSL\t101\n",
      "    SSSSSS\t96\n",
      "    GQGGYT\t86\n",
      "    GQGGHT\t72\n",
      "    YGTSTY\t68\n",
      "    AAAAAV\t63\n",
      "Time for index table init: 0h 0m 1s 117ms\n",
      "Process prefiltering step 1 of 1\n",
      "\n",
      "k-mer similarity threshold: 109\n",
      "Starting prefiltering scores calculation (step 1 of 1)\n",
      "Query db start 1 to 33583\n",
      "Target db start 1 to 33583\n",
      "[=================================================================] 33.58K 36s 804ms\n",
      "\n",
      "388.003728 k-mers per position\n",
      "61331 DB matches per sequence\n",
      "0 overflows\n",
      "0 queries produce too many hits (truncated result)\n",
      "19 sequences passed prefiltering per query sequence\n",
      "20 median result list length\n",
      "0 sequences with 0 size result lists\n",
      "Time for merging to pref_step2: 0h 0m 0s 168ms\n",
      "Time for processing: 0h 0m 38s 386ms\n",
      "align tmp/4508048661228596920/input_step2 tmp/4508048661228596920/input_step2 tmp/4508048661228596920/pref_step2 tmp/4508048661228596920/aln_step2 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' -a 0 --alignment-mode 3 --alignment-output-mode 0 --wrapped-scoring 0 -e 0.001 --min-seq-id 0.1 --min-aln-len 0 --seq-id-mode 0 --alt-ali 0 -c 0.5 --cov-mode 1 --max-seq-len 65535 --comp-bias-corr 1 --comp-bias-corr-scale 1 --max-rejected 2147483647 --max-accept 2147483647 --add-self-matches 0 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --score-bias 0 --realign 0 --realign-score-bias -0.2 --realign-max-seqs 2147483647 --corr-score-weight 0 --gap-open aa:11,nucl:5 --gap-extend aa:1,nucl:2 --zdrop 40 --threads 10 --compressed 0 -v 3 \n",
      "\n",
      "Compute score, coverage and sequence identity\n",
      "Query database size: 33583 type: Aminoacid\n",
      "Target database size: 33583 type: Aminoacid\n",
      "Calculation of alignments\n",
      "[=================================================================] 33.58K 15s 623ms\n",
      "Time for merging to aln_step2: 0h 0m 0s 237ms\n",
      "498957 alignments calculated\n",
      "72418 sequence pairs passed the thresholds (0.145139 of overall calculated)\n",
      "2.156389 hits per query sequence\n",
      "Time for processing: 0h 0m 15s 971ms\n",
      "clust tmp/4508048661228596920/input_step2 tmp/4508048661228596920/aln_step2 tmp/4508048661228596920/clu_step2 --cluster-mode 3 --max-iterations 1000 --similarity-type 2 --threads 10 --compressed 0 -v 3 \n",
      "\n",
      "Clustering mode: Greedy Low Mem\n",
      "Total time: 0h 0m 0s 308ms\n",
      "\n",
      "Size of the sequence database: 33583\n",
      "Size of the alignment database: 33583\n",
      "Number of clusters: 25374\n",
      "\n",
      "Writing results 0h 0m 0s 6ms\n",
      "Time for merging to clu_step2: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 545ms\n",
      "mergeclusters function/function_redundant_db function/function_clusters_redundant tmp/4508048661228596920/clu_redundancy tmp/4508048661228596920/clu_step0 tmp/4508048661228596920/clu_step1 tmp/4508048661228596920/clu_step2 --threads 10 --compressed 0 -v 3 \n",
      "\n",
      "Clustering step 1\n",
      "[=================================================================] 125.60K 0s 22ms\n",
      "Clustering step 2\n",
      "[=================================================================] 58.42K 0s 119ms\n",
      "Clustering step 3\n",
      "[=================================================================] 33.58K 0s 202ms\n",
      "Clustering step 4\n",
      "[=================================================================] 25.37K 0s 301ms\n",
      "Write merged clustering\n",
      "[=================================================================] 352.50K 0s 364ms\n",
      "Time for merging to function_clusters_redundant: 0h 0m 0s 203ms\n",
      "Time for processing: 0h 0m 0s 830ms\n",
      "cluster component/component_redundant_db component/component_clusters_redundant tmp --min-seq-id 0.1 -c 0.5 --cov-mode 1 \n",
      "\n",
      "MMseqs Version:                     \t14.7e284\n",
      "Substitution matrix                 \taa:blosum62.out,nucl:nucleotide.out\n",
      "Seed substitution matrix            \taa:VTML80.out,nucl:nucleotide.out\n",
      "Sensitivity                         \t4\n",
      "k-mer length                        \t0\n",
      "k-score                             \tseq:2147483647,prof:2147483647\n",
      "Alphabet size                       \taa:21,nucl:5\n",
      "Max sequence length                 \t65535\n",
      "Max results per query               \t20\n",
      "Split database                      \t0\n",
      "Split mode                          \t2\n",
      "Split memory limit                  \t0\n",
      "Coverage threshold                  \t0.5\n",
      "Coverage mode                       \t1\n",
      "Compositional bias                  \t1\n",
      "Compositional bias                  \t1\n",
      "Diagonal scoring                    \ttrue\n",
      "Exact k-mer matching                \t0\n",
      "Mask residues                       \t1\n",
      "Mask residues probability           \t0.9\n",
      "Mask lower case residues            \t0\n",
      "Minimum diagonal score              \t15\n",
      "Selected taxa                       \t\n",
      "Include identical seq. id.          \tfalse\n",
      "Spaced k-mers                       \t1\n",
      "Preload mode                        \t0\n",
      "Pseudo count a                      \tsubstitution:1.100,context:1.400\n",
      "Pseudo count b                      \tsubstitution:4.100,context:5.800\n",
      "Spaced k-mer pattern                \t\n",
      "Local temporary path                \t\n",
      "Threads                             \t10\n",
      "Compressed                          \t0\n",
      "Verbosity                           \t3\n",
      "Add backtrace                       \tfalse\n",
      "Alignment mode                      \t3\n",
      "Alignment mode                      \t0\n",
      "Allow wrapped scoring               \tfalse\n",
      "E-value threshold                   \t0.001\n",
      "Seq. id. threshold                  \t0.1\n",
      "Min alignment length                \t0\n",
      "Seq. id. mode                       \t0\n",
      "Alternative alignments              \t0\n",
      "Max reject                          \t2147483647\n",
      "Max accept                          \t2147483647\n",
      "Score bias                          \t0\n",
      "Realign hits                        \tfalse\n",
      "Realign score bias                  \t-0.2\n",
      "Realign max seqs                    \t2147483647\n",
      "Correlation score weight            \t0\n",
      "Gap open cost                       \taa:11,nucl:5\n",
      "Gap extension cost                  \taa:1,nucl:2\n",
      "Zdrop                               \t40\n",
      "Rescore mode                        \t0\n",
      "Remove hits by seq. id. and coverage\tfalse\n",
      "Sort results                        \t0\n",
      "Cluster mode                        \t0\n",
      "Max connected component depth       \t1000\n",
      "Similarity type                     \t2\n",
      "Single step clustering              \tfalse\n",
      "Cascaded clustering steps           \t3\n",
      "Cluster reassign                    \tfalse\n",
      "Remove temporary files              \tfalse\n",
      "Force restart with latest tmp       \tfalse\n",
      "MPI runner                          \t\n",
      "k-mers per sequence                 \t21\n",
      "Scale k-mers per sequence           \taa:0.000,nucl:0.200\n",
      "Adjust k-mer length                 \tfalse\n",
      "Shift hash                          \t67\n",
      "Include only extendable             \tfalse\n",
      "Skip repeating k-mers               \tfalse\n",
      "\n",
      "Set cluster sensitivity to -s 6.000000\n",
      "Set cluster mode GREEDY MEM\n",
      "Set cluster iterations to 3\n",
      "linclust component/component_redundant_db tmp/14807645540069108862/clu_redundancy tmp/14807645540069108862/linclust --cluster-mode 3 --max-iterations 1000 --similarity-type 2 --threads 10 --compressed 0 -v 3 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' -a 0 --alignment-mode 3 --alignment-output-mode 0 --wrapped-scoring 0 -e 0.001 --min-seq-id 0.1 --min-aln-len 0 --seq-id-mode 0 --alt-ali 0 -c 0.5 --cov-mode 1 --max-seq-len 65535 --comp-bias-corr 1 --comp-bias-corr-scale 1 --max-rejected 2147483647 --max-accept 2147483647 --add-self-matches 0 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --score-bias 0 --realign 0 --realign-score-bias -0.2 --realign-max-seqs 2147483647 --corr-score-weight 0 --gap-open aa:11,nucl:5 --gap-extend aa:1,nucl:2 --zdrop 40 --alph-size aa:13,nucl:5 --kmer-per-seq 21 --spaced-kmer-mode 1 --kmer-per-seq-scale aa:0.000,nucl:0.200 --adjust-kmer-len 0 --mask 0 --mask-prob 0.9 --mask-lower-case 0 -k 0 --hash-shift 67 --split-memory-limit 0 --include-only-extendable 0 --ignore-multi-kmer 0 --rescore-mode 0 --filter-hits 0 --sort-results 0 --remove-tmp-files 0 --force-reuse 0 \n",
      "\n",
      "kmermatcher component/component_redundant_db tmp/14807645540069108862/linclust/11716136117351556594/pref --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' --alph-size aa:13,nucl:5 --min-seq-id 0.1 --kmer-per-seq 21 --spaced-kmer-mode 1 --kmer-per-seq-scale aa:0.000,nucl:0.200 --adjust-kmer-len 0 --mask 0 --mask-prob 0.9 --mask-lower-case 0 --cov-mode 1 -k 0 -c 0.5 --max-seq-len 65535 --hash-shift 67 --split-memory-limit 0 --include-only-extendable 0 --ignore-multi-kmer 0 --threads 10 --compressed 0 -v 3 \n",
      "\n",
      "kmermatcher component/component_redundant_db tmp/14807645540069108862/linclust/11716136117351556594/pref --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' --alph-size aa:13,nucl:5 --min-seq-id 0.1 --kmer-per-seq 21 --spaced-kmer-mode 1 --kmer-per-seq-scale aa:0.000,nucl:0.200 --adjust-kmer-len 0 --mask 0 --mask-prob 0.9 --mask-lower-case 0 --cov-mode 1 -k 0 -c 0.5 --max-seq-len 65535 --hash-shift 67 --split-memory-limit 0 --include-only-extendable 0 --ignore-multi-kmer 0 --threads 10 --compressed 0 -v 3 \n",
      "\n",
      "Database size: 304622 type: Aminoacid\n",
      "Reduced amino acid alphabet: (A S T) (C) (D B N) (E Q Z) (F Y) (G) (H) (I V) (K R) (L J M) (P) (W) (X) \n",
      "\n",
      "Generate k-mers list for 1 split\n",
      "[=================================================================] 304.62K 0s 605ms\n",
      "Sort kmer 0h 0m 0s 323ms\n",
      "Sort by rep. sequence 0h 0m 0s 222ms\n",
      "Time for fill: 0h 0m 0s 78ms\n",
      "Time for merging to pref: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 1s 403ms\n",
      "rescorediagonal component/component_redundant_db component/component_redundant_db tmp/14807645540069108862/linclust/11716136117351556594/pref tmp/14807645540069108862/linclust/11716136117351556594/pref_rescore1 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' --rescore-mode 0 --wrapped-scoring 0 --filter-hits 0 -e 0.001 -c 0.5 -a 0 --cov-mode 1 --min-seq-id 0.5 --min-aln-len 0 --seq-id-mode 0 --add-self-matches 0 --sort-results 0 --db-load-mode 0 --threads 10 --compressed 0 -v 3 \n",
      "\n",
      "[=================================================================] 304.62K 0s 111ms\n",
      "Time for merging to pref_rescore1: 0h 0m 0s 150ms\n",
      "Time for processing: 0h 0m 0s 436ms\n",
      "clust component/component_redundant_db tmp/14807645540069108862/linclust/11716136117351556594/pref_rescore1 tmp/14807645540069108862/linclust/11716136117351556594/pre_clust --cluster-mode 3 --max-iterations 1000 --similarity-type 2 --threads 10 --compressed 0 -v 3 \n",
      "\n",
      "Clustering mode: Greedy Low Mem\n",
      "Total time: 0h 0m 0s 45ms\n",
      "\n",
      "Size of the sequence database: 304622\n",
      "Size of the alignment database: 304622\n",
      "Number of clusters: 210629\n",
      "\n",
      "Writing results 0h 0m 0s 55ms\n",
      "Time for merging to pre_clust: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 138ms\n",
      "createsubdb tmp/14807645540069108862/linclust/11716136117351556594/order_redundancy component/component_redundant_db tmp/14807645540069108862/linclust/11716136117351556594/input_step_redundancy -v 3 --subdb-mode 1 \n",
      "\n",
      "Time for merging to input_step_redundancy: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 73ms\n",
      "createsubdb tmp/14807645540069108862/linclust/11716136117351556594/order_redundancy tmp/14807645540069108862/linclust/11716136117351556594/pref tmp/14807645540069108862/linclust/11716136117351556594/pref_filter1 -v 3 --subdb-mode 1 \n",
      "\n",
      "Time for merging to pref_filter1: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 76ms\n",
      "filterdb tmp/14807645540069108862/linclust/11716136117351556594/pref_filter1 tmp/14807645540069108862/linclust/11716136117351556594/pref_filter2 --filter-file tmp/14807645540069108862/linclust/11716136117351556594/order_redundancy --threads 10 --compressed 0 -v 3 \n",
      "\n",
      "Filtering using file(s)\n",
      "[=================================================================] 210.63K 0s 55ms\n",
      "Time for merging to pref_filter2: 0h 0m 0s 140ms\n",
      "Time for processing: 0h 0m 0s 368ms\n",
      "rescorediagonal tmp/14807645540069108862/linclust/11716136117351556594/input_step_redundancy tmp/14807645540069108862/linclust/11716136117351556594/input_step_redundancy tmp/14807645540069108862/linclust/11716136117351556594/pref_filter2 tmp/14807645540069108862/linclust/11716136117351556594/pref_rescore2 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' --rescore-mode 1 --wrapped-scoring 0 --filter-hits 1 -e 0.001 -c 0.5 -a 0 --cov-mode 1 --min-seq-id 0.1 --min-aln-len 0 --seq-id-mode 0 --add-self-matches 0 --sort-results 0 --db-load-mode 0 --threads 10 --compressed 0 -v 3 \n",
      "\n",
      "Can not find any score per column for coverage 0.500000 and sequence identity 0.100000. No hit will be filtered.\n",
      "[=================================================================] 210.63K 0s 102ms\n",
      "Time for merging to pref_rescore2: 0h 0m 0s 120ms\n",
      "Time for processing: 0h 0m 0s 326ms\n",
      "align tmp/14807645540069108862/linclust/11716136117351556594/input_step_redundancy tmp/14807645540069108862/linclust/11716136117351556594/input_step_redundancy tmp/14807645540069108862/linclust/11716136117351556594/pref_rescore2 tmp/14807645540069108862/linclust/11716136117351556594/aln --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' -a 0 --alignment-mode 3 --alignment-output-mode 0 --wrapped-scoring 0 -e 0.001 --min-seq-id 0.1 --min-aln-len 0 --seq-id-mode 0 --alt-ali 0 -c 0.5 --cov-mode 1 --max-seq-len 65535 --comp-bias-corr 1 --comp-bias-corr-scale 1 --max-rejected 2147483647 --max-accept 2147483647 --add-self-matches 0 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --score-bias 0 --realign 0 --realign-score-bias -0.2 --realign-max-seqs 2147483647 --corr-score-weight 0 --gap-open aa:11,nucl:5 --gap-extend aa:1,nucl:2 --zdrop 40 --threads 10 --compressed 0 -v 3 \n",
      "\n",
      "Compute score, coverage and sequence identity\n",
      "Query database size: 210629 type: Aminoacid\n",
      "Target database size: 210629 type: Aminoacid\n",
      "Calculation of alignments\n",
      "[=================================================================] 210.63K 30s 941ms\n",
      "Time for merging to aln: 0h 0m 0s 125ms\n",
      "580810 alignments calculated\n",
      "455755 sequence pairs passed the thresholds (0.784689 of overall calculated)\n",
      "2.163781 hits per query sequence\n",
      "Time for processing: 0h 0m 31s 292ms\n",
      "clust tmp/14807645540069108862/linclust/11716136117351556594/input_step_redundancy tmp/14807645540069108862/linclust/11716136117351556594/aln tmp/14807645540069108862/linclust/11716136117351556594/clust --cluster-mode 3 --max-iterations 1000 --similarity-type 2 --threads 10 --compressed 0 -v 3 \n",
      "\n",
      "Clustering mode: Greedy Low Mem\n",
      "Total time: 0h 0m 0s 33ms\n",
      "\n",
      "Size of the sequence database: 210629\n",
      "Size of the alignment database: 210629\n",
      "Number of clusters: 115899\n",
      "\n",
      "Writing results 0h 0m 0s 31ms\n",
      "Time for merging to clust: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 97ms\n",
      "mergeclusters component/component_redundant_db tmp/14807645540069108862/clu_redundancy tmp/14807645540069108862/linclust/11716136117351556594/pre_clust tmp/14807645540069108862/linclust/11716136117351556594/clust --threads 10 --compressed 0 -v 3 \n",
      "\n",
      "Clustering step 1\n",
      "[=================================================================] 210.63K 0s 19ms\n",
      "Clustering step 2\n",
      "[=================================================================] 115.90K 0s 40ms\n",
      "Write merged clustering\n",
      "[=================================================================] 304.62K 0s 64ms\n",
      "Time for merging to clu_redundancy: 0h 0m 0s 108ms\n",
      "Time for processing: 0h 0m 0s 244ms\n",
      "createsubdb tmp/14807645540069108862/clu_redundancy component/component_redundant_db tmp/14807645540069108862/input_step_redundancy -v 3 --subdb-mode 1 \n",
      "\n",
      "Time for merging to input_step_redundancy: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 50ms\n",
      "prefilter tmp/14807645540069108862/input_step_redundancy tmp/14807645540069108862/input_step_redundancy tmp/14807645540069108862/pref_step0 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' --seed-sub-mat 'aa:VTML80.out,nucl:nucleotide.out' -s 1 -k 0 --k-score seq:2147483647,prof:2147483647 --alph-size aa:21,nucl:5 --max-seq-len 65535 --max-seqs 20 --split 0 --split-mode 2 --split-memory-limit 0 -c 0.5 --cov-mode 1 --comp-bias-corr 0 --comp-bias-corr-scale 1 --diag-score 0 --exact-kmer-matching 0 --mask 1 --mask-prob 0.9 --mask-lower-case 0 --min-ungapped-score 0 --add-self-matches 0 --spaced-kmer-mode 1 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --threads 10 --compressed 0 -v 3 \n",
      "\n",
      "Query database size: 115899 type: Aminoacid\n",
      "Estimated memory consumption: 1G\n",
      "Target database size: 115899 type: Aminoacid\n",
      "Index table k-mer threshold: 154 at k-mer size 6 \n",
      "Index table: counting k-mers\n",
      "[=================================================================] 115.90K 0s 805ms\n",
      "Index table: Masked residues: 1349809\n",
      "Index table: fill\n",
      "[=================================================================] 115.90K 0s 572ms\n",
      "Index statistics\n",
      "Entries:          24093583\n",
      "DB size:          626 MB\n",
      "Avg k-mer size:   0.376462\n",
      "Top 10 k-mers\n",
      "    FAGTGK\t602\n",
      "    AYQGKT\t584\n",
      "    VHDKNI\t300\n",
      "    GQGGHT\t259\n",
      "    GQGGYT\t242\n",
      "    IPRSTR\t227\n",
      "    YGTSTY\t226\n",
      "    LGGFVY\t190\n",
      "    HHPICV\t153\n",
      "    PQLWEW\t143\n",
      "Time for index table init: 0h 0m 1s 667ms\n",
      "Process prefiltering step 1 of 1\n",
      "\n",
      "k-mer similarity threshold: 154\n",
      "Starting prefiltering scores calculation (step 1 of 1)\n",
      "Query db start 1 to 115899\n",
      "Target db start 1 to 115899\n",
      "[=================================================================] 115.90K 1s 951ms\n",
      "\n",
      "2.049775 k-mers per position\n",
      "794 DB matches per sequence\n",
      "0 overflows\n",
      "0 queries produce too many hits (truncated result)\n",
      "9 sequences passed prefiltering per query sequence\n",
      "8 median result list length\n",
      "0 sequences with 0 size result lists\n",
      "Time for merging to pref_step0: 0h 0m 0s 40ms\n",
      "Time for processing: 0h 0m 3s 961ms\n",
      "align tmp/14807645540069108862/input_step_redundancy tmp/14807645540069108862/input_step_redundancy tmp/14807645540069108862/pref_step0 tmp/14807645540069108862/aln_step0 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' -a 0 --alignment-mode 3 --alignment-output-mode 0 --wrapped-scoring 0 -e 0.001 --min-seq-id 0.1 --min-aln-len 0 --seq-id-mode 0 --alt-ali 0 -c 0.5 --cov-mode 1 --max-seq-len 65535 --comp-bias-corr 0 --comp-bias-corr-scale 1 --max-rejected 2147483647 --max-accept 2147483647 --add-self-matches 0 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --score-bias 0 --realign 0 --realign-score-bias -0.2 --realign-max-seqs 2147483647 --corr-score-weight 0 --gap-open aa:11,nucl:5 --gap-extend aa:1,nucl:2 --zdrop 40 --threads 10 --compressed 0 -v 3 \n",
      "\n",
      "Compute score, coverage and sequence identity\n",
      "Query database size: 115899 type: Aminoacid\n",
      "Target database size: 115899 type: Aminoacid\n",
      "Calculation of alignments\n",
      "[=================================================================] 115.90K 1m 24s 87ms\n",
      "Time for merging to aln_step0: 0h 0m 0s 219ms\n",
      "1022176 alignments calculated\n",
      "703355 sequence pairs passed the thresholds (0.688096 of overall calculated)\n",
      "6.068689 hits per query sequence\n",
      "Time for processing: 0h 1m 24s 455ms\n",
      "clust tmp/14807645540069108862/input_step_redundancy tmp/14807645540069108862/aln_step0 tmp/14807645540069108862/clu_step0 --cluster-mode 3 --max-iterations 1000 --similarity-type 2 --threads 10 --compressed 0 -v 3 \n",
      "\n",
      "Clustering mode: Greedy Low Mem\n",
      "Total time: 0h 0m 0s 298ms\n",
      "\n",
      "Size of the sequence database: 115899\n",
      "Size of the alignment database: 115899\n",
      "Number of clusters: 57209\n",
      "\n",
      "Writing results 0h 0m 0s 16ms\n",
      "Time for merging to clu_step0: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 556ms\n",
      "createsubdb tmp/14807645540069108862/clu_step0 tmp/14807645540069108862/input_step_redundancy tmp/14807645540069108862/input_step1 -v 3 --subdb-mode 1 \n",
      "\n",
      "Time for merging to input_step1: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 25ms\n",
      "prefilter tmp/14807645540069108862/input_step1 tmp/14807645540069108862/input_step1 tmp/14807645540069108862/pref_step1 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' --seed-sub-mat 'aa:VTML80.out,nucl:nucleotide.out' -s 3.5 -k 0 --k-score seq:2147483647,prof:2147483647 --alph-size aa:21,nucl:5 --max-seq-len 65535 --max-seqs 20 --split 0 --split-mode 2 --split-memory-limit 0 -c 0.5 --cov-mode 1 --comp-bias-corr 1 --comp-bias-corr-scale 1 --diag-score 1 --exact-kmer-matching 0 --mask 1 --mask-prob 0.9 --mask-lower-case 0 --min-ungapped-score 15 --add-self-matches 0 --spaced-kmer-mode 1 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --threads 10 --compressed 0 -v 3 \n",
      "\n",
      "Query database size: 57209 type: Aminoacid\n",
      "Estimated memory consumption: 1G\n",
      "Target database size: 57209 type: Aminoacid\n",
      "Index table k-mer threshold: 131 at k-mer size 6 \n",
      "Index table: counting k-mers\n",
      "[=================================================================] 57.21K 0s 510ms\n",
      "Index table: Masked residues: 858739\n",
      "Index table: fill\n",
      "[=================================================================] 57.21K 0s 613ms\n",
      "Index statistics\n",
      "Entries:          26585862\n",
      "DB size:          640 MB\n",
      "Avg k-mer size:   0.415404\n",
      "Top 10 k-mers\n",
      "    FAGTGK\t191\n",
      "    AYQGKT\t191\n",
      "    QTSKTM\t134\n",
      "    KLLDGS\t112\n",
      "    GPGGTL\t104\n",
      "    AAAAAP\t93\n",
      "    GQGGYT\t90\n",
      "    GQGGHT\t79\n",
      "    YGTSTY\t79\n",
      "    LGGFVY\t78\n",
      "Time for index table init: 0h 0m 1s 445ms\n",
      "Process prefiltering step 1 of 1\n",
      "\n",
      "k-mer similarity threshold: 131\n",
      "Starting prefiltering scores calculation (step 1 of 1)\n",
      "Query db start 1 to 57209\n",
      "Target db start 1 to 57209\n",
      "[=================================================================] 57.21K 7s 270ms\n",
      "\n",
      "34.706102 k-mers per position\n",
      "5784 DB matches per sequence\n",
      "0 overflows\n",
      "0 queries produce too many hits (truncated result)\n",
      "17 sequences passed prefiltering per query sequence\n",
      "20 median result list length\n",
      "0 sequences with 0 size result lists\n",
      "Time for merging to pref_step1: 0h 0m 0s 212ms\n",
      "Time for processing: 0h 0m 9s 310ms\n",
      "align tmp/14807645540069108862/input_step1 tmp/14807645540069108862/input_step1 tmp/14807645540069108862/pref_step1 tmp/14807645540069108862/aln_step1 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' -a 0 --alignment-mode 3 --alignment-output-mode 0 --wrapped-scoring 0 -e 0.001 --min-seq-id 0.1 --min-aln-len 0 --seq-id-mode 0 --alt-ali 0 -c 0.5 --cov-mode 1 --max-seq-len 65535 --comp-bias-corr 1 --comp-bias-corr-scale 1 --max-rejected 2147483647 --max-accept 2147483647 --add-self-matches 0 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --score-bias 0 --realign 0 --realign-score-bias -0.2 --realign-max-seqs 2147483647 --corr-score-weight 0 --gap-open aa:11,nucl:5 --gap-extend aa:1,nucl:2 --zdrop 40 --threads 10 --compressed 0 -v 3 \n",
      "\n",
      "Compute score, coverage and sequence identity\n",
      "Query database size: 57209 type: Aminoacid\n",
      "Target database size: 57209 type: Aminoacid\n",
      "Calculation of alignments\n",
      "[=================================================================] 57.21K 30s 755ms\n",
      "Time for merging to aln_step1: 0h 0m 0s 204ms\n",
      "773872 alignments calculated\n",
      "216641 sequence pairs passed the thresholds (0.279944 of overall calculated)\n",
      "3.786834 hits per query sequence\n",
      "Time for processing: 0h 0m 31s 71ms\n",
      "clust tmp/14807645540069108862/input_step1 tmp/14807645540069108862/aln_step1 tmp/14807645540069108862/clu_step1 --cluster-mode 3 --max-iterations 1000 --similarity-type 2 --threads 10 --compressed 0 -v 3 \n",
      "\n",
      "Clustering mode: Greedy Low Mem\n",
      "Total time: 0h 0m 0s 269ms\n",
      "\n",
      "Size of the sequence database: 57209\n",
      "Size of the alignment database: 57209\n",
      "Number of clusters: 33911\n",
      "\n",
      "Writing results 0h 0m 0s 11ms\n",
      "Time for merging to clu_step1: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 494ms\n",
      "createsubdb tmp/14807645540069108862/clu_step1 tmp/14807645540069108862/input_step1 tmp/14807645540069108862/input_step2 -v 3 --subdb-mode 1 \n",
      "\n",
      "Time for merging to input_step2: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 16ms\n",
      "prefilter tmp/14807645540069108862/input_step2 tmp/14807645540069108862/input_step2 tmp/14807645540069108862/pref_step2 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' --seed-sub-mat 'aa:VTML80.out,nucl:nucleotide.out' -s 6 -k 0 --k-score seq:2147483647,prof:2147483647 --alph-size aa:21,nucl:5 --max-seq-len 65535 --max-seqs 20 --split 0 --split-mode 2 --split-memory-limit 0 -c 0.5 --cov-mode 1 --comp-bias-corr 1 --comp-bias-corr-scale 1 --diag-score 1 --exact-kmer-matching 0 --mask 1 --mask-prob 0.9 --mask-lower-case 0 --min-ungapped-score 15 --add-self-matches 0 --spaced-kmer-mode 1 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --threads 10 --compressed 0 -v 3 \n",
      "\n",
      "Query database size: 33911 type: Aminoacid\n",
      "Estimated memory consumption: 1G\n",
      "Target database size: 33911 type: Aminoacid\n",
      "Index table k-mer threshold: 109 at k-mer size 6 \n",
      "Index table: counting k-mers\n",
      "[=================================================================] 33.91K 0s 367ms\n",
      "Index table: Masked residues: 653344\n",
      "Index table: fill\n",
      "[=================================================================] 33.91K 0s 387ms\n",
      "Index statistics\n",
      "Entries:          17818250\n",
      "DB size:          590 MB\n",
      "Avg k-mer size:   0.278410\n",
      "Top 10 k-mers\n",
      "    AAAAAA\t157\n",
      "    SFRLYG\t147\n",
      "    AYQGKT\t144\n",
      "    YRSLLL\t118\n",
      "    QTSKTM\t103\n",
      "    SSSSSS\t99\n",
      "    KLLDGS\t78\n",
      "    GQGGYT\t72\n",
      "    YGTSTY\t64\n",
      "    RDKTLQ\t61\n",
      "Time for index table init: 0h 0m 1s 85ms\n",
      "Process prefiltering step 1 of 1\n",
      "\n",
      "k-mer similarity threshold: 109\n",
      "Starting prefiltering scores calculation (step 1 of 1)\n",
      "Query db start 1 to 33911\n",
      "Target db start 1 to 33911\n",
      "[=================================================================] 33.91K 34s 845ms\n",
      "\n",
      "372.219593 k-mers per position\n",
      "51711 DB matches per sequence\n",
      "0 overflows\n",
      "0 queries produce too many hits (truncated result)\n",
      "19 sequences passed prefiltering per query sequence\n",
      "20 median result list length\n",
      "0 sequences with 0 size result lists\n",
      "Time for merging to pref_step2: 0h 0m 0s 12ms\n",
      "Time for processing: 0h 0m 36s 295ms\n",
      "align tmp/14807645540069108862/input_step2 tmp/14807645540069108862/input_step2 tmp/14807645540069108862/pref_step2 tmp/14807645540069108862/aln_step2 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' -a 0 --alignment-mode 3 --alignment-output-mode 0 --wrapped-scoring 0 -e 0.001 --min-seq-id 0.1 --min-aln-len 0 --seq-id-mode 0 --alt-ali 0 -c 0.5 --cov-mode 1 --max-seq-len 65535 --comp-bias-corr 1 --comp-bias-corr-scale 1 --max-rejected 2147483647 --max-accept 2147483647 --add-self-matches 0 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --score-bias 0 --realign 0 --realign-score-bias -0.2 --realign-max-seqs 2147483647 --corr-score-weight 0 --gap-open aa:11,nucl:5 --gap-extend aa:1,nucl:2 --zdrop 40 --threads 10 --compressed 0 -v 3 \n",
      "\n",
      "Compute score, coverage and sequence identity\n",
      "Query database size: 33911 type: Aminoacid\n",
      "Target database size: 33911 type: Aminoacid\n",
      "Calculation of alignments\n",
      "[=================================================================] 33.91K 13s 297ms\n",
      "Time for merging to aln_step2: 0h 0m 0s 217ms\n",
      "488727 alignments calculated\n",
      "73499 sequence pairs passed the thresholds (0.150389 of overall calculated)\n",
      "2.167409 hits per query sequence\n",
      "Time for processing: 0h 0m 13s 566ms\n",
      "clust tmp/14807645540069108862/input_step2 tmp/14807645540069108862/aln_step2 tmp/14807645540069108862/clu_step2 --cluster-mode 3 --max-iterations 1000 --similarity-type 2 --threads 10 --compressed 0 -v 3 \n",
      "\n",
      "Clustering mode: Greedy Low Mem\n",
      "Total time: 0h 0m 0s 26ms\n",
      "\n",
      "Size of the sequence database: 33911\n",
      "Size of the alignment database: 33911\n",
      "Number of clusters: 25315\n",
      "\n",
      "Writing results 0h 0m 0s 6ms\n",
      "Time for merging to clu_step2: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 40ms\n",
      "mergeclusters component/component_redundant_db component/component_clusters_redundant tmp/14807645540069108862/clu_redundancy tmp/14807645540069108862/clu_step0 tmp/14807645540069108862/clu_step1 tmp/14807645540069108862/clu_step2 --threads 10 --compressed 0 -v 3 \n",
      "\n",
      "Clustering step 1\n",
      "[=================================================================] 115.90K 0s 14ms\n",
      "Clustering step 2\n",
      "[=================================================================] 57.21K 0s 24ms\n",
      "Clustering step 3\n",
      "[=================================================================] 33.91K 0s 30ms\n",
      "Clustering step 4\n",
      "[=================================================================] 25.32K 0s 33ms\n",
      "Write merged clustering\n",
      "[=================================================================] 304.62K 0s 58ms\n",
      "Time for merging to component_clusters_redundant: 0h 0m 0s 66ms\n",
      "Time for processing: 0h 0m 0s 212ms\n",
      "cluster process/process_redundant_db process/process_clusters_redundant tmp --min-seq-id 0.1 -c 0.5 --cov-mode 1 \n",
      "\n",
      "MMseqs Version:                     \t14.7e284\n",
      "Substitution matrix                 \taa:blosum62.out,nucl:nucleotide.out\n",
      "Seed substitution matrix            \taa:VTML80.out,nucl:nucleotide.out\n",
      "Sensitivity                         \t4\n",
      "k-mer length                        \t0\n",
      "k-score                             \tseq:2147483647,prof:2147483647\n",
      "Alphabet size                       \taa:21,nucl:5\n",
      "Max sequence length                 \t65535\n",
      "Max results per query               \t20\n",
      "Split database                      \t0\n",
      "Split mode                          \t2\n",
      "Split memory limit                  \t0\n",
      "Coverage threshold                  \t0.5\n",
      "Coverage mode                       \t1\n",
      "Compositional bias                  \t1\n",
      "Compositional bias                  \t1\n",
      "Diagonal scoring                    \ttrue\n",
      "Exact k-mer matching                \t0\n",
      "Mask residues                       \t1\n",
      "Mask residues probability           \t0.9\n",
      "Mask lower case residues            \t0\n",
      "Minimum diagonal score              \t15\n",
      "Selected taxa                       \t\n",
      "Include identical seq. id.          \tfalse\n",
      "Spaced k-mers                       \t1\n",
      "Preload mode                        \t0\n",
      "Pseudo count a                      \tsubstitution:1.100,context:1.400\n",
      "Pseudo count b                      \tsubstitution:4.100,context:5.800\n",
      "Spaced k-mer pattern                \t\n",
      "Local temporary path                \t\n",
      "Threads                             \t10\n",
      "Compressed                          \t0\n",
      "Verbosity                           \t3\n",
      "Add backtrace                       \tfalse\n",
      "Alignment mode                      \t3\n",
      "Alignment mode                      \t0\n",
      "Allow wrapped scoring               \tfalse\n",
      "E-value threshold                   \t0.001\n",
      "Seq. id. threshold                  \t0.1\n",
      "Min alignment length                \t0\n",
      "Seq. id. mode                       \t0\n",
      "Alternative alignments              \t0\n",
      "Max reject                          \t2147483647\n",
      "Max accept                          \t2147483647\n",
      "Score bias                          \t0\n",
      "Realign hits                        \tfalse\n",
      "Realign score bias                  \t-0.2\n",
      "Realign max seqs                    \t2147483647\n",
      "Correlation score weight            \t0\n",
      "Gap open cost                       \taa:11,nucl:5\n",
      "Gap extension cost                  \taa:1,nucl:2\n",
      "Zdrop                               \t40\n",
      "Rescore mode                        \t0\n",
      "Remove hits by seq. id. and coverage\tfalse\n",
      "Sort results                        \t0\n",
      "Cluster mode                        \t0\n",
      "Max connected component depth       \t1000\n",
      "Similarity type                     \t2\n",
      "Single step clustering              \tfalse\n",
      "Cascaded clustering steps           \t3\n",
      "Cluster reassign                    \tfalse\n",
      "Remove temporary files              \tfalse\n",
      "Force restart with latest tmp       \tfalse\n",
      "MPI runner                          \t\n",
      "k-mers per sequence                 \t21\n",
      "Scale k-mers per sequence           \taa:0.000,nucl:0.200\n",
      "Adjust k-mer length                 \tfalse\n",
      "Shift hash                          \t67\n",
      "Include only extendable             \tfalse\n",
      "Skip repeating k-mers               \tfalse\n",
      "\n",
      "Set cluster sensitivity to -s 6.000000\n",
      "Set cluster mode GREEDY MEM\n",
      "Set cluster iterations to 3\n",
      "linclust process/process_redundant_db tmp/9968614989288362083/clu_redundancy tmp/9968614989288362083/linclust --cluster-mode 3 --max-iterations 1000 --similarity-type 2 --threads 10 --compressed 0 -v 3 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' -a 0 --alignment-mode 3 --alignment-output-mode 0 --wrapped-scoring 0 -e 0.001 --min-seq-id 0.1 --min-aln-len 0 --seq-id-mode 0 --alt-ali 0 -c 0.5 --cov-mode 1 --max-seq-len 65535 --comp-bias-corr 1 --comp-bias-corr-scale 1 --max-rejected 2147483647 --max-accept 2147483647 --add-self-matches 0 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --score-bias 0 --realign 0 --realign-score-bias -0.2 --realign-max-seqs 2147483647 --corr-score-weight 0 --gap-open aa:11,nucl:5 --gap-extend aa:1,nucl:2 --zdrop 40 --alph-size aa:13,nucl:5 --kmer-per-seq 21 --spaced-kmer-mode 1 --kmer-per-seq-scale aa:0.000,nucl:0.200 --adjust-kmer-len 0 --mask 0 --mask-prob 0.9 --mask-lower-case 0 -k 0 --hash-shift 67 --split-memory-limit 0 --include-only-extendable 0 --ignore-multi-kmer 0 --rescore-mode 0 --filter-hits 0 --sort-results 0 --remove-tmp-files 0 --force-reuse 0 \n",
      "\n",
      "kmermatcher process/process_redundant_db tmp/9968614989288362083/linclust/17894407855695214801/pref --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' --alph-size aa:13,nucl:5 --min-seq-id 0.1 --kmer-per-seq 21 --spaced-kmer-mode 1 --kmer-per-seq-scale aa:0.000,nucl:0.200 --adjust-kmer-len 0 --mask 0 --mask-prob 0.9 --mask-lower-case 0 --cov-mode 1 -k 0 -c 0.5 --max-seq-len 65535 --hash-shift 67 --split-memory-limit 0 --include-only-extendable 0 --ignore-multi-kmer 0 --threads 10 --compressed 0 -v 3 \n",
      "\n",
      "kmermatcher process/process_redundant_db tmp/9968614989288362083/linclust/17894407855695214801/pref --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' --alph-size aa:13,nucl:5 --min-seq-id 0.1 --kmer-per-seq 21 --spaced-kmer-mode 1 --kmer-per-seq-scale aa:0.000,nucl:0.200 --adjust-kmer-len 0 --mask 0 --mask-prob 0.9 --mask-lower-case 0 --cov-mode 1 -k 0 -c 0.5 --max-seq-len 65535 --hash-shift 67 --split-memory-limit 0 --include-only-extendable 0 --ignore-multi-kmer 0 --threads 10 --compressed 0 -v 3 \n",
      "\n",
      "Database size: 279080 type: Aminoacid\n",
      "Reduced amino acid alphabet: (A S T) (C) (D B N) (E Q Z) (F Y) (G) (H) (I V) (K R) (L J M) (P) (W) (X) \n",
      "\n",
      "Generate k-mers list for 1 split\n",
      "[=================================================================] 279.08K 0s 557ms\n",
      "Sort kmer 0h 0m 0s 121ms\n",
      "Sort by rep. sequence 0h 0m 0s 34ms\n",
      "Time for fill: 0h 0m 0s 72ms\n",
      "Time for merging to pref: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 958ms\n",
      "rescorediagonal process/process_redundant_db process/process_redundant_db tmp/9968614989288362083/linclust/17894407855695214801/pref tmp/9968614989288362083/linclust/17894407855695214801/pref_rescore1 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' --rescore-mode 0 --wrapped-scoring 0 --filter-hits 0 -e 0.001 -c 0.5 -a 0 --cov-mode 1 --min-seq-id 0.5 --min-aln-len 0 --seq-id-mode 0 --add-self-matches 0 --sort-results 0 --db-load-mode 0 --threads 10 --compressed 0 -v 3 \n",
      "\n",
      "[=================================================================] 279.08K 0s 98ms\n",
      "Time for merging to pref_rescore1: 0h 0m 0s 146ms\n",
      "Time for processing: 0h 0m 0s 348ms\n",
      "clust process/process_redundant_db tmp/9968614989288362083/linclust/17894407855695214801/pref_rescore1 tmp/9968614989288362083/linclust/17894407855695214801/pre_clust --cluster-mode 3 --max-iterations 1000 --similarity-type 2 --threads 10 --compressed 0 -v 3 \n",
      "\n",
      "Clustering mode: Greedy Low Mem\n",
      "Total time: 0h 0m 0s 33ms\n",
      "\n",
      "Size of the sequence database: 279080\n",
      "Size of the alignment database: 279080\n",
      "Number of clusters: 190102\n",
      "\n",
      "Writing results 0h 0m 0s 50ms\n",
      "Time for merging to pre_clust: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 119ms\n",
      "createsubdb tmp/9968614989288362083/linclust/17894407855695214801/order_redundancy process/process_redundant_db tmp/9968614989288362083/linclust/17894407855695214801/input_step_redundancy -v 3 --subdb-mode 1 \n",
      "\n",
      "Time for merging to input_step_redundancy: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 66ms\n",
      "createsubdb tmp/9968614989288362083/linclust/17894407855695214801/order_redundancy tmp/9968614989288362083/linclust/17894407855695214801/pref tmp/9968614989288362083/linclust/17894407855695214801/pref_filter1 -v 3 --subdb-mode 1 \n",
      "\n",
      "Time for merging to pref_filter1: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 76ms\n",
      "filterdb tmp/9968614989288362083/linclust/17894407855695214801/pref_filter1 tmp/9968614989288362083/linclust/17894407855695214801/pref_filter2 --filter-file tmp/9968614989288362083/linclust/17894407855695214801/order_redundancy --threads 10 --compressed 0 -v 3 \n",
      "\n",
      "Filtering using file(s)\n",
      "[=================================================================] 190.10K 0s 58ms\n",
      "Time for merging to pref_filter2: 0h 0m 0s 147ms\n",
      "Time for processing: 0h 0m 0s 383ms\n",
      "rescorediagonal tmp/9968614989288362083/linclust/17894407855695214801/input_step_redundancy tmp/9968614989288362083/linclust/17894407855695214801/input_step_redundancy tmp/9968614989288362083/linclust/17894407855695214801/pref_filter2 tmp/9968614989288362083/linclust/17894407855695214801/pref_rescore2 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' --rescore-mode 1 --wrapped-scoring 0 --filter-hits 1 -e 0.001 -c 0.5 -a 0 --cov-mode 1 --min-seq-id 0.1 --min-aln-len 0 --seq-id-mode 0 --add-self-matches 0 --sort-results 0 --db-load-mode 0 --threads 10 --compressed 0 -v 3 \n",
      "\n",
      "Can not find any score per column for coverage 0.500000 and sequence identity 0.100000. No hit will be filtered.\n",
      "[=================================================================] 190.10K 0s 95ms\n",
      "Time for merging to pref_rescore2: 0h 0m 0s 123ms\n",
      "Time for processing: 0h 0m 0s 315ms\n",
      "align tmp/9968614989288362083/linclust/17894407855695214801/input_step_redundancy tmp/9968614989288362083/linclust/17894407855695214801/input_step_redundancy tmp/9968614989288362083/linclust/17894407855695214801/pref_rescore2 tmp/9968614989288362083/linclust/17894407855695214801/aln --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' -a 0 --alignment-mode 3 --alignment-output-mode 0 --wrapped-scoring 0 -e 0.001 --min-seq-id 0.1 --min-aln-len 0 --seq-id-mode 0 --alt-ali 0 -c 0.5 --cov-mode 1 --max-seq-len 65535 --comp-bias-corr 1 --comp-bias-corr-scale 1 --max-rejected 2147483647 --max-accept 2147483647 --add-self-matches 0 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --score-bias 0 --realign 0 --realign-score-bias -0.2 --realign-max-seqs 2147483647 --corr-score-weight 0 --gap-open aa:11,nucl:5 --gap-extend aa:1,nucl:2 --zdrop 40 --threads 10 --compressed 0 -v 3 \n",
      "\n",
      "Compute score, coverage and sequence identity\n",
      "Query database size: 190102 type: Aminoacid\n",
      "Target database size: 190102 type: Aminoacid\n",
      "Calculation of alignments\n",
      "[=================================================================] 190.10K 30s 567ms\n",
      "Time for merging to aln: 0h 0m 0s 183ms\n",
      "518535 alignments calculated\n",
      "416394 sequence pairs passed the thresholds (0.803020 of overall calculated)\n",
      "2.190372 hits per query sequence\n",
      "Time for processing: 0h 0m 30s 816ms\n",
      "clust tmp/9968614989288362083/linclust/17894407855695214801/input_step_redundancy tmp/9968614989288362083/linclust/17894407855695214801/aln tmp/9968614989288362083/linclust/17894407855695214801/clust --cluster-mode 3 --max-iterations 1000 --similarity-type 2 --threads 10 --compressed 0 -v 3 \n",
      "\n",
      "Clustering mode: Greedy Low Mem\n",
      "Total time: 0h 0m 0s 25ms\n",
      "\n",
      "Size of the sequence database: 190102\n",
      "Size of the alignment database: 190102\n",
      "Number of clusters: 101520\n",
      "\n",
      "Writing results 0h 0m 0s 28ms\n",
      "Time for merging to clust: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 76ms\n",
      "mergeclusters process/process_redundant_db tmp/9968614989288362083/clu_redundancy tmp/9968614989288362083/linclust/17894407855695214801/pre_clust tmp/9968614989288362083/linclust/17894407855695214801/clust --threads 10 --compressed 0 -v 3 \n",
      "\n",
      "Clustering step 1\n",
      "[=================================================================] 190.10K 0s 18ms\n",
      "Clustering step 2\n",
      "[=================================================================] 101.52K 0s 32ms\n",
      "Write merged clustering\n",
      "[=================================================================] 279.08K 0s 54ms\n",
      "Time for merging to clu_redundancy: 0h 0m 0s 104ms\n",
      "Time for processing: 0h 0m 0s 226ms\n",
      "createsubdb tmp/9968614989288362083/clu_redundancy process/process_redundant_db tmp/9968614989288362083/input_step_redundancy -v 3 --subdb-mode 1 \n",
      "\n",
      "Time for merging to input_step_redundancy: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 44ms\n",
      "prefilter tmp/9968614989288362083/input_step_redundancy tmp/9968614989288362083/input_step_redundancy tmp/9968614989288362083/pref_step0 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' --seed-sub-mat 'aa:VTML80.out,nucl:nucleotide.out' -s 1 -k 0 --k-score seq:2147483647,prof:2147483647 --alph-size aa:21,nucl:5 --max-seq-len 65535 --max-seqs 20 --split 0 --split-mode 2 --split-memory-limit 0 -c 0.5 --cov-mode 1 --comp-bias-corr 0 --comp-bias-corr-scale 1 --diag-score 0 --exact-kmer-matching 0 --mask 1 --mask-prob 0.9 --mask-lower-case 0 --min-ungapped-score 0 --add-self-matches 0 --spaced-kmer-mode 1 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --threads 10 --compressed 0 -v 3 \n",
      "\n",
      "Query database size: 101520 type: Aminoacid\n",
      "Estimated memory consumption: 1G\n",
      "Target database size: 101520 type: Aminoacid\n",
      "Index table k-mer threshold: 154 at k-mer size 6 \n",
      "Index table: counting k-mers\n",
      "[=================================================================] 101.52K 0s 712ms\n",
      "Index table: Masked residues: 1162670\n",
      "Index table: fill\n",
      "[=================================================================] 101.52K 0s 535ms\n",
      "Index statistics\n",
      "Entries:          21859836\n",
      "DB size:          613 MB\n",
      "Avg k-mer size:   0.341560\n",
      "Top 10 k-mers\n",
      "    FAGTGK\t710\n",
      "    AYQGKT\t653\n",
      "    GQGGYT\t268\n",
      "    YGTSTY\t248\n",
      "    HVYDLT\t182\n",
      "    HIYDLT\t169\n",
      "    LGGFVY\t158\n",
      "    SGTTGP\t136\n",
      "    NTFNLQ\t136\n",
      "    PGGTAR\t132\n",
      "Time for index table init: 0h 0m 1s 557ms\n",
      "Process prefiltering step 1 of 1\n",
      "\n",
      "k-mer similarity threshold: 154\n",
      "Starting prefiltering scores calculation (step 1 of 1)\n",
      "Query db start 1 to 101520\n",
      "Target db start 1 to 101520\n",
      "[=================================================================] 101.52K 1s 758ms\n",
      "\n",
      "2.027772 k-mers per position\n",
      "769 DB matches per sequence\n",
      "0 overflows\n",
      "0 queries produce too many hits (truncated result)\n",
      "10 sequences passed prefiltering per query sequence\n",
      "9 median result list length\n",
      "0 sequences with 0 size result lists\n",
      "Time for merging to pref_step0: 0h 0m 0s 31ms\n",
      "Time for processing: 0h 0m 3s 644ms\n",
      "align tmp/9968614989288362083/input_step_redundancy tmp/9968614989288362083/input_step_redundancy tmp/9968614989288362083/pref_step0 tmp/9968614989288362083/aln_step0 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' -a 0 --alignment-mode 3 --alignment-output-mode 0 --wrapped-scoring 0 -e 0.001 --min-seq-id 0.1 --min-aln-len 0 --seq-id-mode 0 --alt-ali 0 -c 0.5 --cov-mode 1 --max-seq-len 65535 --comp-bias-corr 0 --comp-bias-corr-scale 1 --max-rejected 2147483647 --max-accept 2147483647 --add-self-matches 0 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --score-bias 0 --realign 0 --realign-score-bias -0.2 --realign-max-seqs 2147483647 --corr-score-weight 0 --gap-open aa:11,nucl:5 --gap-extend aa:1,nucl:2 --zdrop 40 --threads 10 --compressed 0 -v 3 \n",
      "\n",
      "Compute score, coverage and sequence identity\n",
      "Query database size: 101520 type: Aminoacid\n",
      "Target database size: 101520 type: Aminoacid\n",
      "Calculation of alignments\n",
      "[=================================================================] 101.52K 1m 23s 101ms\n",
      "Time for merging to aln_step0: 0h 0m 0s 216ms\n",
      "934590 alignments calculated\n",
      "659192 sequence pairs passed the thresholds (0.705327 of overall calculated)\n",
      "6.493223 hits per query sequence\n",
      "Time for processing: 0h 1m 23s 449ms\n",
      "clust tmp/9968614989288362083/input_step_redundancy tmp/9968614989288362083/aln_step0 tmp/9968614989288362083/clu_step0 --cluster-mode 3 --max-iterations 1000 --similarity-type 2 --threads 10 --compressed 0 -v 3 \n",
      "\n",
      "Clustering mode: Greedy Low Mem\n",
      "Total time: 0h 0m 0s 50ms\n",
      "\n",
      "Size of the sequence database: 101520\n",
      "Size of the alignment database: 101520\n",
      "Number of clusters: 48238\n",
      "\n",
      "Writing results 0h 0m 0s 13ms\n",
      "Time for merging to clu_step0: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 113ms\n",
      "createsubdb tmp/9968614989288362083/clu_step0 tmp/9968614989288362083/input_step_redundancy tmp/9968614989288362083/input_step1 -v 3 --subdb-mode 1 \n",
      "\n",
      "Time for merging to input_step1: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 21ms\n",
      "prefilter tmp/9968614989288362083/input_step1 tmp/9968614989288362083/input_step1 tmp/9968614989288362083/pref_step1 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' --seed-sub-mat 'aa:VTML80.out,nucl:nucleotide.out' -s 3.5 -k 0 --k-score seq:2147483647,prof:2147483647 --alph-size aa:21,nucl:5 --max-seq-len 65535 --max-seqs 20 --split 0 --split-mode 2 --split-memory-limit 0 -c 0.5 --cov-mode 1 --comp-bias-corr 1 --comp-bias-corr-scale 1 --diag-score 1 --exact-kmer-matching 0 --mask 1 --mask-prob 0.9 --mask-lower-case 0 --min-ungapped-score 15 --add-self-matches 0 --spaced-kmer-mode 1 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --threads 10 --compressed 0 -v 3 \n",
      "\n",
      "Query database size: 48238 type: Aminoacid\n",
      "Estimated memory consumption: 1G\n",
      "Target database size: 48238 type: Aminoacid\n",
      "Index table k-mer threshold: 131 at k-mer size 6 \n",
      "Index table: counting k-mers\n",
      "[=================================================================] 48.24K 0s 459ms\n",
      "Index table: Masked residues: 715986\n",
      "Index table: fill\n",
      "[=================================================================] 48.24K 0s 477ms\n",
      "Index statistics\n",
      "Entries:          23319005\n",
      "DB size:          621 MB\n",
      "Avg k-mer size:   0.364359\n",
      "Top 10 k-mers\n",
      "    FAGTGK\t224\n",
      "    AYQGKT\t210\n",
      "    QTSKTM\t172\n",
      "    VPRSTR\t129\n",
      "    KLLDGS\t102\n",
      "    GQGGYT\t102\n",
      "    YGTSTY\t90\n",
      "    LGGFVY\t67\n",
      "    PGGTAR\t59\n",
      "    LVCASR\t59\n",
      "Time for index table init: 0h 0m 1s 216ms\n",
      "Process prefiltering step 1 of 1\n",
      "\n",
      "k-mer similarity threshold: 131\n",
      "Starting prefiltering scores calculation (step 1 of 1)\n",
      "Query db start 1 to 48238\n",
      "Target db start 1 to 48238\n",
      "[=================================================================] 48.24K 6s 173ms\n",
      "\n",
      "35.646166 k-mers per position\n",
      "5485 DB matches per sequence\n",
      "0 overflows\n",
      "0 queries produce too many hits (truncated result)\n",
      "17 sequences passed prefiltering per query sequence\n",
      "20 median result list length\n",
      "0 sequences with 0 size result lists\n",
      "Time for merging to pref_step1: 0h 0m 0s 33ms\n",
      "Time for processing: 0h 0m 7s 766ms\n",
      "align tmp/9968614989288362083/input_step1 tmp/9968614989288362083/input_step1 tmp/9968614989288362083/pref_step1 tmp/9968614989288362083/aln_step1 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' -a 0 --alignment-mode 3 --alignment-output-mode 0 --wrapped-scoring 0 -e 0.001 --min-seq-id 0.1 --min-aln-len 0 --seq-id-mode 0 --alt-ali 0 -c 0.5 --cov-mode 1 --max-seq-len 65535 --comp-bias-corr 1 --comp-bias-corr-scale 1 --max-rejected 2147483647 --max-accept 2147483647 --add-self-matches 0 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --score-bias 0 --realign 0 --realign-score-bias -0.2 --realign-max-seqs 2147483647 --corr-score-weight 0 --gap-open aa:11,nucl:5 --gap-extend aa:1,nucl:2 --zdrop 40 --threads 10 --compressed 0 -v 3 \n",
      "\n",
      "Compute score, coverage and sequence identity\n",
      "Query database size: 48238 type: Aminoacid\n",
      "Target database size: 48238 type: Aminoacid\n",
      "Calculation of alignments\n",
      "[=================================================================] 48.24K 28s 331ms\n",
      "Time for merging to aln_step1: 0h 0m 0s 197ms\n",
      "656141 alignments calculated\n",
      "188872 sequence pairs passed the thresholds (0.287853 of overall calculated)\n",
      "3.915419 hits per query sequence\n",
      "Time for processing: 0h 0m 28s 639ms\n",
      "clust tmp/9968614989288362083/input_step1 tmp/9968614989288362083/aln_step1 tmp/9968614989288362083/clu_step1 --cluster-mode 3 --max-iterations 1000 --similarity-type 2 --threads 10 --compressed 0 -v 3 \n",
      "\n",
      "Clustering mode: Greedy Low Mem\n",
      "Total time: 0h 0m 0s 33ms\n",
      "\n",
      "Size of the sequence database: 48238\n",
      "Size of the alignment database: 48238\n",
      "Number of clusters: 27968\n",
      "\n",
      "Writing results 0h 0m 0s 8ms\n",
      "Time for merging to clu_step1: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 56ms\n",
      "createsubdb tmp/9968614989288362083/clu_step1 tmp/9968614989288362083/input_step1 tmp/9968614989288362083/input_step2 -v 3 --subdb-mode 1 \n",
      "\n",
      "Time for merging to input_step2: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 14ms\n",
      "prefilter tmp/9968614989288362083/input_step2 tmp/9968614989288362083/input_step2 tmp/9968614989288362083/pref_step2 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' --seed-sub-mat 'aa:VTML80.out,nucl:nucleotide.out' -s 6 -k 0 --k-score seq:2147483647,prof:2147483647 --alph-size aa:21,nucl:5 --max-seq-len 65535 --max-seqs 20 --split 0 --split-mode 2 --split-memory-limit 0 -c 0.5 --cov-mode 1 --comp-bias-corr 1 --comp-bias-corr-scale 1 --diag-score 1 --exact-kmer-matching 0 --mask 1 --mask-prob 0.9 --mask-lower-case 0 --min-ungapped-score 15 --add-self-matches 0 --spaced-kmer-mode 1 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --threads 10 --compressed 0 -v 3 \n",
      "\n",
      "Query database size: 27968 type: Aminoacid\n",
      "Estimated memory consumption: 1G\n",
      "Target database size: 27968 type: Aminoacid\n",
      "Index table k-mer threshold: 109 at k-mer size 6 \n",
      "Index table: counting k-mers\n",
      "[=================================================================] 27.97K 0s 297ms\n",
      "Index table: Masked residues: 527130\n",
      "Index table: fill\n",
      "[=================================================================] 27.97K 0s 327ms\n",
      "Index statistics\n",
      "Entries:          15466191\n",
      "DB size:          576 MB\n",
      "Avg k-mer size:   0.241659\n",
      "Top 10 k-mers\n",
      "    FAGTGK\t168\n",
      "    AYQGKT\t163\n",
      "    QTSKTM\t126\n",
      "    VPRSTR\t93\n",
      "    SSSSSS\t92\n",
      "    GQGGYT\t83\n",
      "    YGTSTY\t66\n",
      "    HVYDLT\t57\n",
      "    AAAAAV\t55\n",
      "    VHDKNI\t50\n",
      "Time for index table init: 0h 0m 0s 915ms\n",
      "Process prefiltering step 1 of 1\n",
      "\n",
      "k-mer similarity threshold: 109\n",
      "Starting prefiltering scores calculation (step 1 of 1)\n",
      "Query db start 1 to 27968\n",
      "Target db start 1 to 27968\n",
      "[=================================================================] 27.97K 28s 389ms\n",
      "\n",
      "381.915693 k-mers per position\n",
      "48472 DB matches per sequence\n",
      "0 overflows\n",
      "0 queries produce too many hits (truncated result)\n",
      "19 sequences passed prefiltering per query sequence\n",
      "20 median result list length\n",
      "0 sequences with 0 size result lists\n",
      "Time for merging to pref_step2: 0h 0m 0s 11ms\n",
      "Time for processing: 0h 0m 29s 590ms\n",
      "align tmp/9968614989288362083/input_step2 tmp/9968614989288362083/input_step2 tmp/9968614989288362083/pref_step2 tmp/9968614989288362083/aln_step2 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' -a 0 --alignment-mode 3 --alignment-output-mode 0 --wrapped-scoring 0 -e 0.001 --min-seq-id 0.1 --min-aln-len 0 --seq-id-mode 0 --alt-ali 0 -c 0.5 --cov-mode 1 --max-seq-len 65535 --comp-bias-corr 1 --comp-bias-corr-scale 1 --max-rejected 2147483647 --max-accept 2147483647 --add-self-matches 0 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --score-bias 0 --realign 0 --realign-score-bias -0.2 --realign-max-seqs 2147483647 --corr-score-weight 0 --gap-open aa:11,nucl:5 --gap-extend aa:1,nucl:2 --zdrop 40 --threads 10 --compressed 0 -v 3 \n",
      "\n",
      "Compute score, coverage and sequence identity\n",
      "Query database size: 27968 type: Aminoacid\n",
      "Target database size: 27968 type: Aminoacid\n",
      "Calculation of alignments\n",
      "[=================================================================] 27.97K 12s 94ms\n",
      "Time for merging to aln_step2: 0h 0m 0s 57ms\n",
      "409986 alignments calculated\n",
      "60303 sequence pairs passed the thresholds (0.147086 of overall calculated)\n",
      "2.156143 hits per query sequence\n",
      "Time for processing: 0h 0m 12s 260ms\n",
      "clust tmp/9968614989288362083/input_step2 tmp/9968614989288362083/aln_step2 tmp/9968614989288362083/clu_step2 --cluster-mode 3 --max-iterations 1000 --similarity-type 2 --threads 10 --compressed 0 -v 3 \n",
      "\n",
      "Clustering mode: Greedy Low Mem\n",
      "Total time: 0h 0m 0s 6ms\n",
      "\n",
      "Size of the sequence database: 27968\n",
      "Size of the alignment database: 27968\n",
      "Number of clusters: 20805\n",
      "\n",
      "Writing results 0h 0m 0s 5ms\n",
      "Time for merging to clu_step2: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 33ms\n",
      "mergeclusters process/process_redundant_db process/process_clusters_redundant tmp/9968614989288362083/clu_redundancy tmp/9968614989288362083/clu_step0 tmp/9968614989288362083/clu_step1 tmp/9968614989288362083/clu_step2 --threads 10 --compressed 0 -v 3 \n",
      "\n",
      "Clustering step 1\n",
      "[=================================================================] 101.52K 0s 13ms\n",
      "Clustering step 2\n",
      "[=================================================================] 48.24K 0s 22ms\n",
      "Clustering step 3\n",
      "[=================================================================] 27.97K 0s 28ms\n",
      "Clustering step 4\n",
      "[=================================================================] 20.81K 0s 30ms\n",
      "Write merged clustering\n",
      "[=================================================================] 279.08K 0s 51ms\n",
      "Time for merging to process_clusters_redundant: 0h 0m 0s 57ms\n",
      "Time for processing: 0h 0m 0s 198ms\n",
      "Creating TSV files...\n",
      "createtsv function/function_redundant_db function/function_redundant_db function/function_clusters_redundant function/function_clusters_redundant.tsv \n",
      "\n",
      "MMseqs Version:                 \t14.7e284\n",
      "First sequence as representative\tfalse\n",
      "Target column                   \t1\n",
      "Add full header                 \tfalse\n",
      "Sequence source                 \t0\n",
      "Database output                 \tfalse\n",
      "Threads                         \t10\n",
      "Compressed                      \t0\n",
      "Verbosity                       \t3\n",
      "\n",
      "Time for merging to function_clusters_redundant.tsv: 0h 0m 0s 37ms\n",
      "Time for processing: 0h 0m 0s 163ms\n",
      "createtsv component/component_redundant_db component/component_redundant_db component/component_clusters_redundant component/component_clusters_redundant.tsv \n",
      "\n",
      "MMseqs Version:                 \t14.7e284\n",
      "First sequence as representative\tfalse\n",
      "Target column                   \t1\n",
      "Add full header                 \tfalse\n",
      "Sequence source                 \t0\n",
      "Database output                 \tfalse\n",
      "Threads                         \t10\n",
      "Compressed                      \t0\n",
      "Verbosity                       \t3\n",
      "\n",
      "Time for merging to component_clusters_redundant.tsv: 0h 0m 0s 19ms\n",
      "Time for processing: 0h 0m 0s 97ms\n",
      "createtsv process/process_redundant_db process/process_redundant_db process/process_clusters_redundant process/process_clusters_redundant.tsv \n",
      "\n",
      "MMseqs Version:                 \t14.7e284\n",
      "First sequence as representative\tfalse\n",
      "Target column                   \t1\n",
      "Add full header                 \tfalse\n",
      "Sequence source                 \t0\n",
      "Database output                 \tfalse\n",
      "Threads                         \t10\n",
      "Compressed                      \t0\n",
      "Verbosity                       \t3\n",
      "\n",
      "Time for merging to process_clusters_redundant.tsv: 0h 0m 0s 92ms\n",
      "Time for processing: 0h 0m 0s 162ms\n",
      "All tasks completed successfully.\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "CompletedProcess(args=['./mmseq_cluster.sh', '/Users/harsh/Documents/PFP_PUBLISH/raw_data_from_uniprot/reduced_90_similarity_10', '/Users/harsh/Documents/PFP_PUBLISH/raw_data_from_uniprot/reduced_90_function_propagated.fasta', '/Users/harsh/Documents/PFP_PUBLISH/raw_data_from_uniprot/reduced_90_component_propagated.fasta', '/Users/harsh/Documents/PFP_PUBLISH/raw_data_from_uniprot/reduced_90_process_propagated.fasta', '0.1'], returncode=0)"
      ]
     },
     "execution_count": 84,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "subprocess.run([\"./mmseq_cluster.sh\", \"reduced_90_similarity_10\", \"reduced_90_function_propagated.fasta\", \"reduced_90_component_propagated.fasta\", \"reduced_90_process_propagated.fasta\", \"0.1\"])"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Train, dev, test split"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 94,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "There are 36398 clusters in the function dataset.\n",
      "There are 36571 clusters in the component dataset.\n",
      "There are 30239 clusters in the process dataset.\n"
     ]
    }
   ],
   "source": [
    "function_mmseq_out = pd.read_csv(\"reduced_90_similarity_30/function/function_clusters_redundant.tsv\", sep='\\t', header=None)\n",
    "function_mmseq_out.columns = ['Cluster_2', 'Entry']\n",
    "\n",
    "component_mmseq_out = pd.read_csv(\"reduced_90_similarity_30/component/component_clusters_redundant.tsv\", sep='\\t', header=None)\n",
    "component_mmseq_out.columns = ['Cluster_2', 'Entry']\n",
    "\n",
    "process_mmseq_out = pd.read_csv(\"reduced_90_similarity_30/process/process_clusters_redundant.tsv\", sep='\\t', header=None)\n",
    "process_mmseq_out.columns = ['Cluster_2', 'Entry']\n",
    "\n",
    "print(\"There are {} clusters in the function dataset.\".format(function_mmseq_out['Cluster_2'].nunique()))\n",
    "print(\"There are {} clusters in the component dataset.\".format(component_mmseq_out['Cluster_2'].nunique()))\n",
    "print(\"There are {} clusters in the process dataset.\".format(process_mmseq_out['Cluster_2'].nunique()))\n",
    "\n",
    "assert len(function_mmseq_out) == len(total_SAR_function_GO)\n",
    "assert len(component_mmseq_out) == len(total_SAR_component_GO)\n",
    "assert len(process_mmseq_out) == len(total_SAR_process_GO)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 95,
   "metadata": {},
   "outputs": [],
   "source": [
    "#merge with the original datasets\n",
    "function_merged = pd.merge(function_mmseq_out, total_SAR_function_GO, on='Entry')\n",
    "component_merged = pd.merge(component_mmseq_out, total_SAR_component_GO, on='Entry')\n",
    "process_merged = pd.merge(process_mmseq_out, total_SAR_process_GO, on='Entry')\n",
    "\n",
    "assert len(function_merged) == len(function_mmseq_out)\n",
    "assert len(component_merged) == len(component_mmseq_out)\n",
    "assert len(process_merged) == len(process_mmseq_out)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 87,
   "metadata": {},
   "outputs": [],
   "source": [
    "#read in RNA_associated_proteins_UniProtIDs.pkl\n",
    "RNA_associated_proteins = pd.read_pickle(\"/RNA_associated_proteins_UniProtIDs.pkl\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 96,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Original shapes:\n",
      "Function: (352499, 14)\n",
      "Component: (304622, 14)\n",
      "Process: (279080, 14)\n",
      "After removing RNA associated proteins:\n",
      "Function: (352461, 14)\n",
      "Component: (304567, 14)\n",
      "Process: (279033, 14)\n"
     ]
    }
   ],
   "source": [
    "seqs_to_remove = RNA_associated_proteins\n",
    "\n",
    "#print the original shapes \n",
    "print(\"Original shapes:\")\n",
    "print(\"Function:\", function_merged.shape)\n",
    "print(\"Component:\", component_merged.shape)\n",
    "print(\"Process:\", process_merged.shape)\n",
    "\n",
    "function_merged = function_merged[~function_merged['Entry'].isin(seqs_to_remove)]\n",
    "component_merged = component_merged[~component_merged['Entry'].isin(seqs_to_remove)]\n",
    "process_merged = process_merged[~process_merged['Entry'].isin(seqs_to_remove)]\n",
    "\n",
    "print(\"After removing RNA associated proteins:\")\n",
    "print(\"Function:\", function_merged.shape)\n",
    "print(\"Component:\", component_merged.shape)\n",
    "print(\"Process:\", process_merged.shape)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 99,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Processing rows: 100%|██████████| 352461/352461 [00:09<00:00, 37125.60it/s]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Identifying unique and multi-occurrence clusters\n",
      "ShuffleSplit for training, validation, and test sets from multi occurrence clusters\n",
      "Extracting rows for each set\n",
      "Moving clusters with missing terms from validation/test to training\n",
      "Validation terms not in training: set()\n",
      "Test terms not in training: set()\n",
      "Terms below threshold (50 occurrences): {'GO:0170054', 'GO:0035325', 'GO:0042083', 'GO:0016723', 'GO:0015616', 'GO:0099106', 'GO:0004673', 'GO:0016815', 'GO:0015220', 'GO:0046592', 'GO:0004325', 'GO:0008920', 'GO:0034338', 'GO:0015378', 'GO:0016869', 'GO:0140359', 'GO:0002061', 'GO:0042171', 'GO:0035515', 'GO:0016664', 'GO:0019903', 'GO:0005345', 'GO:0005337', 'GO:0015296', 'GO:0043047', 'GO:0050815', 'GO:0140666', 'GO:0030291', 'GO:0033840', 'GO:0051731', 'GO:0019199', 'GO:0005343', 'GO:0036080', 'GO:0036009', 'GO:0005272', 'GO:0016885', 'GO:0015020', 'GO:0016672', 'GO:0030611', 'GO:0003905', 'GO:0033764', 'GO:0070137', 'GO:0015173', 'GO:0043394', 'GO:0030283', 'GO:0015091', 'GO:0019777', 'GO:0004104', 'GO:0016662', 'GO:0004115', 'GO:0016801', 'GO:0015355', 'GO:0000993', 'GO:0035671', 'GO:0003964', 'GO:0140674', 'GO:0009669', 'GO:0031544', 'GO:0170053', 'GO:0002060', 'GO:0001653', 'GO:0045289', 'GO:0055056', 'GO:0004344', 'GO:0004738', 'GO:0015095', 'GO:0005365', 'GO:0051734', 'GO:0106345', 'GO:0016663', 'GO:0099604', 'GO:0005287', 'GO:0016406', 'GO:0019904', 'GO:0015185', 'GO:0005245', 'GO:0015152', 'GO:0004486', 'GO:0051032', 'GO:0016421', 'GO:0044390', 'GO:0031593', 'GO:0140939', 'GO:0016824', 'GO:0042910', 'GO:0008486', 'GO:0015526', 'GO:0050309', 'GO:0047555', 'GO:0015154', 'GO:0016715', 'GO:0000099', 'GO:0043364', 'GO:0141052', 'GO:0015141', 'GO:0004407', 'GO:0140658', 'GO:0005353', 'GO:0009013', 'GO:0004860', 'GO:0047617', 'GO:0031543', 'GO:0050486', 'GO:0043225', 'GO:0015149', 'GO:0030276', 'GO:0070004', 'GO:0015119', 'GO:0016420', 'GO:0008907', 'GO:0050664', 'GO:0004806', 'GO:0052738', 'GO:0004930', 'GO:0034596', 'GO:0015373', 'GO:0019798', 'GO:0016803', 'GO:0080084', 'GO:0004564', 'GO:0015172', 'GO:0004396', 'GO:0070548', 'GO:0008554', 'GO:0140721', 'GO:0010354', 'GO:0008556', 'GO:0004463', 'GO:0016713', 'GO:0106018', 'GO:0008376', 'GO:0004888', 'GO:0043893', 'GO:0015116', 'GO:0140776', 'GO:0140033', 'GO:0004563', 'GO:0051219', 'GO:0030492', 'GO:0051864', 'GO:0004022', 'GO:0035673', 'GO:0016898', 'GO:0016805', 'GO:0015106', 'GO:0140940', 'GO:0140135', 'GO:0140906', 'GO:0016416', 'GO:0015112', 'GO:0004084', 'GO:0034594', 'GO:0003730', 'GO:0005416', 'GO:0016759', 'GO:0035312', 'GO:0141050', 'GO:0015157', 'GO:0016703', 'GO:0033677', 'GO:0070492', 'GO:0003857', 'GO:0016247', 'GO:0015269', 'GO:0015175', 'GO:0019212', 'GO:0001217', 'GO:0052880', 'GO:0005179', 'GO:0141218', 'GO:0000182', 'GO:0005227', 'GO:0004731', 'GO:0004127', 'GO:0031369', 'GO:0016917', 'GO:0140161', 'GO:0005452', 'GO:0008381', 'GO:0046997', 'GO:0016215', 'GO:0034595', 'GO:0003865', 'GO:0140905', 'GO:0015928', 'GO:0016154', 'GO:0106019', 'GO:0070815', 'GO:0033931', 'GO:0004630', 'GO:0015315', 'GO:0003954', 'GO:0035591', 'GO:0017022', 'GO:0016794', 'GO:0016695', 'GO:0004709', 'GO:0016732', 'GO:0004040', 'GO:0008373', 'GO:0004864', 'GO:0004629', 'GO:0004753', 'GO:0015125', 'GO:0001025', 'GO:0140517', 'GO:0051499', 'GO:0015292', 'GO:0016827', 'GO:1901683', 'GO:0048030', 'GO:0008516', 'GO:0016632', 'GO:0046030', 'GO:0015666', 'GO:0050463', 'GO:0030572', 'GO:0099122', 'GO:0033164', 'GO:0045309', 'GO:0016454', 'GO:0120159', 'GO:0160041', 'GO:0008940', 'GO:0005385', 'GO:0034483', 'GO:0015210', 'GO:0034701', 'GO:0061459', 'GO:0016312', 'GO:0005186', 'GO:0016419', 'GO:0051377', 'GO:0140713', 'GO:0016636', 'GO:0017136', 'GO:0004114', 'GO:0005350', 'GO:0061665', 'GO:0015174', 'GO:0016716', 'GO:0141126', 'GO:0051903', 'GO:0016699', 'GO:0005539', 'GO:0005184', 'GO:0015930', 'GO:0015123', 'GO:0015391', 'GO:0120517', 'GO:0070735', 'GO:0015450', 'GO:0016647', 'GO:0043176', 'GO:0051018', 'GO:0010485', 'GO:0016899', 'GO:0004623', 'GO:0005217', 'GO:0106017', 'GO:0016972', 'GO:0015138', 'GO:0004844', 'GO:0015377', 'GO:0016997', 'GO:0042979', 'GO:0001163', 'GO:0008477', 'GO:0016002', 'GO:0005351', 'GO:0042085', 'GO:0008796', 'GO:0016843', 'GO:0030547', 'GO:0030058', 'GO:0017150', 'GO:0016750', 'GO:0004713', 'GO:0016435', 'GO:0031545', 'GO:0050263', 'GO:0022829', 'GO:0000179', 'GO:0048763', 'GO:0003714', 'GO:0016618', 'GO:0035014', 'GO:0046923', 'GO:0034318', 'GO:0004772', 'GO:0019210', 'GO:0015140', 'GO:0000033', 'GO:0008603', 'GO:0008481', 'GO:0008988', 'GO:0001664', 'GO:0015647', 'GO:0004029', 'GO:0031127', 'GO:0050839', 'GO:0003924', 'GO:0031559', 'GO:0010436', 'GO:0004683', 'GO:0016825', 'GO:0043733', 'GO:0061783', 'GO:0008515', 'GO:0009881', 'GO:0015203', 'GO:0034979', 'GO:0002054', 'GO:0003724', 'GO:0001046', 'GO:0141108', 'GO:0001091', 'GO:0015067', 'GO:0015278', 'GO:0016937', 'GO:0005283', 'GO:0000217'}\n",
      "Filtering out 322 terms from the training set.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/var/folders/ky/jqpqz88565l8qdqz4jf3387c0000gn/T/ipykernel_61963/2548981061.py:92: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  train_df[propagated_terms_col] = train_df[propagated_terms_col].apply(lambda terms: [term for term in terms if term not in terms_below_threshold])\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Filtering validation and test sets to remove terms below training threshold.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/var/folders/ky/jqpqz88565l8qdqz4jf3387c0000gn/T/ipykernel_61963/2548981061.py:97: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  val_df[propagated_terms_col] = val_df[propagated_terms_col].apply(lambda terms: [term for term in terms if term not in terms_below_threshold])\n",
      "/var/folders/ky/jqpqz88565l8qdqz4jf3387c0000gn/T/ipykernel_61963/2548981061.py:98: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  test_df[propagated_terms_col] = test_df[propagated_terms_col].apply(lambda terms: [term for term in terms if term not in terms_below_threshold])\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Filtered function training set terms with less than 50 occurrences: {}\n",
      "Filtered function training set total terms: 744\n",
      "Function training set shape: (289930, 14)\n",
      "Function validation set shape: (29188, 14)\n",
      "Function test set shape: (33343, 14)\n",
      "----\n"
     ]
    }
   ],
   "source": [
    "def prepare_data_and_split(go_terms_df, clusters_col='NA', terms_col='NA'):\n",
    "    # Mapping clusters to GO terms and vice versa\n",
    "    cluster_to_terms = defaultdict(list)\n",
    "    term_to_clusters = defaultdict(set)\n",
    "\n",
    "    for _, row in tqdm(go_terms_df.iterrows(), total=go_terms_df.shape[0], desc=\"Processing rows\"):\n",
    "        cluster, terms = row[clusters_col], row[terms_col]\n",
    "        cluster_to_terms[cluster].extend(terms)\n",
    "        for term in terms:\n",
    "            term_to_clusters[term].add(cluster)\n",
    "\n",
    "    # Identifying unique and multi-occurrence clusters\n",
    "    print(\"Identifying unique and multi-occurrence clusters\")\n",
    "\n",
    "    unique_clusters = {cluster for cluster, terms in cluster_to_terms.items()\n",
    "                       if all(len(term_to_clusters[term]) == 1 for term in terms)}\n",
    "    multi_clusters = set(cluster_to_terms.keys()) - unique_clusters\n",
    "\n",
    "    # Binarizing the GO terms\n",
    "    mlb = MultiLabelBinarizer()\n",
    "    all_terms = [terms for terms in cluster_to_terms.values()]\n",
    "    mlb.fit(all_terms)\n",
    "\n",
    "    # ShuffleSplit for training, validation, and test sets from multi occurrence clusters\n",
    "    print(\"ShuffleSplit for training, validation, and test sets from multi occurrence clusters\")\n",
    "    ss = ShuffleSplit(n_splits=1, test_size=0.1, random_state=38)\n",
    "    multi_clusters_list = list(multi_clusters)\n",
    "    train_val_idx, test_idx = next(ss.split(multi_clusters_list))\n",
    "\n",
    "    train_val_clusters = [multi_clusters_list[i] for i in train_val_idx]\n",
    "    ss_train_val = ShuffleSplit(n_splits=1, test_size=0.1, random_state=38)\n",
    "    train_idx, val_idx = next(ss_train_val.split(train_val_clusters))\n",
    "\n",
    "    train_clusters = [train_val_clusters[i] for i in train_idx] + list(unique_clusters)\n",
    "    val_clusters = [train_val_clusters[i] for i in val_idx]\n",
    "    test_clusters = [multi_clusters_list[i] for i in test_idx]\n",
    "\n",
    "    # Initial data extraction\n",
    "    print(\"Extracting rows for each set\")\n",
    "    train_df = go_terms_df[go_terms_df[clusters_col].isin(train_clusters)]\n",
    "    val_df = go_terms_df[go_terms_df[clusters_col].isin(val_clusters)]\n",
    "    test_df = go_terms_df[go_terms_df[clusters_col].isin(test_clusters)]\n",
    "\n",
    "    # Check and adjust for missing terms in training set\n",
    "    train_terms = set(sum(train_df[terms_col].tolist(), []))\n",
    "    val_terms = set(sum(val_df[terms_col].tolist(), []))\n",
    "    test_terms = set(sum(test_df[terms_col].tolist(), []))\n",
    "\n",
    "    missing_val_terms = val_terms - train_terms\n",
    "    missing_test_terms = test_terms - train_terms\n",
    "\n",
    "    # Move clusters with missing terms from val/test to train\n",
    "    print(\"Moving clusters with missing terms from validation/test to training\")\n",
    "    if missing_val_terms:\n",
    "        val_clusters_to_move = [cluster for cluster in val_clusters if any(term in missing_val_terms for term in cluster_to_terms[cluster])]\n",
    "        train_clusters.extend(val_clusters_to_move)\n",
    "        val_clusters = list(set(val_clusters) - set(val_clusters_to_move))\n",
    "\n",
    "    if missing_test_terms:\n",
    "        test_clusters_to_move = [cluster for cluster in test_clusters if any(term in missing_test_terms for term in cluster_to_terms[cluster])]\n",
    "        train_clusters.extend(test_clusters_to_move)\n",
    "        test_clusters = list(set(test_clusters) - set(test_clusters_to_move))\n",
    "\n",
    "    # Re-extract dataframes after adjustment\n",
    "    train_df = go_terms_df[go_terms_df[clusters_col].isin(train_clusters)]\n",
    "    val_df = go_terms_df[go_terms_df[clusters_col].isin(val_clusters)]\n",
    "    test_df = go_terms_df[go_terms_df[clusters_col].isin(test_clusters)]\n",
    "\n",
    "    #recalculate missing val/test terms\n",
    "    train_terms = set(sum(train_df[terms_col].tolist(), []))\n",
    "    val_terms = set(sum(val_df[terms_col].tolist(), []))\n",
    "    test_terms = set(sum(test_df[terms_col].tolist(), []))\n",
    "\n",
    "    missing_val_terms = val_terms - train_terms\n",
    "    missing_test_terms = test_terms - train_terms\n",
    "\n",
    "    print(\"Validation terms not in training:\", missing_val_terms)\n",
    "    print(\"Test terms not in training:\", missing_test_terms)\n",
    "\n",
    "    ###### \n",
    "    propagated_terms_col = 'Raw propagated GO terms'  # Replace with the actual column name if different\n",
    "\n",
    "    min_training_instances = 50 \n",
    "    # Filter out terms from the training set that don't meet the minimum instance threshold\n",
    "    train_terms_counter = Counter([term for terms in train_df[propagated_terms_col].tolist() for term in terms])\n",
    "    terms_below_threshold = {term for term, count in train_terms_counter.items() if count < min_training_instances}\n",
    "\n",
    "    print(f\"Terms below threshold ({min_training_instances} occurrences):\", terms_below_threshold)\n",
    "\n",
    "    if terms_below_threshold:\n",
    "        print(f\"Filtering out {len(terms_below_threshold)} terms from the training set.\")\n",
    "        train_df[propagated_terms_col] = train_df[propagated_terms_col].apply(lambda terms: [term for term in terms if term not in terms_below_threshold])\n",
    "        train_df = train_df[train_df[propagated_terms_col].apply(len) > 0]  # Remove rows with no terms after filtering\n",
    "\n",
    "    # Optionally, filter validation and test sets to remove terms below the training threshold\n",
    "    print(\"Filtering validation and test sets to remove terms below training threshold.\")\n",
    "    val_df[propagated_terms_col] = val_df[propagated_terms_col].apply(lambda terms: [term for term in terms if term not in terms_below_threshold])\n",
    "    test_df[propagated_terms_col] = test_df[propagated_terms_col].apply(lambda terms: [term for term in terms if term not in terms_below_threshold])\n",
    "\n",
    "    # Remove rows with no terms after filtering\n",
    "    val_df = val_df[val_df[propagated_terms_col].apply(len) > 0]\n",
    "    test_df = test_df[test_df[propagated_terms_col].apply(len) > 0]\n",
    "\n",
    "    # Check the results after filtering\n",
    "    filtered_train_counts = Counter([term for terms in train_df[propagated_terms_col].tolist() for term in terms])\n",
    "    filtered_train_counter_terms_less_than_X = {term: count for term, count in filtered_train_counts.items() if count < min_training_instances}\n",
    "    print(\"Filtered function training set terms with less than {} occurrences:\".format(min_training_instances), filtered_train_counter_terms_less_than_X)\n",
    "    print(\"Filtered function training set total terms:\", len(filtered_train_counts))\n",
    "\n",
    "    return train_df, val_df, test_df, mlb\n",
    "\n",
    "\n",
    "function_train_df, function_val_df, function_test_df, function_mlb = prepare_data_and_split(function_merged, clusters_col='Cluster_2', terms_col='Raw GO terms')\n",
    "print(\"Function training set shape:\", function_train_df.shape)\n",
    "print(\"Function validation set shape:\", function_val_df.shape)\n",
    "print(\"Function test set shape:\", function_test_df.shape)\n",
    "print(\"----\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 100,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Processing rows: 100%|██████████| 304567/304567 [00:08<00:00, 37979.03it/s]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Identifying unique and multi-occurrence clusters\n",
      "ShuffleSplit for training, validation, and test sets from multi occurrence clusters\n",
      "Extracting rows for each set\n",
      "Moving clusters with missing terms from validation/test to training\n",
      "Validation terms not in training: set()\n",
      "Test terms not in training: set()\n",
      "Terms below threshold (50 occurrences): {'GO:0005765', 'GO:0005775', 'GO:0008278', 'GO:0017053', 'GO:0033061', 'GO:0019897', 'GO:0065010', 'GO:0000775', 'GO:0005789', 'GO:0097729', 'GO:0030863', 'GO:0034705', 'GO:0000322', 'GO:0032116', 'GO:0031514', 'GO:0098802', 'GO:0038037', 'GO:0051286', 'GO:0042646', 'GO:0055028', 'GO:0005876', 'GO:0016459', 'GO:0005618', 'GO:1902687', 'GO:0098857', 'GO:0002178', 'GO:0070112', 'GO:0000164', 'GO:0097651', 'GO:0033655', 'GO:0009527', 'GO:0043190', 'GO:0030141', 'GO:0010009', 'GO:0031519', 'GO:0098636', 'GO:0097648', 'GO:0098858', 'GO:0043235', 'GO:0005657', 'GO:0034706', 'GO:0031010', 'GO:0030427', 'GO:1990391', 'GO:0036387', 'GO:0016465', 'GO:0044094', 'GO:0032156', 'GO:0031082', 'GO:0098562', 'GO:0030312', 'GO:0005737', 'GO:0005634', 'GO:0000785', 'GO:0090545', 'GO:0030055', 'GO:0071014', 'GO:0030076', 'GO:0000177', 'GO:0005798', 'GO:0090576', 'GO:0060187', 'GO:0043264', 'GO:1905862', 'GO:0030136', 'GO:0000783', 'GO:0043659', 'GO:0042709', 'GO:0005891', 'GO:0044297', 'GO:0098827', 'GO:0043230', 'GO:0005816', 'GO:0000235', 'GO:0043660', 'GO:0008076', 'GO:0000782', 'GO:0036125', 'GO:0000940', 'GO:0072379', 'GO:0000792', 'GO:0070419', 'GO:0044232', 'GO:0071782', 'GO:0005783', 'GO:0080008', 'GO:0005742', 'GO:0044796', 'GO:0009526', 'GO:0098576', 'GO:0030133'}\n",
      "Filtering out 91 terms from the training set.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/var/folders/ky/jqpqz88565l8qdqz4jf3387c0000gn/T/ipykernel_61963/2548981061.py:92: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  train_df[propagated_terms_col] = train_df[propagated_terms_col].apply(lambda terms: [term for term in terms if term not in terms_below_threshold])\n",
      "/var/folders/ky/jqpqz88565l8qdqz4jf3387c0000gn/T/ipykernel_61963/2548981061.py:97: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  val_df[propagated_terms_col] = val_df[propagated_terms_col].apply(lambda terms: [term for term in terms if term not in terms_below_threshold])\n",
      "/var/folders/ky/jqpqz88565l8qdqz4jf3387c0000gn/T/ipykernel_61963/2548981061.py:98: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  test_df[propagated_terms_col] = test_df[propagated_terms_col].apply(lambda terms: [term for term in terms if term not in terms_below_threshold])\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Filtering validation and test sets to remove terms below training threshold.\n",
      "Filtered function training set terms with less than 50 occurrences: {}\n",
      "Filtered function training set total terms: 255\n"
     ]
    }
   ],
   "source": [
    "component_train_df, component_val_df, component_test_df, component_mlb = prepare_data_and_split(component_merged, clusters_col='Cluster_2', terms_col='Raw GO terms')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 101,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Processing rows: 100%|██████████| 279033/279033 [00:06<00:00, 40265.36it/s]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Identifying unique and multi-occurrence clusters\n",
      "ShuffleSplit for training, validation, and test sets from multi occurrence clusters\n",
      "Extracting rows for each set\n",
      "Moving clusters with missing terms from validation/test to training\n",
      "Validation terms not in training: set()\n",
      "Test terms not in training: set()\n",
      "Terms below threshold (50 occurrences): {'GO:0006060', 'GO:0019835', 'GO:1901028', 'GO:0044406', 'GO:0051336', 'GO:0030335', 'GO:0043062', 'GO:0006116', 'GO:0033567', 'GO:0000103', 'GO:0009690', 'GO:1901159', 'GO:0098656', 'GO:0043447', 'GO:0015766', 'GO:0051056', 'GO:0006426', 'GO:0031589', 'GO:0017121', 'GO:0042399', 'GO:0046271', 'GO:0035725', 'GO:1902115', 'GO:0051231', 'GO:1904030', 'GO:0031554', 'GO:0070601', 'GO:0006864', 'GO:1903522', 'GO:0008016', 'GO:0009217', 'GO:0071709', 'GO:0042044', 'GO:0045595', 'GO:0010466', 'GO:0042173', 'GO:0052553', 'GO:0009303', 'GO:1900704', 'GO:0099046', 'GO:0050804', 'GO:0043954', 'GO:0033875', 'GO:0016310', 'GO:0006600', 'GO:0031365', 'GO:1901873', 'GO:0006801', 'GO:0030653', 'GO:0044088', 'GO:0045785', 'GO:0051446', 'GO:0034410', 'GO:0006523', 'GO:0120178', 'GO:0009146', 'GO:0042938', 'GO:0071900', 'GO:1903312', 'GO:0006208', 'GO:1903321', 'GO:0042127', 'GO:0010507', 'GO:0030334', 'GO:0035795', 'GO:1902410', 'GO:0140718', 'GO:0007548', 'GO:0042365', 'GO:0140467', 'GO:0006289', 'GO:0033619', 'GO:0043066', 'GO:0015879', 'GO:0051051', 'GO:0006671', 'GO:0046444', 'GO:0006064', 'GO:0051588', 'GO:0045123', 'GO:0044355', 'GO:0048608', 'GO:0001819', 'GO:0055062', 'GO:0010467', 'GO:0052372', 'GO:0035442', 'GO:0019043', 'GO:0006177', 'GO:0016441', 'GO:0015669', 'GO:0034411', 'GO:0031122', 'GO:0051495', 'GO:1901678', 'GO:0051126', 'GO:0051790', 'GO:0032456', 'GO:1901970', 'GO:0006468', 'GO:0006584', 'GO:0015947', 'GO:0071969', 'GO:0009306', 'GO:0009310', 'GO:0045337', 'GO:0001523', 'GO:0072340', 'GO:0010570', 'GO:0072598', 'GO:0008211', 'GO:0002573', 'GO:0061620', 'GO:0009612', 'GO:0052173', 'GO:0051094', 'GO:0070596', 'GO:0007169', 'GO:0009136', 'GO:0010476', 'GO:1990849', 'GO:0009953', 'GO:0051788', 'GO:0006829', 'GO:0141060', 'GO:0008204', 'GO:0009093', 'GO:0019401', 'GO:0006216', 'GO:0006077', 'GO:0050871', 'GO:0019376', 'GO:0072388', 'GO:0006706', 'GO:0046108', 'GO:0019585', 'GO:0046436', 'GO:0009237', 'GO:0009409', 'GO:0048580', 'GO:0019530', 'GO:0006415', 'GO:1905709', 'GO:0044872', 'GO:0097292', 'GO:1900373', 'GO:0035461', 'GO:0043549', 'GO:0048609', 'GO:0048513', 'GO:0051262', 'GO:0002694', 'GO:0099177', 'GO:0046928', 'GO:0018342', 'GO:0017001', 'GO:0051938', 'GO:0018200', 'GO:0019586', 'GO:0071763', 'GO:2001239', 'GO:0006308', 'GO:0031456', 'GO:0046484', 'GO:0044057', 'GO:1903320', 'GO:0050778', 'GO:0030810', 'GO:0001120', 'GO:0090069', 'GO:0002696', 'GO:0000098', 'GO:0009698', 'GO:2001293', 'GO:0072490', 'GO:0043570', 'GO:0043112', 'GO:0018201', 'GO:0090481', 'GO:0002685', 'GO:0032527', 'GO:0060296', 'GO:0046462', 'GO:0030968', 'GO:0042373', 'GO:2000220', 'GO:0000730', 'GO:0045981', 'GO:0030473', 'GO:0030307', 'GO:2000765', 'GO:0097354', 'GO:0017003', 'GO:0090169', 'GO:0016094', 'GO:0042374', 'GO:0010039', 'GO:0006497', 'GO:0050790', 'GO:0050731', 'GO:0006955', 'GO:0042157', 'GO:0006517', 'GO:0006693', 'GO:0019605', 'GO:0046461', 'GO:0016259', 'GO:0002700', 'GO:0050867', 'GO:0140205', 'GO:0065009', 'GO:0034053', 'GO:0009791', 'GO:0002130', 'GO:0016032', 'GO:0050654', 'GO:0033673', 'GO:0046452', 'GO:0032785', 'GO:2000222', 'GO:1900377', 'GO:0009120', 'GO:0019406', 'GO:0072351', 'GO:0006531', 'GO:0006857', 'GO:1903799', 'GO:0006491', 'GO:1901337', 'GO:0072338', 'GO:0002687', 'GO:0051300', 'GO:0046149', 'GO:0071966', 'GO:1903173', 'GO:0097190', 'GO:0071589', 'GO:0051642', 'GO:0030201', 'GO:0009617', 'GO:0052548', 'GO:0048468', 'GO:0030212', 'GO:1903726', 'GO:0019934', 'GO:0070784', 'GO:0046464', 'GO:0006901', 'GO:1901910', 'GO:0032881', 'GO:0032986', 'GO:0046070', 'GO:0030449', 'GO:0048582', 'GO:0051230', 'GO:0008217', 'GO:0048469', 'GO:0051301', 'GO:0018131', 'GO:1902905', 'GO:0009696', 'GO:0040011', 'GO:0097052', 'GO:0097164', 'GO:0071322', 'GO:0070880', 'GO:0070252', 'GO:0099024', 'GO:1902373', 'GO:1902751', 'GO:0070199', 'GO:0140053', 'GO:0006820', 'GO:0018210', 'GO:0009140', 'GO:0070598', 'GO:0006515', 'GO:0010594', 'GO:0015876', 'GO:0008589', 'GO:0051047', 'GO:0033212', 'GO:0030493', 'GO:0001692', 'GO:0097293', 'GO:0019630', 'GO:0019695', 'GO:0015884', 'GO:0045840', 'GO:0046373', 'GO:0141067', 'GO:0006434', 'GO:0000957', 'GO:0071453', 'GO:0098838', 'GO:0034964', 'GO:0046475', 'GO:0045026', 'GO:0060236', 'GO:0060147', 'GO:0051418', 'GO:0002699', 'GO:0009180', 'GO:0044837', 'GO:0002521', 'GO:0033609', 'GO:0022409', 'GO:0043446', 'GO:0019859', 'GO:0034486', 'GO:0141061', 'GO:0003008', 'GO:1901804', 'GO:1903083', 'GO:1902440', 'GO:0033015', 'GO:0043949', 'GO:0048259', 'GO:0000967', 'GO:0043132', 'GO:0046937', 'GO:0097485', 'GO:0015916', 'GO:0002831', 'GO:0042839', 'GO:0070813', 'GO:0000270', 'GO:0006797', 'GO:0032103', 'GO:0051316', 'GO:0006498', 'GO:0045922', 'GO:0120251', 'GO:0071591', 'GO:0034030', 'GO:0051791', 'GO:0016101', 'GO:0016131', 'GO:0034284', 'GO:1900430', 'GO:0071695', 'GO:0009615', 'GO:0046306', 'GO:0070900', 'GO:0051014', 'GO:0140404', 'GO:0048200', 'GO:0033542', 'GO:0000912', 'GO:0071577', 'GO:0042955', 'GO:0006050', 'GO:1900864', 'GO:0008361', 'GO:0015888', 'GO:0043201', 'GO:0048255', 'GO:0032944', 'GO:0051093', 'GO:0033993', 'GO:0046605', 'GO:0071495', 'GO:1901073', 'GO:2000631', 'GO:0015772', 'GO:0001933', 'GO:0015961', 'GO:1900369', 'GO:0034765', 'GO:0019216', 'GO:0070988', 'GO:0000052', 'GO:0042618', 'GO:0051785', 'GO:0030111', 'GO:0032309', 'GO:0035672', 'GO:0007389', 'GO:0009158', 'GO:0034367', 'GO:2000047', 'GO:0006499', 'GO:1902117', 'GO:0099402', 'GO:0009125', 'GO:0000019', 'GO:0071715', 'GO:0046513', 'GO:0006598', 'GO:0046457', 'GO:0032946', 'GO:0031455', 'GO:0010563', 'GO:0006436', 'GO:0050427', 'GO:0002691', 'GO:0046482', 'GO:0008284', 'GO:1904429', 'GO:0033979', 'GO:0006051', 'GO:1901796', 'GO:0097576', 'GO:0019677', 'GO:0010586', 'GO:0051239', 'GO:0042326', 'GO:0006692', 'GO:1903963', 'GO:0071459', 'GO:0045859', 'GO:0031347', 'GO:0002684', 'GO:0071139', 'GO:0046378', 'GO:0052547', 'GO:0016122', 'GO:0009061', 'GO:0015840', 'GO:0010959', 'GO:0051896', 'GO:0042844', 'GO:0043516', 'GO:0015749', 'GO:0120256', 'GO:1900619', 'GO:0019407', 'GO:0006059', 'GO:0052572', 'GO:0050671', 'GO:0009803', 'GO:2000145', 'GO:0046456', 'GO:0046333', 'GO:0042120', 'GO:0033967', 'GO:0046900', 'GO:0007154', 'GO:0006304', 'GO:0042402', 'GO:0097722', 'GO:0010322', 'GO:0060149', 'GO:0002833', 'GO:0070989', 'GO:0072396', 'GO:0044092', 'GO:0032917', 'GO:0042509', 'GO:0009169', 'GO:1901993', 'GO:0046278', 'GO:0009441', 'GO:0001112', 'GO:0034077', 'GO:0019649', 'GO:0052158', 'GO:0030036', 'GO:0034389', 'GO:2000147', 'GO:0000965', 'GO:2001234', 'GO:0044414', 'GO:0000422', 'GO:0061512', 'GO:0009190', 'GO:1901336', 'GO:0018973', 'GO:0071901', 'GO:0090567', 'GO:0042845', 'GO:0051055', 'GO:0001906', 'GO:0071786', 'GO:0008291', 'GO:0048367', 'GO:0120254', 'GO:0050864', 'GO:0002693', 'GO:0046520', 'GO:0071230', 'GO:0045833', 'GO:0009111', 'GO:0016138', 'GO:0043934', 'GO:0036294', 'GO:0006858', 'GO:0048753', 'GO:1901994', 'GO:0006528', 'GO:0071669', 'GO:0009888', 'GO:1903580', 'GO:0015791', 'GO:0003356', 'GO:0043939', 'GO:0031047', 'GO:0042371', 'GO:0000079', 'GO:0051647', 'GO:1901099', 'GO:0043543', 'GO:0002702', 'GO:0005980', 'GO:0052167', 'GO:0045923', 'GO:0006356', 'GO:0010821', 'GO:0019483', 'GO:1903722', 'GO:0007188', 'GO:0051291', 'GO:0006469', 'GO:0140354', 'GO:0042000', 'GO:0071170', 'GO:0006688', 'GO:0022412', 'GO:1901888', 'GO:2000278', 'GO:0052838', 'GO:0046107', 'GO:0046495', 'GO:0071375', 'GO:1901052', 'GO:0019326', 'GO:0051348', 'GO:0045921', 'GO:0009452', 'GO:0048638', 'GO:0016075', 'GO:0033866', 'GO:0006114', 'GO:0030182', 'GO:0001907', 'GO:0046099', 'GO:0044111', 'GO:0090435', 'GO:0015801', 'GO:0010324', 'GO:0043455', 'GO:0098609', 'GO:0030682', 'GO:0052652', 'GO:0019566', 'GO:0030155', 'GO:0031929', 'GO:0046397', 'GO:0090693', 'GO:0009094', 'GO:0070663', 'GO:0034285', 'GO:0001817', 'GO:0006936', 'GO:0010960', 'GO:0042783', 'GO:0045827', 'GO:0018377', 'GO:1900620', 'GO:0097551', 'GO:0046134', 'GO:0042744', 'GO:0006828', 'GO:0034502', 'GO:0006419', 'GO:0044127', 'GO:0003012', 'GO:0051282', 'GO:0019858', 'GO:1905475', 'GO:1901699', 'GO:1903828', 'GO:0035196', 'GO:0070070', 'GO:0016128', 'GO:1903564', 'GO:0006722', 'GO:1903305', 'GO:0034198', 'GO:0045048', 'GO:0071496', 'GO:0015846', 'GO:0046601', 'GO:0045861', 'GO:0034035', 'GO:0006744', 'GO:0110030', 'GO:0046189', 'GO:0046156', 'GO:0019482', 'GO:0000707', 'GO:0046599', 'GO:0009214', 'GO:1902019', 'GO:0046836', 'GO:0061615', 'GO:0001678', 'GO:0015880', 'GO:2000638', 'GO:0034963', 'GO:0019954', 'GO:0001192', 'GO:1900368', 'GO:0030177', 'GO:0036065', 'GO:0019860', 'GO:0044779', 'GO:0052170', 'GO:0046439', 'GO:0007020', 'GO:0170037', 'GO:0007163', 'GO:0071266', 'GO:0051249', 'GO:0060211', 'GO:0072676', 'GO:0043489', 'GO:0075015', 'GO:0046443', 'GO:0006054', 'GO:0070599', 'GO:0140415', 'GO:0016136', 'GO:0015810', 'GO:0009799', 'GO:0006937', 'GO:1902776', 'GO:0040012', 'GO:1905820', 'GO:0035966', 'GO:0009128', 'GO:0009106', 'GO:0032196', 'GO:0046379', 'GO:0018209', 'GO:1901004', 'GO:0019614', 'GO:0075136', 'GO:0032501', 'GO:0043122', 'GO:0042868', 'GO:0051283', 'GO:0043266', 'GO:0042423', 'GO:1905132', 'GO:0046346', 'GO:0016117', 'GO:0046396', 'GO:0052200', 'GO:0071792', 'GO:0036507', 'GO:2001237', 'GO:1903514', 'GO:0033388', 'GO:0042546', 'GO:0060294', 'GO:1900544', 'GO:0030908', 'GO:0051924', 'GO:0002718', 'GO:0090140', 'GO:0046822', 'GO:0016139', 'GO:0001525', 'GO:0030099', 'GO:0046058', 'GO:0001558', 'GO:0099098', 'GO:0006911', 'GO:0006997', 'GO:0043086', 'GO:0044001', 'GO:0009642', 'GO:0071331', 'GO:0017006', 'GO:0030417', 'GO:0006403', 'GO:0045911', 'GO:0015693', 'GO:0034032', 'GO:0034406', 'GO:0006670', 'GO:0070597', 'GO:0071326', 'GO:1902624', 'GO:0030575', 'GO:0032329', 'GO:1900140', 'GO:0015744', 'GO:0015959', 'GO:0009713', 'GO:1901874', 'GO:0043100', 'GO:1901022', 'GO:0033477', 'GO:0071028', 'GO:0060828', 'GO:0019626', 'GO:0098927', 'GO:0032868', 'GO:0006470', 'GO:0045776', 'GO:0043937', 'GO:1903131', 'GO:0061842', 'GO:2000045', 'GO:0019748', 'GO:0070920', 'GO:1903532', 'GO:0031400', 'GO:0017157', 'GO:0060964', 'GO:1903798', 'GO:0051278', 'GO:0030100', 'GO:0045117', 'GO:0071852', 'GO:0031221', 'GO:0007155', 'GO:0071674', 'GO:0019288', 'GO:0044145', 'GO:0042362', 'GO:0072334', 'GO:0071072', 'GO:1900428', 'GO:0019755', 'GO:0006282', 'GO:0051251', 'GO:0051657', 'GO:0050865', 'GO:0048666', 'GO:0071168', 'GO:0043903', 'GO:0006041', 'GO:0021700', 'GO:0052027', 'GO:0030979', 'GO:0070879', 'GO:0006681', 'GO:0046135', 'GO:0015774', 'GO:1900376', 'GO:0072596', 'GO:2000389', 'GO:0051127', 'GO:0032204', 'GO:0051346', 'GO:1903725', 'GO:0005988', 'GO:0009151', 'GO:1904062', 'GO:0019448', 'GO:0019634', 'GO:0009204', 'GO:0030174', 'GO:0006446', 'GO:0051338', 'GO:0071934', 'GO:0019217', 'GO:1904029', 'GO:0006360', 'GO:0098658', 'GO:0034067', 'GO:0032101', 'GO:0055078', 'GO:0007276', 'GO:0034036', 'GO:0016134', 'GO:0009668', 'GO:0051255', 'GO:0044149', 'GO:0009233', 'GO:0042439', 'GO:0044417', 'GO:1902622', 'GO:0010824', 'GO:0019747', 'GO:0006658', 'GO:0080171', 'GO:0006915', 'GO:0031032', 'GO:0061572', 'GO:0002920', 'GO:0030042', 'GO:0015800', 'GO:0009699', 'GO:0002697', 'GO:0043434', 'GO:0060567', 'GO:0003002', 'GO:0009798', 'GO:0065001', 'GO:0010411', 'GO:0071071', 'GO:0048646', 'GO:2000243', 'GO:0051715', 'GO:0030638', 'GO:0080164', 'GO:0061492', 'GO:0009195', 'GO:0140418', 'GO:1903371', 'GO:0090224', 'GO:0006879', 'GO:0048308', 'GO:0046718', 'GO:0006636', 'GO:0050776', 'GO:0000915', 'GO:0052562', 'GO:0098869', 'GO:0097193', 'GO:0006007', 'GO:0042424', 'GO:0042304', 'GO:1904872', 'GO:0006787', 'GO:0060965', 'GO:0030178', 'GO:0071216', 'GO:0001574', 'GO:0006891', 'GO:0016477', 'GO:0006684', 'GO:0019740', 'GO:0009855', 'GO:0051240', 'GO:0062013', 'GO:0006846', 'GO:0019348', 'GO:0046902', 'GO:0009749', 'GO:2001236', 'GO:1902103', 'GO:1901160', 'GO:0140590', 'GO:0048731', 'GO:0001325', 'GO:0009719', 'GO:0009657', 'GO:1905133', 'GO:0002682', 'GO:0061687', 'GO:0060285', 'GO:0006063', 'GO:0055070', 'GO:0050730', 'GO:0045912', 'GO:2000779', 'GO:0071921', 'GO:0044550', 'GO:0060967', 'GO:0051209', 'GO:1902767', 'GO:0009743', 'GO:0015838', 'GO:0031640', 'GO:0097305', 'GO:0089718', 'GO:0034315', 'GO:0042330', 'GO:0030866', 'GO:0006075', 'GO:0000947', 'GO:0019374', 'GO:0007063', 'GO:0050670', 'GO:1904407', 'GO:0051177', 'GO:0035498', 'GO:0018974', 'GO:1901440', 'GO:0032870', 'GO:0010029', 'GO:0001897', 'GO:0045428', 'GO:0019572', 'GO:0010565', 'GO:0019694', 'GO:0034407', 'GO:0019478', 'GO:0030978', 'GO:1901342', 'GO:0070637', 'GO:0009268', 'GO:0010508', 'GO:0046087', 'GO:0070595', 'GO:0006735', 'GO:0016077', 'GO:0031055', 'GO:0030091', 'GO:1905634', 'GO:0019276', 'GO:0051180', 'GO:0072387', 'GO:0032387', 'GO:0015697', 'GO:0042415', 'GO:2000026', 'GO:0046263', 'GO:0015828', 'GO:1903830', 'GO:0009215', 'GO:0051645', 'GO:0044409', 'GO:0042417', 'GO:0071034', 'GO:0009725', 'GO:0010037', 'GO:0042692', 'GO:0007411', 'GO:0009812', 'GO:0071041', 'GO:0007097', 'GO:0045927', 'GO:0044650', 'GO:0098704', 'GO:0034033', 'GO:0060295', 'GO:0031396', 'GO:0032042', 'GO:0042178', 'GO:0044396', 'GO:1905508', 'GO:0055075', 'GO:0034086', 'GO:0090158', 'GO:0051103', 'GO:0043401', 'GO:1902644', 'GO:0000184', 'GO:0045936', 'GO:0002128', 'GO:0015995', 'GO:0040008', 'GO:0001666', 'GO:0035349', 'GO:0019543', 'GO:0009756', 'GO:0015948', 'GO:1902101', 'GO:0019594', 'GO:1901805', 'GO:0060966', 'GO:0070358', 'GO:0031407', 'GO:0023052', 'GO:0051299', 'GO:0016129', 'GO:0019568', 'GO:2000105', 'GO:1902102', 'GO:0046125', 'GO:1903513', 'GO:0014706', 'GO:0019279', 'GO:0000165', 'GO:0031349', 'GO:0043069', 'GO:0051303', 'GO:0055090', 'GO:0044873', 'GO:0045088', 'GO:0090257', 'GO:0030154', 'GO:1901798', 'GO:0006074', 'GO:0140568', 'GO:0072491', 'GO:0022407', 'GO:0046854', 'GO:0046068', 'GO:0061718', 'GO:0090033', 'GO:0062014', 'GO:0072531', 'GO:0031175', 'GO:1901906', 'GO:1902777', 'GO:0070296', 'GO:0006942', 'GO:0046889', 'GO:0040017', 'GO:0046084', 'GO:0009155', 'GO:0033865', 'GO:0000963', 'GO:0045834', 'GO:1903533', 'GO:1901995', 'GO:0030952', 'GO:0043248', 'GO:1901891', 'GO:0015975', 'GO:0051260', 'GO:0032107', 'GO:0070786', 'GO:0046361', 'GO:1904356', 'GO:0042982', 'GO:1904461', 'GO:0016076', 'GO:0046260', 'GO:1901908', 'GO:0009222', 'GO:0042853', 'GO:0009832', 'GO:0046890', 'GO:0015743', 'GO:1901441', 'GO:0072402', 'GO:0048856', 'GO:0007263', 'GO:0019418', 'GO:0036508', 'GO:0070665', 'GO:0090559', 'GO:0042939', 'GO:0006096', 'GO:0051167', 'GO:0016127', 'GO:0071988', 'GO:0045332', 'GO:0009746', 'GO:0030888', 'GO:1905323', 'GO:0042446', 'GO:0061640', 'GO:1902369', 'GO:0048033', 'GO:0060537', 'GO:0000955', 'GO:1902807', 'GO:0046133', 'GO:0050900', 'GO:0020027', 'GO:0001934', 'GO:0015780', 'GO:0045740', 'GO:0035967', 'GO:1902407', 'GO:0061678', 'GO:0008207', 'GO:0106259', 'GO:0090646', 'GO:0052042', 'GO:0072678'}\n",
      "Filtering out 1050 terms from the training set.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/var/folders/ky/jqpqz88565l8qdqz4jf3387c0000gn/T/ipykernel_61963/2548981061.py:92: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  train_df[propagated_terms_col] = train_df[propagated_terms_col].apply(lambda terms: [term for term in terms if term not in terms_below_threshold])\n",
      "/var/folders/ky/jqpqz88565l8qdqz4jf3387c0000gn/T/ipykernel_61963/2548981061.py:97: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  val_df[propagated_terms_col] = val_df[propagated_terms_col].apply(lambda terms: [term for term in terms if term not in terms_below_threshold])\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Filtering validation and test sets to remove terms below training threshold.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/var/folders/ky/jqpqz88565l8qdqz4jf3387c0000gn/T/ipykernel_61963/2548981061.py:98: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  test_df[propagated_terms_col] = test_df[propagated_terms_col].apply(lambda terms: [term for term in terms if term not in terms_below_threshold])\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Filtered function training set terms with less than 50 occurrences: {}\n",
      "Filtered function training set total terms: 1548\n"
     ]
    }
   ],
   "source": [
    "process_train_df, process_val_df, process_test_df, process_mlb = prepare_data_and_split(process_merged, clusters_col='Cluster_2', terms_col='Raw GO terms')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 104,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total number of GO terms in the function training set: 3096062\n",
      "Total number of unique GO terms in the function training set: 744\n"
     ]
    }
   ],
   "source": [
    "#how many GO terms are there in the propogated colmmn for function train\n",
    "temp_terms = function_train_df['Raw propagated GO terms'].tolist()\n",
    "#flatten the list\n",
    "temp_terms = [item for sublist in temp_terms for item in sublist]\n",
    "print(\"Total number of GO terms in the function training set:\", len(temp_terms))\n",
    "#unique terms\n",
    "print(\"Total number of unique GO terms in the function training set:\", len(set(temp_terms)))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 64,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Terms in validation set but not in training set: set()\n",
      "Terms in test set but not in training set: set()\n"
     ]
    }
   ],
   "source": [
    "#are there any terms that are not in the training set but are in the validation set and test set\n",
    "\n",
    "temp_terms_train = function_train_df['Propagated GO terms'].tolist()\n",
    "temp_terms_train = [item for sublist in temp_terms_train for item in sublist]\n",
    "temp_terms_val = function_val_df['Propagated GO terms'].tolist()\n",
    "temp_terms_val = [item for sublist in temp_terms_val for item in sublist]\n",
    "temp_terms_test = function_test_df['Propagated GO terms'].tolist()\n",
    "temp_terms_test = [item for sublist in temp_terms_test for item in sublist]\n",
    "\n",
    "print(\"Terms in validation set but not in training set:\", set(temp_terms_val) - set(temp_terms_train))\n",
    "print(\"Terms in test set but not in training set:\", set(temp_terms_test) - set(temp_terms_train))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 106,
   "metadata": {},
   "outputs": [],
   "source": [
    "os.chdir(\"/PFP_PUBLISH/processed_data_90_30\")\n",
    "function_train_df.to_csv(\"function_train.tsv\", sep=\"\\t\", index=False)\n",
    "function_val_df.to_csv(\"function_val.tsv\", sep=\"\\t\", index=False)\n",
    "function_test_df.to_csv(\"function_test.tsv\", sep=\"\\t\", index=False)\n",
    "\n",
    "component_train_df.to_csv(\"component_train.tsv\", sep=\"\\t\", index=False)\n",
    "component_val_df.to_csv(\"component_val.tsv\", sep=\"\\t\", index=False)\n",
    "component_test_df.to_csv(\"component_test.tsv\", sep=\"\\t\", index=False)\n",
    "\n",
    "process_train_df.to_csv(\"process_train.tsv\", sep=\"\\t\", index=False)\n",
    "process_val_df.to_csv(\"process_val.tsv\", sep=\"\\t\", index=False)\n",
    "process_test_df.to_csv(\"process_test.tsv\", sep=\"\\t\", index=False)\n",
    "\n",
    "#write file that has contains how many unique GO terms are in the training set\n",
    "function_train_terms = function_train_df['Raw propagated GO terms'].tolist()\n",
    "function_train_terms = [item for sublist in function_train_terms for item in sublist]\n",
    "component_train_terms = component_train_df['Raw propagated GO terms'].tolist()\n",
    "component_train_terms = [item for sublist in component_train_terms for item in sublist]\n",
    "process_train_terms = process_train_df['Raw propagated GO terms'].tolist()\n",
    "process_train_terms = [item for sublist in process_train_terms for item in sublist]\n",
    "function_train_terms_unique = len(set(function_train_terms))\n",
    "component_train_terms_unique = len(set(component_train_terms))\n",
    "process_train_terms_unique = len(set(process_train_terms))\n",
    "\n",
    "with open(\"number_of_terms_trained_on.txt\", \"w\") as f:\n",
    "    print(\"Function:\", len(set(function_train_terms)), file=f)\n",
    "    print(\"Component:\", len(set(component_train_terms)), file=f)\n",
    "    print(\"Process:\", len(set(process_train_terms)), file=f)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 107,
   "metadata": {},
   "outputs": [],
   "source": [
    "with open(\"df_shapes.txt\", \"w\") as f:\n",
    "    print(\"Function train:\", function_train_df.shape, file=f)\n",
    "    print(\"Function val:\", function_val_df.shape, file=f)\n",
    "    print(\"Function test:\", function_test_df.shape, file=f)\n",
    "    print(\"Component train:\", component_train_df.shape, file=f)\n",
    "    print(\"Component val:\", component_val_df.shape, file=f)\n",
    "    print(\"Component test:\", component_test_df.shape, file=f)\n",
    "    print(\"Process train:\", process_train_df.shape, file=f)\n",
    "    print(\"Process val:\", process_val_df.shape, file=f)\n",
    "    print(\"Process test:\", process_test_df.shape, file=f)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 110,
   "metadata": {},
   "outputs": [],
   "source": [
    "function_merged.to_csv(\"presplit_function_merged.tsv\", sep=\"\\t\", index=False)\n",
    "component_merged.to_csv(\"presplit_component_merged.tsv\", sep=\"\\t\", index=False)\n",
    "process_merged.to_csv(\"presplit_process_merged.tsv\", sep=\"\\t\", index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 112,
   "metadata": {},
   "outputs": [],
   "source": [
    "assert len(function_train_df) + len(function_val_df) + len(function_test_df) == len(function_merged)\n",
    "assert len(component_train_df) + len(component_val_df) + len(component_test_df) == len(component_merged)\n",
    "assert len(process_train_df) + len(process_val_df) + len(process_test_df) == len(process_merged)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 220,
   "metadata": {},
   "outputs": [],
   "source": [
    "#create fasta files \n",
    "seq_records_function_train = create_seq_records(function_train_df)\n",
    "seq_records_function_val = create_seq_records(function_val_df)\n",
    "seq_records_function_test = create_seq_records(function_test_df)\n",
    "\n",
    "seq_records_component_train = create_seq_records(component_train_df)\n",
    "seq_records_component_val = create_seq_records(component_val_df)\n",
    "seq_records_component_test = create_seq_records(component_test_df)\n",
    "\n",
    "seq_records_process_train = create_seq_records(process_train_df)\n",
    "seq_records_process_val = create_seq_records(process_val_df)\n",
    "seq_records_process_test = create_seq_records(process_test_df)\n",
    "\n",
    "assert len(seq_records_function_train) == function_train_df.shape[0]\n",
    "assert len(seq_records_function_val) == function_val_df.shape[0]\n",
    "assert len(seq_records_function_test) == function_test_df.shape[0]\n",
    "\n",
    "assert len(seq_records_component_train) == component_train_df.shape[0]\n",
    "assert len(seq_records_component_val) == component_val_df.shape[0]\n",
    "assert len(seq_records_component_test) == component_test_df.shape[0]\n",
    "\n",
    "assert len(seq_records_process_train) == process_train_df.shape[0]\n",
    "assert len(seq_records_process_val) == process_val_df.shape[0]\n",
    "assert len(seq_records_process_test) == process_test_df.shape[0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 224,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "26746"
      ]
     },
     "execution_count": 224,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#write out the fasta files\n",
    "SeqIO.write(seq_records_function_train, \"function_train.fasta\", \"fasta\")\n",
    "SeqIO.write(seq_records_function_val, \"function_val.fasta\", \"fasta\")\n",
    "SeqIO.write(seq_records_function_test, \"function_test.fasta\", \"fasta\")\n",
    "\n",
    "SeqIO.write(seq_records_component_train, \"component_train.fasta\", \"fasta\")\n",
    "SeqIO.write(seq_records_component_val, \"component_val.fasta\", \"fasta\")\n",
    "SeqIO.write(seq_records_component_test, \"component_test.fasta\", \"fasta\")\n",
    "\n",
    "SeqIO.write(seq_records_process_train, \"process_train.fasta\", \"fasta\")\n",
    "SeqIO.write(seq_records_process_val, \"process_val.fasta\", \"fasta\")\n",
    "SeqIO.write(seq_records_process_test, \"process_test.fasta\", \"fasta\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Recreate ancestors dict with the terms trained on "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "This needs to be done since there are ancestors that are introduced that are not part of the unique GO terms. \n",
    "\n",
    "For example GO term X is present in the pre-propogated dataset. Y is a parent of X. Y is not a key then in the existing dict configuration. "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 238,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(744, 809, 2287)"
      ]
     },
     "execution_count": 238,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(set(function_train_terms)), len(component_mlb.classes_), len(process_mlb.classes_)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 245,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "100%|██████████| 15/15 [00:15<00:00,  1.01s/it]\n",
      "100%|██████████| 6/6 [00:05<00:00,  1.19it/s]\n",
      "100%|██████████| 31/31 [00:27<00:00,  1.12it/s]\n"
     ]
    }
   ],
   "source": [
    "batch_size = 50  # You can adjust the batch size here\n",
    "\n",
    "UNIQUE_FUNCTION_TERMS = list(set(function_train_terms))\n",
    "UNIQUE_COMPONENT_TERMS = list(set(component_train_terms))\n",
    "UNIQUE_PROCESS_TERMS = list(set(process_train_terms))\n",
    "\n",
    "trained_function_json_dict, trained_function_ancestors_dict = process_go_terms_in_batches(UNIQUE_FUNCTION_TERMS, batch_size=batch_size)\n",
    "assert len(trained_function_json_dict) == len(UNIQUE_FUNCTION_TERMS)\n",
    "\n",
    "trained_component_json_dict, trained_component_ancestors_dict = process_go_terms_in_batches(UNIQUE_COMPONENT_TERMS, batch_size=batch_size)\n",
    "assert len(trained_component_json_dict) == len(UNIQUE_COMPONENT_TERMS)\n",
    "\n",
    "trained_process_json_dict, trained_process_ancestors_dict = process_go_terms_in_batches(UNIQUE_PROCESS_TERMS, batch_size=batch_size)\n",
    "assert len(trained_process_json_dict) == len(UNIQUE_PROCESS_TERMS)\n",
    "\n",
    "with open('trained_function_ALL_data_ancestors_dict.json', 'w') as f:\n",
    "    json.dump(trained_function_ancestors_dict, f, indent=4)\n",
    "\n",
    "with open('trained_function_ALL_data_json_dict.json', 'w') as f:\n",
    "    json.dump(trained_function_json_dict, f, indent=4)\n",
    "\n",
    "with open('trained_component_ALL_data_ancestors_dict.json', 'w') as f:\n",
    "    json.dump(trained_component_ancestors_dict, f, indent=4)\n",
    "\n",
    "with open('trained_component_ALL_data_json_dict.json', 'w') as f:\n",
    "    json.dump(trained_component_json_dict, f, indent=4)\n",
    "\n",
    "with open('trained_process_ALL_data_ancestors_dict.json', 'w') as f:\n",
    "    json.dump(trained_process_ancestors_dict, f, indent=4)\n",
    "\n",
    "with open('trained_process_ALL_data_json_dict.json', 'w') as f:\n",
    "    json.dump(trained_process_json_dict, f, indent=4)\n",
    "\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Get embeddings"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 123,
   "metadata": {},
   "outputs": [],
   "source": [
    "#function numpy array\n",
    "function_all_np = np.load(\"TM_vec_embeddings/reduced_90_function_TM_Vec.npy\", allow_pickle=True)\n",
    "function_all_df = pd.read_csv(\"/Users/harsh/Documents/PFP_PUBLISH/raw_data_from_uniprot/reduced_90_function.tsv\", sep='\\t')\n",
    "assert function_all_np.shape[0] == function_all_df.shape[0]\n",
    "\n",
    "#component numpy array\n",
    "component_all_np = np.load(\"TM_vec_embeddings/reduced_90_component_TM_Vec.npy\", allow_pickle=True)\n",
    "component_all_df = pd.read_csv(\"/Users/harsh/Documents/PFP_PUBLISH/raw_data_from_uniprot/reduced_90_component.tsv\", sep='\\t')\n",
    "assert component_all_np.shape[0] == component_all_df.shape[0]\n",
    "\n",
    "#process numpy array\n",
    "process_all_np = np.load(\"TM_vec_embeddings/reduced_90_process_TM_Vec.npy\", allow_pickle=True)\n",
    "process_all_df = pd.read_csv(\"/Users/harsh/Documents/PFP_PUBLISH/raw_data_from_uniprot/reduced_90_process.tsv\", sep='\\t')\n",
    "assert process_all_np.shape[0] == process_all_df.shape[0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 124,
   "metadata": {},
   "outputs": [],
   "source": [
    "function_all_dict = dict(zip(function_all_df['Entry'], function_all_np))\n",
    "component_all_dict = dict(zip(component_all_df['Entry'], component_all_np))\n",
    "process_all_dict = dict(zip(process_all_df['Entry'], process_all_np))\n",
    "\n",
    "function_train_np = np.array([function_all_dict[entry] for entry in function_train_df['Entry']])\n",
    "function_val_np = np.array([function_all_dict[entry] for entry in function_val_df['Entry']])\n",
    "function_test_np = np.array([function_all_dict[entry] for entry in function_test_df['Entry']])\n",
    "\n",
    "component_train_np = np.array([component_all_dict[entry] for entry in component_train_df['Entry']])\n",
    "component_val_np = np.array([component_all_dict[entry] for entry in component_val_df['Entry']])\n",
    "component_test_np = np.array([component_all_dict[entry] for entry in component_test_df['Entry']])\n",
    "\n",
    "process_train_np = np.array([process_all_dict[entry] for entry in process_train_df['Entry']])\n",
    "process_val_np = np.array([process_all_dict[entry] for entry in process_val_df['Entry']])\n",
    "process_test_np = np.array([process_all_dict[entry] for entry in process_test_df['Entry']])\n",
    "\n",
    "assert function_train_np.shape[0] == function_train_df.shape[0]\n",
    "assert function_val_np.shape[0] == function_val_df.shape[0]\n",
    "assert function_test_np.shape[0] == function_test_df.shape[0]\n",
    "\n",
    "assert component_train_np.shape[0] == component_train_df.shape[0]\n",
    "assert component_val_np.shape[0] == component_val_df.shape[0]\n",
    "assert component_test_np.shape[0] == component_test_df.shape[0]\n",
    "\n",
    "assert process_train_np.shape[0] == process_train_df.shape[0]\n",
    "assert process_val_np.shape[0] == process_val_df.shape[0]\n",
    "assert process_test_np.shape[0] == process_test_df.shape[0]\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 126,
   "metadata": {},
   "outputs": [],
   "source": [
    "np.save(\"function_train.npy\", function_train_np)\n",
    "np.save(\"function_val.npy\", function_val_np)\n",
    "np.save(\"function_test.npy\", function_test_np)\n",
    "\n",
    "np.save(\"component_train.npy\", component_train_np)\n",
    "np.save(\"component_val.npy\", component_val_np)\n",
    "np.save(\"component_test.npy\", component_test_np)\n",
    "\n",
    "np.save(\"process_train.npy\", process_train_np)\n",
    "np.save(\"process_val.npy\", process_val_np)\n",
    "np.save(\"process_test.npy\", process_test_np)\n",
    "\n",
    "#file with expected shapes of embeddings \n",
    "with open(\"expected_shapes.txt\", \"w\") as f:\n",
    "    print(\"Function train:\", function_train_np.shape, file=f)\n",
    "    print(\"Function val:\", function_val_np.shape, file=f)\n",
    "    print(\"Function test:\", function_test_np.shape, file=f)\n",
    "    print(\"Component train:\", component_train_np.shape, file=f)\n",
    "    print(\"Component val:\", component_val_np.shape, file=f)\n",
    "    print(\"Component test:\", component_test_np.shape, file=f)\n",
    "    print(\"Process train:\", process_train_np.shape, file=f)\n",
    "    print(\"Process val:\", process_val_np.shape, file=f)\n",
    "    print(\"Process test:\", process_test_np.shape, file=f)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Work with the Plasmodium Sequences now "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 142,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total number of GO terms in the function training set: 744\n",
      "Total number of GO terms in the component training set: 255\n",
      "Total number of GO terms in the process training set: 1548\n"
     ]
    }
   ],
   "source": [
    "function_train_terms = function_train_df['Raw propagated GO terms'].tolist()\n",
    "function_train_terms = [item for sublist in function_train_terms for item in sublist]\n",
    "function_train_terms = list(set(function_train_terms))\n",
    "print(\"Total number of GO terms in the function training set:\", len(function_train_terms))\n",
    "\n",
    "component_train_terms = component_train_df['Raw propagated GO terms'].tolist()\n",
    "component_train_terms = [item for sublist in component_train_terms for item in sublist]\n",
    "component_train_terms = list(set(component_train_terms))\n",
    "print(\"Total number of GO terms in the component training set:\", len(component_train_terms))\n",
    "\n",
    "process_train_terms = process_train_df['Raw propagated GO terms'].tolist()\n",
    "process_train_terms = [item for sublist in process_train_terms for item in sublist]\n",
    "process_train_terms = list(set(process_train_terms))\n",
    "print(\"Total number of GO terms in the process training set:\", len(process_train_terms))\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 227,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(1328, 9)\n",
      "(1360, 9)\n",
      "(1413, 9)\n"
     ]
    }
   ],
   "source": [
    "#sequences for Plasmodium MA to embedd \n",
    "Plasmodium_MA_seqs = create_seq_records(Plasmodium_MA)\n",
    "SeqIO.write(Plasmodium_MA_seqs, \"Plasmodium_MA_ALL.fasta\", \"fasta\")\n",
    "\n",
    "Plasmodium_MA_function = Plasmodium_MA[Plasmodium_MA[\"Gene Ontology (molecular function)\"].notnull()]\n",
    "Plasmodium_MA_process = Plasmodium_MA[Plasmodium_MA[\"Gene Ontology (biological process)\"].notnull()]\n",
    "Plasmodium_MA_component = Plasmodium_MA[Plasmodium_MA[\"Gene Ontology (cellular component)\"].notnull()]\n",
    "\n",
    "print(Plasmodium_MA_function.shape)\n",
    "print(Plasmodium_MA_process.shape)\n",
    "print(Plasmodium_MA_component.shape)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 228,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/var/folders/ky/jqpqz88565l8qdqz4jf3387c0000gn/T/ipykernel_61963/2354573566.py:2: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  Plasmodium_MA_function['Raw GO terms'] = Plasmodium_MA_function['Gene Ontology (molecular function)'].str.findall(r\"GO:\\d+\")\n",
      "/var/folders/ky/jqpqz88565l8qdqz4jf3387c0000gn/T/ipykernel_61963/2354573566.py:3: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  Plasmodium_MA_process['Raw GO terms'] = Plasmodium_MA_process['Gene Ontology (biological process)'].str.findall(r\"GO:\\d+\")\n",
      "/var/folders/ky/jqpqz88565l8qdqz4jf3387c0000gn/T/ipykernel_61963/2354573566.py:4: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  Plasmodium_MA_component['Raw GO terms'] = Plasmodium_MA_component['Gene Ontology (cellular component)'].str.findall(r\"GO:\\d+\")\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Adding ancestors: 100%|██████████| 1328/1328 [00:00<00:00, 153842.89it/s]\n",
      "Adding ancestors: 100%|██████████| 1360/1360 [00:00<00:00, 214067.38it/s]\n",
      "Adding ancestors: 100%|██████████| 1413/1413 [00:00<00:00, 300382.74it/s]\n",
      "/var/folders/ky/jqpqz88565l8qdqz4jf3387c0000gn/T/ipykernel_61963/2354573566.py:10: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  Plasmodium_MA_function['Raw propagated GO terms'] = Plasmodium_MA_function_raw_terms_propogated\n",
      "/var/folders/ky/jqpqz88565l8qdqz4jf3387c0000gn/T/ipykernel_61963/2354573566.py:11: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  Plasmodium_MA_process['Raw propagated GO terms'] = Plasmodium_MA_process_raw_terms_propogated\n",
      "/var/folders/ky/jqpqz88565l8qdqz4jf3387c0000gn/T/ipykernel_61963/2354573566.py:12: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  Plasmodium_MA_component['Raw propagated GO terms'] = Plasmodium_MA_component_raw_terms_propogated\n"
     ]
    }
   ],
   "source": [
    "#propogate the annotations for each of the datasets\n",
    "Plasmodium_MA_function['Raw GO terms'] = Plasmodium_MA_function['Gene Ontology (molecular function)'].str.findall(r\"GO:\\d+\")\n",
    "Plasmodium_MA_process['Raw GO terms'] = Plasmodium_MA_process['Gene Ontology (biological process)'].str.findall(r\"GO:\\d+\")\n",
    "Plasmodium_MA_component['Raw GO terms'] = Plasmodium_MA_component['Gene Ontology (cellular component)'].str.findall(r\"GO:\\d+\")\n",
    "\n",
    "Plasmodium_MA_function_raw_terms_propogated = add_ancestors(list(Plasmodium_MA_function['Raw GO terms']), function_ancestors_dict)\n",
    "Plasmodium_MA_process_raw_terms_propogated = add_ancestors(list(Plasmodium_MA_process['Raw GO terms']), process_ancestors_dict)\n",
    "Plasmodium_MA_component_raw_terms_propogated = add_ancestors(list(Plasmodium_MA_component['Raw GO terms']), component_ancestors_dict)\n",
    "\n",
    "Plasmodium_MA_function['Raw propagated GO terms'] = Plasmodium_MA_function_raw_terms_propogated\n",
    "Plasmodium_MA_process['Raw propagated GO terms'] = Plasmodium_MA_process_raw_terms_propogated\n",
    "Plasmodium_MA_component['Raw propagated GO terms'] = Plasmodium_MA_component_raw_terms_propogated"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 232,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Unique terms in Plasmodium MA function propogated terms but not in the training set: 55\n",
      "Unique terms in Plasmodium MA process propogated terms but not in the training set: 233\n",
      "Unique terms in Plasmodium MA component propogated terms but not in the training set: 21\n"
     ]
    }
   ],
   "source": [
    "#In the propogated terms, how many unique terms are present there but not in the training set\n",
    "Plasmodium_MA_function_raw_terms_propogated_unique = Plasmodium_MA_function['Raw propagated GO terms'].tolist()\n",
    "Plasmodium_MA_function_raw_terms_propogated_unique = [item for sublist in Plasmodium_MA_function_raw_terms_propogated_unique for item in sublist]\n",
    "Plasmodium_MA_function_raw_terms_propogated_unique = list(set(Plasmodium_MA_function_raw_terms_propogated_unique))\n",
    "unique_terms_not_in_training_function = set(Plasmodium_MA_function_raw_terms_propogated_unique) - set(function_train_terms)\n",
    "print(\"Unique terms in Plasmodium MA function propogated terms but not in the training set:\", len(unique_terms_not_in_training_function))\n",
    "\n",
    "Plasmodium_MA_process_raw_terms_propogated = Plasmodium_MA_process['Raw propagated GO terms'].tolist()\n",
    "Plasmodium_MA_process_raw_terms_propogated = [item for sublist in Plasmodium_MA_process_raw_terms_propogated for item in sublist]\n",
    "Plasmodium_MA_process_raw_terms_propogated = list(set(Plasmodium_MA_process_raw_terms_propogated))\n",
    "unique_terms_not_in_training_process = set(Plasmodium_MA_process_raw_terms_propogated) - set(process_train_terms)\n",
    "print(\"Unique terms in Plasmodium MA process propogated terms but not in the training set:\", len(unique_terms_not_in_training_process))\n",
    "\n",
    "Plasmodium_MA_component_raw_terms_propogated = Plasmodium_MA_component['Raw propagated GO terms'].tolist()\n",
    "Plasmodium_MA_component_raw_terms_propogated = [item for sublist in Plasmodium_MA_component_raw_terms_propogated for item in sublist]\n",
    "Plasmodium_MA_component_raw_terms_propogated = list(set(Plasmodium_MA_component_raw_terms_propogated))\n",
    "unique_terms_not_in_training_component = set(Plasmodium_MA_component_raw_terms_propogated) - set(component_train_terms)\n",
    "print(\"Unique terms in Plasmodium MA component propogated terms but not in the training set:\", len(unique_terms_not_in_training_component))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 154,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/var/folders/ky/jqpqz88565l8qdqz4jf3387c0000gn/T/ipykernel_61963/997023908.py:2: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  Plasmodium_MA_function['Raw propagated GO terms'] = Plasmodium_MA_function['Raw propagated GO terms'].apply(lambda terms: [term for term in terms if term not in unique_terms_not_in_training_function])\n",
      "/var/folders/ky/jqpqz88565l8qdqz4jf3387c0000gn/T/ipykernel_61963/997023908.py:3: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  Plasmodium_MA_process['Raw propagated GO terms'] = Plasmodium_MA_process['Raw propagated GO terms'].apply(lambda terms: [term for term in terms if term not in unique_terms_not_in_training_function])\n",
      "/var/folders/ky/jqpqz88565l8qdqz4jf3387c0000gn/T/ipykernel_61963/997023908.py:4: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  Plasmodium_MA_component['Raw propagated GO terms'] = Plasmodium_MA_component['Raw propagated GO terms'].apply(lambda terms: [term for term in terms if term not in unique_terms_not_in_training_function])\n"
     ]
    }
   ],
   "source": [
    "# #filter out terms in raw propogated terms that are not in the unique_terms_not_in_training_function\n",
    "# Plasmodium_MA_function['Raw propagated GO terms'] = Plasmodium_MA_function['Raw propagated GO terms'].apply(lambda terms: [term for term in terms if term not in unique_terms_not_in_training_function])\n",
    "# Plasmodium_MA_process['Raw propagated GO terms'] = Plasmodium_MA_process['Raw propagated GO terms'].apply(lambda terms: [term for term in terms if term not in unique_terms_not_in_training_function])\n",
    "# Plasmodium_MA_component['Raw propagated GO terms'] = Plasmodium_MA_component['Raw propagated GO terms'].apply(lambda terms: [term for term in terms if term not in unique_terms_not_in_training_function])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 229,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Original shapes:\n",
      "Function: (1328, 11)\n",
      "Process: (1360, 11)\n",
      "Component: (1413, 11)\n",
      "After filtering out terms not in the training set:\n",
      "Function: (1328, 11)\n",
      "Process: (1355, 11)\n",
      "Component: (1410, 11)\n"
     ]
    }
   ],
   "source": [
    "#first print the original shape \n",
    "print(\"Original shapes:\")\n",
    "print(\"Function:\", Plasmodium_MA_function.shape)\n",
    "print(\"Process:\", Plasmodium_MA_process.shape)\n",
    "print(\"Component:\", Plasmodium_MA_component.shape)\n",
    "\n",
    "Plasmodium_MA_function = Plasmodium_MA_function[Plasmodium_MA_function['Raw propagated GO terms'].str.len() > 0]\n",
    "Plasmodium_MA_process = Plasmodium_MA_process[Plasmodium_MA_process['Raw propagated GO terms'].str.len() > 0]\n",
    "Plasmodium_MA_component = Plasmodium_MA_component[Plasmodium_MA_component['Raw propagated GO terms'].str.len() > 0]\n",
    "\n",
    "print(\"After filtering out terms not in the training set:\")\n",
    "print(\"Function:\", Plasmodium_MA_function.shape)\n",
    "print(\"Process:\", Plasmodium_MA_process.shape)\n",
    "print(\"Component:\", Plasmodium_MA_component.shape)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 230,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(1527, 512)\n"
     ]
    }
   ],
   "source": [
    "Plasmodium_MA_embedd_all = np.load(\"/Users/harsh/Documents/PFP_PUBLISH/processed_data_90_30/TM_vec_embeddings/Plasmodium_MA_TM_Vec.npy\", allow_pickle=True)\n",
    "print(Plasmodium_MA_embedd_all.shape)\n",
    "Plasmodium_MA_ids = Plasmodium_MA['Entry'].tolist()\n",
    "\n",
    "Plasmodium_MA_dict = dict(zip(Plasmodium_MA_ids, Plasmodium_MA_embedd_all))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 231,
   "metadata": {},
   "outputs": [],
   "source": [
    "#filter the dict based on the entries in the Plasmodium_MA_function, Plasmodium_MA_process, and Plasmodium_MA_component\n",
    "Plasmodium_MA_function_np = np.array([Plasmodium_MA_dict[entry] for entry in Plasmodium_MA_function['Entry']])\n",
    "Plasmodium_MA_process_np = np.array([Plasmodium_MA_dict[entry] for entry in Plasmodium_MA_process['Entry']])\n",
    "Plasmodium_MA_component_np = np.array([Plasmodium_MA_dict[entry] for entry in Plasmodium_MA_component['Entry']])\n",
    "\n",
    "assert Plasmodium_MA_function_np.shape[0] == Plasmodium_MA_function.shape[0]\n",
    "assert Plasmodium_MA_process_np.shape[0] == Plasmodium_MA_process.shape[0]\n",
    "assert Plasmodium_MA_component_np.shape[0] == Plasmodium_MA_component.shape[0]\n",
    "\n",
    "np.save(\"Plasmodium_MA_function.npy\", Plasmodium_MA_function_np)\n",
    "np.save(\"Plasmodium_MA_process.npy\", Plasmodium_MA_process_np)\n",
    "np.save(\"Plasmodium_MA_component.npy\", Plasmodium_MA_component_np)\n",
    "\n",
    "#save the dataframes\n",
    "Plasmodium_MA_function.to_csv(\"Plasmodium_MA_function.tsv\", sep=\"\\t\", index=False)\n",
    "Plasmodium_MA_process.to_csv(\"Plasmodium_MA_process.tsv\", sep=\"\\t\", index=False)\n",
    "Plasmodium_MA_component.to_csv(\"Plasmodium_MA_component.tsv\", sep=\"\\t\", index=False)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 233,
   "metadata": {},
   "outputs": [],
   "source": [
    "Plasmodium_MA_function_seqs = create_seq_records(Plasmodium_MA_function)\n",
    "Plasmodium_MA_process_seqs = create_seq_records(Plasmodium_MA_process)\n",
    "Plasmodium_MA_component_seqs = create_seq_records(Plasmodium_MA_component)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 234,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "1410"
      ]
     },
     "execution_count": 234,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "SeqIO.write(Plasmodium_MA_function_seqs, \"Plasmodium_MA_function.fasta\", \"fasta\")\n",
    "SeqIO.write(Plasmodium_MA_process_seqs, \"Plasmodium_MA_process.fasta\", \"fasta\")\n",
    "SeqIO.write(Plasmodium_MA_component_seqs, \"Plasmodium_MA_component.fasta\", \"fasta\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Generate IA TSV using the training data "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 212,
   "metadata": {},
   "outputs": [],
   "source": [
    "function_train_df_subset = total_SAR_function_GO[['Entry', 'Raw GO terms']].copy()\n",
    "function_train_df_subset.columns = ['EntryID', 'term']\n",
    "\n",
    "component_train_df_subset = total_SAR_component_GO[['Entry', 'Raw GO terms']].copy()\n",
    "component_train_df_subset.columns = ['EntryID', 'term']\n",
    "\n",
    "process_train_df_subset = total_SAR_process_GO[['Entry', 'Raw GO terms']].copy()\n",
    "process_train_df_subset.columns = ['EntryID', 'term']\n",
    "\n",
    "# Adding aspect to each respective dataframe\n",
    "function_train_df_subset['aspect'] = 'MFO'\n",
    "component_train_df_subset['aspect'] = 'CCO'\n",
    "process_train_df_subset['aspect'] = 'BPO'\n",
    "\n",
    "# Concatenating all dataframes to create a final combined dataframe\n",
    "combined_df = pd.concat([function_train_df_subset, component_train_df_subset, process_train_df_subset], ignore_index=True)\n",
    "\n",
    "exploded_df = combined_df.explode('term')\n",
    "\n",
    "exploded_df.head()\n",
    "\n",
    "#REMOVE ALL_OBSELETE_TERMS\n",
    "exploded_df = exploded_df[~exploded_df['term'].isin(ALL_OBSELETE_TERMS)]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 213,
   "metadata": {},
   "outputs": [],
   "source": [
    "exploded_df.to_csv(\"terms_for_IA_calculation.tsv\", sep=\"\\t\", index=False)\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "python ia.py -a /Users/harsh/Documents/PFP_PUBLISH/terms_for_IA_calculation.tsv -o /Users/harsh/Documents/PFP_PUBLISH/IA_all.tsv -g /Users/harsh/Documents/PFP_PUBLISH/go-basic.obo"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "base",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.12"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
